---
title: "Using Tweedieverse for differential abundance analysis of microbiome data"
author: "Himel Mallick"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Tweedieverse}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

This vignette highlights some example workflows for performing differential abundance analysis using `Tweedieverse` on microbiome counts or relative abundances (both taxonomic and functional profiles are supported). It should be noted that `Tweedieverse` is a sister package of [MaAsLin](https://huttenhower.sph.harvard.edu/maaslin/) which is developed by many of the same authors as this package and likewise, they share certain similarities but also important differences as described in the in-depth [tutorials](https://github.com/himelmallick/TweedieLabs).

## Install and load packages

```{r, warning=FALSE, message = FALSE}
# Load the Tweedieverse package
library(Tweedieverse)

# Load other essential packages
library(tidyverse)

```

## The Data

`Tweedieverse` requires two tab-delimited input files, one for microbial taxonomic or functional profiles (`input_features`) and one for metadata or covariates (`input_metadata`). The rows of the `input_features` correspond to features (e.g., microbial species, genes, or pathways) and the columns correspond to samples (and vice versa). Similarly, the columns of the `input_metadata` correspond to sample-specific covariates (e.g., disease status) and the rows correspond to samples.

The `input_features` file can contain samples not included in the `input_metadata` file (along with the reverse case) although it is expected that they have matching samples. For both cases, samples not included in either of the files will be removed from the analysis. Also samples do not need to be in the same order in the two files as `Tweedieverse` will automatically detect the right order based on the sample names (assuming common samples IDs across files).

For the purpose of this vignette, it is assumed that these two input data have already been quality-controlled with necessary preprocessing steps, e.g.,, as described in the [iHMP manuscript](https://www.nature.com/articles/s41586-019-1237-9). For demonstration purposes, in Example 2, we will be using a cleaned version of the iHMP (or HMP2) dataset available from the R/Bioconductor package [MaAsLin2](https://github.com/biobakery/Maaslin2/tree/master/inst/extdata). The original  dataset is available from the [Integrative Human Microbiome Project](https://www.hmpdacc.org/ihmp/). For Example 1, we will be using a synthetic dataset generated by [SparseDOSSA](https://github.com/biobakery/sparseDOSSA).

### Example 1 - Differential Abundance on Synthetic Microbiome Counts

Let's first simulate 100 synthetic metagenomes using [SparseDOSSA](https://github.com/biobakery/sparseDOSSA). By default, SparseDOSSA uses a pre-trained human microbiome dataset as a template (the calibration dataset sparseDOSSA trains its model on); however, the tool is environment-agnostic and can estimate community properties from any user-provided template data (e.g., non-human microbiome profiles). 

Here is a basic example of simulating a synthetic dataset with 200 features (2% spiked-in), 100 samples, and 1 binary metadata. 

```{r, warning=FALSE, message = FALSE}
# Install and load libraries (not required if SparseDOSSA is already installed)
library(devtools)
devtools::install_github('biobakery/sparseDOSSA@varyLibSize')
library(sparseDOSSA)
library(stringi)

# Specify simulation parameters 
n.microbes <- 200 # Number of features
n.samples <- 100 # Number of samples
spike.perc <- 0.02 # Percentage of spiked-in features
spikeStrength<-"20" # Effect size

# Specify binary metadata 
n.metadata <- 1 
UserMetadata<-as.matrix(rep(c(0,1), each=n.samples/2))
UserMetadata<-t(UserMetadata) # Transpose

# Spiked-in metadata (which metadata to spike-in) 
Metadatafrozenidx<-1
spikeCount<-as.character(length(Metadatafrozenidx))
significant_metadata<-paste('Metadata', Metadatafrozenidx, sep='')

# Generate SparseDOSSA synthetic abundances 
DD<-sparseDOSSA::sparseDOSSA(number_features = n.microbes,
                            number_samples = n.samples,
                            UserMetadata = UserMetadata,
                            Metadatafrozenidx = Metadatafrozenidx,
                            datasetCount = 1,
                            spikeCount = spikeCount,
                            spikeStrength = spikeStrength,
                            noZeroInflate = TRUE,
                            percent_spiked = spike.perc, 
                            seed = 1234)

# Gather SparseDOSSA output
sparsedossa_results <- as.data.frame(DD$OTU_count)
rownames(sparsedossa_results)<-sparsedossa_results$X1
sparsedossa_results<-sparsedossa_results[-1,-1]
colnames(sparsedossa_results)<-paste('Sample', 1:ncol(sparsedossa_results), sep='')
data<-as.matrix(sparsedossa_results[-c((n.metadata+1):(2*n.microbes+n.metadata)),])
data<-data.matrix(data)
class(data) <- "numeric"
truth<-c(unlist(DD$truth))
truth<-truth[!stri_detect_fixed(truth,":")]
truth<-truth[(5+n.metadata):length(truth)]
truth<-as.data.frame(truth)
significant_features<-truth[seq(1, (as.numeric(spikeCount)+1)*(n.microbes*spike.perc), (as.numeric(spikeCount)+1)),]
significant_features<-as.vector(significant_features)

# Remove intermediate files
file.remove('SyntheticMicrobiome-Counts.pcl')
file.remove('SyntheticMicrobiome.pcl')
file.remove('SyntheticMicrobiomeParameterFile.txt')

# Extract features table from SparseDOSSA output
features<-as.data.frame(t(data[-c(1:n.metadata),]))

# Extract metadata table from SparseDOSSA output
metadata<-as.data.frame(data[1,])
colnames(metadata)<-rownames(data)[1]
metadata$Metadata1<-as.character(metadata$Metadata1)

# Mark True Positive (TP) features
wh.TP = colnames(features) %in% significant_features
colnames(features)<-paste("Feature", 1:n.microbes, sep = "")
newname = paste0(colnames(features)[wh.TP], "_TP")
colnames(features)[wh.TP] <- newname;
colnames(features)[grep('TP', colnames(features))]
```

Executing the above code will generate a synthetic microbiome dataset with 100 samples and 200 features, among which 4 features (2%) are differentially abundant with respect to a simulated binary metadata. Note that, the `True Negative` features do not have `_TP` attached to their names. As a quick sanity check, we can plot the distributions of the SparseDOSSA-generated features. To this end, we first create a temporary data frame combining the feature of interest with the metadata to visualize the distribution using a grouped boxplot.

```{r message=FALSE, warning=FALSE}
# Create data frames to plot one TP feature and one TN feature
plot_data_TP<-cbind.data.frame(y = features[,'Feature172_TP'], metadata)
plot_data_TN<-cbind.data.frame(y = features[,'Feature3'], metadata)
colnames(plot_data_TP)<-c('Feature', 'Metadata')
colnames(plot_data_TN)<-c('Feature', 'Metadata')

# TP (We expect significant group differences here)
plot_data_TP %>% ggplot(aes(x = Metadata, y = Feature, fill = Metadata)) + geom_boxplot(outlier.shape = NA) + scale_y_log10() + ylab('True Positive Feature') + xlab('')

# TN (We do not expect significant group differences here)
plot_data_TN %>% ggplot(aes(x = Metadata, y = Feature, fill = Metadata)) + geom_boxplot(outlier.shape = NA) + scale_y_log10() + ylab('True Negative Feature') + xlab('')
```

As expected, we see significant group differences for the true positive features (`Feature46_TP`, `Feature58_TP`, `Feature165_TP`, `Feature172_TP`) but not for the rest. Now that we have validated SparseDOSSA, let's run Tweedieverse on this `'fake'` dataset (assuming that the absolute path of the output directory exists).

```{r, warning=FALSE, message = FALSE}
# Run Tweedieverse using the CPLM model
if (!dir.exists('demo_output')) dir.create('demo_output')
SimData<-Tweedieverse(features, 
                   metadata, 
                   output = './demo_output/SimData', # Assuming demo_output exists
                   base_model = 'CPLM',
                   standardize = FALSE) 
```

After computation, `SimData` is a data frame containing coefficient estimates, p-values, and q-values (multiplicity-adjusted p-values) along with other parameter estimates from the fitted per-feature models. By default, p-values are adjusted with the Benjamini-Hochberg method.


```{r}
# Table with DA analysis results - significant features at 5% FDR
SimData[SimData$qval<0.05, 1:7]
```

Using the adjusted p-values, there are 4 statistically significant differentially abundant features at 5% FDR, which matches exactly with the `ground truth` information from SparseDOSSA indicating that Tweedieverse is able to call out the potential true positive features while also maintaining a low false positive rate.

### Example 2 - Multivariable Association on HMP2 Baseline Microbiomes 

For the second example, we will analyze the microbiome taxonomic data from the [integrative Human Microbiome Project](https://www.hmpdacc.org/ihmp/), available from the [R/Bioconductor package MaAsLin2](https://github.com/biobakery/Maaslin2). Check out the [original paper](https://www.nature.com/articles/s41586-019-1237-9) for details on the related statistical modeling. Compared to Example 1, Example 2 is arguably complex with multiple covariates and repeated measures. Focusing on the HMP2 baseline samples, we will fit a zero-inflated model unlike Example 1. Following original publication, we specify age, antibiotic usage, and disease status (diagnosis) as fixed effects. 

```{r, warning=FALSE, message = FALSE}
# Load data 
library(data.table)
input_features <- fread("https://raw.githubusercontent.com/biobakery/Maaslin2/master/inst/extdata/HMP2_taxonomy.tsv", sep ="\t")
input_metadata <-fread("https://raw.githubusercontent.com/biobakery/Maaslin2/master/inst/extdata/HMP2_metadata.tsv", sep ="\t")

# Format data 
library(tibble)
input_features<- column_to_rownames(input_features, 'ID')
input_metadata<- column_to_rownames(input_metadata, 'ID')

# Extract baseline samples
input_metadata<-input_metadata[order(input_metadata$collection),] # Order to extract first sample

# Create baseline data 
metadata_baseline<- input_metadata[!duplicated(input_metadata$subject),]
metadata_baseline<- metadata_baseline[metadata_baseline$collection < 3,]

# Remove irrelevant columns from both 
metadata_baseline<-metadata_baseline %>% dplyr::select(age, antibiotics, diagnosis)
metadata_final<-metadata_baseline

# Remove missing data and set reference
metadata_final<- metadata_final[complete.cases(metadata_final), ]
metadata_final$diagnosis<-as.factor(metadata_final$diagnosis)
metadata_final$diagnosis<-relevel(metadata_final$diagnosis, ref = "nonIBD") #NonIBD as reference

# Adjust features accordingly 
features_final<-input_features[rownames(metadata_final),]

# Fit a ZICP model 
if (!dir.exists('demo_output')) dir.create('demo_output')
HMP2_base <- Tweedieverse(features_final, 
                     metadata_final,
                     output = './demo_output/HMP2_base', # Assuming demo_output exists
                     fixed_effects = c('age', 'antibiotics','diagnosis'),
                     base_model = 'ZICP',
                     max_significance = 0.25, # To match original study
                     adjust_offset = FALSE) # No offset as the values are relative abundances)

# Table with DA analysis results - significant features at 25% FDR with respect to diagnosis
nrow(HMP2_base[HMP2_base$metadata == 'diagnosis',])
```
The results, as stored in the data frame `HMP2_base`, indicate that there are 37 statistically significant differentially abundant features at 25% FDR including *Subdoligranulum unclassified* and *Bacteroides vulgatus*, both detected as significant in the original publication. Notably, the original publication failed to find significant taxonomic associations at the baseline level, which can be attributed to (i) low sample size, (ii) small effect size, and (iii) lack of detection power using linear models.

It is very satisfactory to observe that even at the baseline level (i.e. using a smaller sample size), Tweedieverse is able to detect several dysbiosis-associated species identified at the longitudinal level (i.e. using a larger sample size) in the original study.

### Example 3 - Multivariable Association on HMP2 Longitudinal Microbiomes 

As a final illustration, we will analyze the same iHMP dataset but without restricting ourselves to baseline samples. Following the original publication, we will include subjects and recruitment sites as random effects to account for the correlations in the repeated measures and model each feature as a function of diagnosis (a categorical variable with non-IBD as the reference group) and dysbiosis state as a nested binary variable (with non-dysbiotic as reference) within each IBD phenotype (UC, CD, and non-IBD), while adjusting for consent age as a continuous covariate, and antibiotics as as binary covariate.

```r

#############
# Load data #
#############

library(data.table)
input_features <- fread("https://raw.githubusercontent.com/biobakery/Maaslin2/master/inst/extdata/HMP2_taxonomy.tsv", sep ="\t")
input_metadata <-fread("https://raw.githubusercontent.com/biobakery/Maaslin2/master/inst/extdata/HMP2_metadata.tsv", sep ="\t")

###############
# Format data #
###############

library(tibble)
input_features<- column_to_rownames(input_features, 'ID')
input_metadata<- column_to_rownames(input_metadata, 'ID')

# Remove missing data and set reference
metadata_final<- input_metadata[complete.cases(input_metadata), ]
metadata_final$diagnosis<-as.factor(metadata_final$diagnosis)
metadata_final$diagnosis<-relevel(metadata_final$diagnosis, ref = "nonIBD") #NonIBD as reference

# Adjust features accordingly 
features_final<-input_features[rownames(metadata_final),]

#################################
# Fit a CPLM mixed effect model #
#################################

if (!dir.exists('demo_output')) dir.create('demo_output')
HMP2_long <- Tweedieverse(features_final, 
                          metadata_final,
                          output = './demo_output/HMP2_long', # Assuming demo_output exists
                          fixed_effects = c('diagnosis', 'dysbiosisnonIBD','dysbiosisUC','dysbiosisCD', 'antibiotics', 'age'),
                          random_effects = c('subject', 'site'),
                          base_model = 'CPLM',
                          adjust_offset = FALSE,
                          reference = c('diagnosis,nonIBD')) 


```
The results from this analysis (`HMP2_long`) indicate that the primary conclusion of the original study is largely recapitulated which can be broadly summarized as **increase in facultative anaerobes including *E. coli* at the expense of anaerobic SCFA producers including *F. prausnitzi***.

## Session information

```{r}
sessionInfo()
```