<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Himel Mallick" />

<meta name="date" content="2022-02-26" />

<title>Using Tweedieverse for differential abundance analysis of microbiome data</title>


<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Using Tweedieverse for differential abundance analysis of microbiome data</h1>
<h4 class="author">Himel Mallick</h4>
<h4 class="date">2022-02-26</h4>



<p>This vignette highlights some example workflows for performing differential abundance analysis using <code>Tweedieverse</code> on microbiome counts or relative abundances (both taxonomic and functional profiles are supported). It should be noted that <code>Tweedieverse</code> is a sister package of <a href="https://huttenhower.sph.harvard.edu/maaslin/">MaAsLin</a> which is developed by many of the same authors as this package and likewise, they share certain similarities but also important differences as described in the in-depth <a href="https://github.com/himelmallick/TweedieLabs">tutorials</a>.</p>
<div id="install-and-load-packages" class="section level2">
<h2>Install and load packages</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co"># Load the Tweedieverse package</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">library</span>(Tweedieverse)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="co"># Load other essential packages</span></a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">library</span>(tidyverse)</a></code></pre></div>
</div>
<div id="the-data" class="section level2">
<h2>The Data</h2>
<p><code>Tweedieverse</code> requires two tab-delimited input files, one for microbial taxonomic or functional profiles (<code>input_features</code>) and one for metadata or covariates (<code>input_metadata</code>). The rows of the <code>input_features</code> correspond to features (e.g., microbial species, genes, or pathways) and the columns correspond to samples (and vice versa). Similarly, the columns of the <code>input_metadata</code> correspond to sample-specific covariates (e.g., disease status) and the rows correspond to samples.</p>
<p>The <code>input_features</code> file can contain samples not included in the <code>input_metadata</code> file (along with the reverse case) although it is expected that they have matching samples. For both cases, samples not included in either of the files will be removed from the analysis. Also samples do not need to be in the same order in the two files as <code>Tweedieverse</code> will automatically detect the right order based on the sample names (assuming common samples IDs across files).</p>
<p>For the purpose of this vignette, it is assumed that these two input data have already been quality-controlled with necessary preprocessing steps, e.g.,, as described in the <a href="https://www.nature.com/articles/s41586-019-1237-9">iHMP manuscript</a>. For demonstration purposes, in Example 2, we will be using a cleaned version of the iHMP (or HMP2) dataset available from the R/Bioconductor package <a href="https://github.com/biobakery/Maaslin2/tree/master/inst/extdata">MaAsLin2</a>. The original dataset is available from the <a href="https://www.hmpdacc.org/ihmp/">Integrative Human Microbiome Project</a>. For Example 1, we will be using a synthetic dataset generated by <a href="https://github.com/biobakery/sparseDOSSA">SparseDOSSA</a>.</p>
<div id="example-1---differential-abundance-on-synthetic-microbiome-counts" class="section level3">
<h3>Example 1 - Differential Abundance on Synthetic Microbiome Counts</h3>
<p>Letâ€™s first simulate 100 synthetic metagenomes using <a href="https://github.com/biobakery/sparseDOSSA">SparseDOSSA</a>. By default, SparseDOSSA uses a pre-trained human microbiome dataset as a template (the calibration dataset sparseDOSSA trains its model on); however, the tool is environment-agnostic and can estimate community properties from any user-provided template data (e.g., non-human microbiome profiles).</p>
<p>Here is a basic example of simulating a synthetic dataset with 200 features (2% spiked-in), 100 samples, and 1 binary metadata.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># Install and load libraries (not required if SparseDOSSA is already installed)</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw">library</span>(devtools)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">devtools<span class="op">::</span><span class="kw">install_github</span>(<span class="st">'biobakery/sparseDOSSA@varyLibSize'</span>)</a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="kw">library</span>(sparseDOSSA)</a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="kw">library</span>(stringi)</a>
<a class="sourceLine" id="cb2-6" data-line-number="6"></a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="co"># Specify simulation parameters </span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8">n.microbes &lt;-<span class="st"> </span><span class="dv">200</span> <span class="co"># Number of features</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9">n.samples &lt;-<span class="st"> </span><span class="dv">100</span> <span class="co"># Number of samples</span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10">spike.perc &lt;-<span class="st"> </span><span class="fl">0.02</span> <span class="co"># Percentage of spiked-in features</span></a>
<a class="sourceLine" id="cb2-11" data-line-number="11">spikeStrength&lt;-<span class="st">&quot;20&quot;</span> <span class="co"># Effect size</span></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"></a>
<a class="sourceLine" id="cb2-13" data-line-number="13"><span class="co"># Specify binary metadata </span></a>
<a class="sourceLine" id="cb2-14" data-line-number="14">n.metadata &lt;-<span class="st"> </span><span class="dv">1</span> </a>
<a class="sourceLine" id="cb2-15" data-line-number="15">UserMetadata&lt;-<span class="kw">as.matrix</span>(<span class="kw">rep</span>(<span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="dt">each=</span>n.samples<span class="op">/</span><span class="dv">2</span>))</a>
<a class="sourceLine" id="cb2-16" data-line-number="16">UserMetadata&lt;-<span class="kw">t</span>(UserMetadata) <span class="co"># Transpose</span></a>
<a class="sourceLine" id="cb2-17" data-line-number="17"></a>
<a class="sourceLine" id="cb2-18" data-line-number="18"><span class="co"># Spiked-in metadata (which metadata to spike-in) </span></a>
<a class="sourceLine" id="cb2-19" data-line-number="19">Metadatafrozenidx&lt;-<span class="dv">1</span></a>
<a class="sourceLine" id="cb2-20" data-line-number="20">spikeCount&lt;-<span class="kw">as.character</span>(<span class="kw">length</span>(Metadatafrozenidx))</a>
<a class="sourceLine" id="cb2-21" data-line-number="21">significant_metadata&lt;-<span class="kw">paste</span>(<span class="st">'Metadata'</span>, Metadatafrozenidx, <span class="dt">sep=</span><span class="st">''</span>)</a>
<a class="sourceLine" id="cb2-22" data-line-number="22"></a>
<a class="sourceLine" id="cb2-23" data-line-number="23"><span class="co"># Generate SparseDOSSA synthetic abundances </span></a>
<a class="sourceLine" id="cb2-24" data-line-number="24">DD&lt;-sparseDOSSA<span class="op">::</span><span class="kw">sparseDOSSA</span>(<span class="dt">number_features =</span> n.microbes,</a>
<a class="sourceLine" id="cb2-25" data-line-number="25">                            <span class="dt">number_samples =</span> n.samples,</a>
<a class="sourceLine" id="cb2-26" data-line-number="26">                            <span class="dt">UserMetadata =</span> UserMetadata,</a>
<a class="sourceLine" id="cb2-27" data-line-number="27">                            <span class="dt">Metadatafrozenidx =</span> Metadatafrozenidx,</a>
<a class="sourceLine" id="cb2-28" data-line-number="28">                            <span class="dt">datasetCount =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb2-29" data-line-number="29">                            <span class="dt">spikeCount =</span> spikeCount,</a>
<a class="sourceLine" id="cb2-30" data-line-number="30">                            <span class="dt">spikeStrength =</span> spikeStrength,</a>
<a class="sourceLine" id="cb2-31" data-line-number="31">                            <span class="dt">noZeroInflate =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb2-32" data-line-number="32">                            <span class="dt">percent_spiked =</span> spike.perc, </a>
<a class="sourceLine" id="cb2-33" data-line-number="33">                            <span class="dt">seed =</span> <span class="dv">1234</span>)</a>
<a class="sourceLine" id="cb2-34" data-line-number="34"></a>
<a class="sourceLine" id="cb2-35" data-line-number="35"><span class="co"># Gather SparseDOSSA output</span></a>
<a class="sourceLine" id="cb2-36" data-line-number="36">sparsedossa_results &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(DD<span class="op">$</span>OTU_count)</a>
<a class="sourceLine" id="cb2-37" data-line-number="37"><span class="kw">rownames</span>(sparsedossa_results)&lt;-sparsedossa_results<span class="op">$</span>X1</a>
<a class="sourceLine" id="cb2-38" data-line-number="38">sparsedossa_results&lt;-sparsedossa_results[<span class="op">-</span><span class="dv">1</span>,<span class="op">-</span><span class="dv">1</span>]</a>
<a class="sourceLine" id="cb2-39" data-line-number="39"><span class="kw">colnames</span>(sparsedossa_results)&lt;-<span class="kw">paste</span>(<span class="st">'Sample'</span>, <span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(sparsedossa_results), <span class="dt">sep=</span><span class="st">''</span>)</a>
<a class="sourceLine" id="cb2-40" data-line-number="40">data&lt;-<span class="kw">as.matrix</span>(sparsedossa_results[<span class="op">-</span><span class="kw">c</span>((n.metadata<span class="op">+</span><span class="dv">1</span>)<span class="op">:</span>(<span class="dv">2</span><span class="op">*</span>n.microbes<span class="op">+</span>n.metadata)),])</a>
<a class="sourceLine" id="cb2-41" data-line-number="41">data&lt;-<span class="kw">data.matrix</span>(data)</a>
<a class="sourceLine" id="cb2-42" data-line-number="42"><span class="kw">class</span>(data) &lt;-<span class="st"> &quot;numeric&quot;</span></a>
<a class="sourceLine" id="cb2-43" data-line-number="43">truth&lt;-<span class="kw">c</span>(<span class="kw">unlist</span>(DD<span class="op">$</span>truth))</a>
<a class="sourceLine" id="cb2-44" data-line-number="44">truth&lt;-truth[<span class="op">!</span><span class="kw">stri_detect_fixed</span>(truth,<span class="st">&quot;:&quot;</span>)]</a>
<a class="sourceLine" id="cb2-45" data-line-number="45">truth&lt;-truth[(<span class="dv">5</span><span class="op">+</span>n.metadata)<span class="op">:</span><span class="kw">length</span>(truth)]</a>
<a class="sourceLine" id="cb2-46" data-line-number="46">truth&lt;-<span class="kw">as.data.frame</span>(truth)</a>
<a class="sourceLine" id="cb2-47" data-line-number="47">significant_features&lt;-truth[<span class="kw">seq</span>(<span class="dv">1</span>, (<span class="kw">as.numeric</span>(spikeCount)<span class="op">+</span><span class="dv">1</span>)<span class="op">*</span>(n.microbes<span class="op">*</span>spike.perc), (<span class="kw">as.numeric</span>(spikeCount)<span class="op">+</span><span class="dv">1</span>)),]</a>
<a class="sourceLine" id="cb2-48" data-line-number="48">significant_features&lt;-<span class="kw">as.vector</span>(significant_features)</a>
<a class="sourceLine" id="cb2-49" data-line-number="49"></a>
<a class="sourceLine" id="cb2-50" data-line-number="50"><span class="co"># Remove intermediate files</span></a>
<a class="sourceLine" id="cb2-51" data-line-number="51"><span class="kw">file.remove</span>(<span class="st">'SyntheticMicrobiome-Counts.pcl'</span>)</a>
<a class="sourceLine" id="cb2-52" data-line-number="52"><span class="co">#&gt; [1] TRUE</span></a>
<a class="sourceLine" id="cb2-53" data-line-number="53"><span class="kw">file.remove</span>(<span class="st">'SyntheticMicrobiome.pcl'</span>)</a>
<a class="sourceLine" id="cb2-54" data-line-number="54"><span class="co">#&gt; [1] TRUE</span></a>
<a class="sourceLine" id="cb2-55" data-line-number="55"><span class="kw">file.remove</span>(<span class="st">'SyntheticMicrobiomeParameterFile.txt'</span>)</a>
<a class="sourceLine" id="cb2-56" data-line-number="56"><span class="co">#&gt; [1] TRUE</span></a>
<a class="sourceLine" id="cb2-57" data-line-number="57"></a>
<a class="sourceLine" id="cb2-58" data-line-number="58"><span class="co"># Extract features table from SparseDOSSA output</span></a>
<a class="sourceLine" id="cb2-59" data-line-number="59">features&lt;-<span class="kw">as.data.frame</span>(<span class="kw">t</span>(data[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span>n.metadata),]))</a>
<a class="sourceLine" id="cb2-60" data-line-number="60"></a>
<a class="sourceLine" id="cb2-61" data-line-number="61"><span class="co"># Extract metadata table from SparseDOSSA output</span></a>
<a class="sourceLine" id="cb2-62" data-line-number="62">metadata&lt;-<span class="kw">as.data.frame</span>(data[<span class="dv">1</span>,])</a>
<a class="sourceLine" id="cb2-63" data-line-number="63"><span class="kw">colnames</span>(metadata)&lt;-<span class="kw">rownames</span>(data)[<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb2-64" data-line-number="64">metadata<span class="op">$</span>Metadata1&lt;-<span class="kw">as.character</span>(metadata<span class="op">$</span>Metadata1)</a>
<a class="sourceLine" id="cb2-65" data-line-number="65"></a>
<a class="sourceLine" id="cb2-66" data-line-number="66"><span class="co"># Mark True Positive (TP) features</span></a>
<a class="sourceLine" id="cb2-67" data-line-number="67">wh.TP =<span class="st"> </span><span class="kw">colnames</span>(features) <span class="op">%in%</span><span class="st"> </span>significant_features</a>
<a class="sourceLine" id="cb2-68" data-line-number="68"><span class="kw">colnames</span>(features)&lt;-<span class="kw">paste</span>(<span class="st">&quot;Feature&quot;</span>, <span class="dv">1</span><span class="op">:</span>n.microbes, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>)</a>
<a class="sourceLine" id="cb2-69" data-line-number="69">newname =<span class="st"> </span><span class="kw">paste0</span>(<span class="kw">colnames</span>(features)[wh.TP], <span class="st">&quot;_TP&quot;</span>)</a>
<a class="sourceLine" id="cb2-70" data-line-number="70"><span class="kw">colnames</span>(features)[wh.TP] &lt;-<span class="st"> </span>newname;</a>
<a class="sourceLine" id="cb2-71" data-line-number="71"><span class="kw">colnames</span>(features)[<span class="kw">grep</span>(<span class="st">'TP'</span>, <span class="kw">colnames</span>(features))]</a>
<a class="sourceLine" id="cb2-72" data-line-number="72"><span class="co">#&gt; [1] &quot;Feature46_TP&quot;  &quot;Feature58_TP&quot;  &quot;Feature165_TP&quot; &quot;Feature172_TP&quot;</span></a></code></pre></div>
<p>Executing the above code will generate a synthetic microbiome dataset with 100 samples and 200 features, among which 4 features (2%) are differentially abundant with respect to a simulated binary metadata. Note that, the <code>True Negative</code> features do not have <code>_TP</code> attached to their names. As a quick sanity check, we can plot the distributions of the SparseDOSSA-generated features. To this end, we first create a temporary data frame combining the feature of interest with the metadata to visualize the distribution using a grouped boxplot.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># Create data frames to plot one TP feature and one TN feature</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">plot_data_TP&lt;-<span class="kw">cbind.data.frame</span>(<span class="dt">y =</span> features[,<span class="st">'Feature172_TP'</span>], metadata)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">plot_data_TN&lt;-<span class="kw">cbind.data.frame</span>(<span class="dt">y =</span> features[,<span class="st">'Feature3'</span>], metadata)</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="kw">colnames</span>(plot_data_TP)&lt;-<span class="kw">c</span>(<span class="st">'Feature'</span>, <span class="st">'Metadata'</span>)</a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="kw">colnames</span>(plot_data_TN)&lt;-<span class="kw">c</span>(<span class="st">'Feature'</span>, <span class="st">'Metadata'</span>)</a>
<a class="sourceLine" id="cb3-6" data-line-number="6"></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co"># TP (We expect significant group differences here)</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8">plot_data_TP <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Metadata, <span class="dt">y =</span> Feature, <span class="dt">fill =</span> Metadata)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_boxplot</span>(<span class="dt">outlier.shape =</span> <span class="ot">NA</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_log10</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">'True Positive Feature'</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">''</span>)</a></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAYAAAAUg66AAAAEDmlDQ1BrQ0dDb2xvclNwYWNlR2VuZXJpY1JHQgAAOI2NVV1oHFUUPpu5syskzoPUpqaSDv41lLRsUtGE2uj+ZbNt3CyTbLRBkMns3Z1pJjPj/KRpKT4UQRDBqOCT4P9bwSchaqvtiy2itFCiBIMo+ND6R6HSFwnruTOzu5O4a73L3PnmnO9+595z7t4LkLgsW5beJQIsGq4t5dPis8fmxMQ6dMF90A190C0rjpUqlSYBG+PCv9rt7yDG3tf2t/f/Z+uuUEcBiN2F2Kw4yiLiZQD+FcWyXYAEQfvICddi+AnEO2ycIOISw7UAVxieD/Cyz5mRMohfRSwoqoz+xNuIB+cj9loEB3Pw2448NaitKSLLRck2q5pOI9O9g/t/tkXda8Tbg0+PszB9FN8DuPaXKnKW4YcQn1Xk3HSIry5ps8UQ/2W5aQnxIwBdu7yFcgrxPsRjVXu8HOh0qao30cArp9SZZxDfg3h1wTzKxu5E/LUxX5wKdX5SnAzmDx4A4OIqLbB69yMesE1pKojLjVdoNsfyiPi45hZmAn3uLWdpOtfQOaVmikEs7ovj8hFWpz7EV6mel0L9Xy23FMYlPYZenAx0yDB1/PX6dledmQjikjkXCxqMJS9WtfFCyH9XtSekEF+2dH+P4tzITduTygGfv58a5VCTH5PtXD7EFZiNyUDBhHnsFTBgE0SQIA9pfFtgo6cKGuhooeilaKH41eDs38Ip+f4At1Rq/sjr6NEwQqb/I/DQqsLvaFUjvAx+eWirddAJZnAj1DFJL0mSg/gcIpPkMBkhoyCSJ8lTZIxk0TpKDjXHliJzZPO50dR5ASNSnzeLvIvod0HG/mdkmOC0z8VKnzcQ2M/Yz2vKldduXjp9bleLu0ZWn7vWc+l0JGcaai10yNrUnXLP/8Jf59ewX+c3Wgz+B34Df+vbVrc16zTMVgp9um9bxEfzPU5kPqUtVWxhs6OiWTVW+gIfywB9uXi7CGcGW/zk98k/kmvJ95IfJn/j3uQ+4c5zn3Kfcd+AyF3gLnJfcl9xH3OfR2rUee80a+6vo7EK5mmXUdyfQlrYLTwoZIU9wsPCZEtP6BWGhAlhL3p2N6sTjRdduwbHsG9kq32sgBepc+xurLPW4T9URpYGJ3ym4+8zA05u44QjST8ZIoVtu3qE7fWmdn5LPdqvgcZz8Ww8BWJ8X3w0PhQ/wnCDGd+LvlHs8dRy6bLLDuKMaZ20tZrqisPJ5ONiCq8yKhYM5cCgKOu66Lsc0aYOtZdo5QCwezI4wm9J/v0X23mlZXOfBjj8Jzv3WrY5D+CsA9D7aMs2gGfjve8ArD6mePZSeCfEYt8CONWDw8FXTxrPqx/r9Vt4biXeANh8vV7/+/16ffMD1N8AuKD/A/8leAvFY9bLAAAAOGVYSWZNTQAqAAAACAABh2kABAAAAAEAAAAaAAAAAAACoAIABAAAAAEAAAEgoAMABAAAAAEAAAEgAAAAAKtAJY0AACaqSURBVHgB7Z0HmFTVGYb/hYUVBBYQVFBYEikxISgoIAkiGAuoYI9CEgOBEDVUERBRBESDkgQ1QRQjYkERKSKBKKg0MQQNIEWKoVcpS1nAXdrmfkfPODs7c/feKbfNd55nuDO3nPKeux+n/n9GoRGEgQRIgARcIFDKhTSZJAmQAAkoAhQgvggkQAKuEaAAuYaeCZMACVCA+A6QAAm4RoAC5Bp6JkwCJEAB4jtAAiTgGoFM11J2MOEjR47IqVOnHEwxdlIZGRmSlZUlBQUFki4rIMqUKSMo94kTJ2KDcfhK1apVHU6RyUUjkBYChBf/5MmT0crv+LnMzEypUqWKHD9+3DN5SjWE8uXLKwHKy8tLdVKM32cE2AXzWYUxuyQQJAIUoCDVJstCAj4jQAHyWYUxuyQQJAIUoCDVJstCAj4jQAHyWYUxuyQQJAIUoCDVJstCAj4jQAHyWYUxuyQQJAIUoCDVJstCAj4jQAHyWYUxuyQQJAIUoCDVJstCAj4jQAHyWYUxuyQQJAIUoCDVJstCAj4jQAHyWYUxuyQQJAIUoCDVJstCAj4jQAHyWYUxuyQQJAIUoCDVJstCAj4jQAHyWYUxuyQQJAIUoCDVJstCAj4jQAHyWYUxuyQQJAJpYRM6SBXmx7LAKQADCUQjwBZQNCo8l1QCjzzyiPTv3z+pcTKyYBCgAAWjHlkKEvAlAQqQL6uNmSaBYBCgAAWjHlkKEvAlAQqQL6uNmSaBYBCgAAWjHlkKEvAlAQqQL6uNmSaBYBCgAAWjHlkKEvAlAQqQL6uNmSaBYBCgAAWjHlkKEvAlAQqQL6uNmSaBYBCgAAWjHlkKEvAlAQqQL6uNmSaBYBCgAAWjHlkKEvAlAQqQL6uNmSaBYBCgAAWjHlkKEvAlAQqQL6uNmSaBYBCgAAWjHlkKEvAlAQqQL6uNmSaBYBCgAAWjHlkKEvAlAUcF6OjRo7J48eJioNavXy9z5syR/fv3F7mG3x988IHgemSI9UzkffxNAiTgXQKOCVB+fr4MHTpUZsyYUYTG6NGjZdSoUbJ8+XLp2rWrbNu2TV3H7y5dusiGDRtkwIABMn369NBzsZ4J3cAvJEACviDgiABt2rRJOnfuLHl5eUWgbNmyRRYtWiTjxo2TgQMHSseOHWXixInqnmeeeUZGjBghPXv2VNfHjx8vJ06cELNnikTOHyRAAp4n4IhfsOPHj8vgwYPlwIEDMnv27BAUCFOjRo2kVKlvdbBJkyYya9YsOXXqlOzYsUNdw83nnXeelC9fXnbu3CmbN2+O+kwoUuPLihUrZN68eaFTN910k5x//vmh325+0WVFeU6fPu1mVhxLW5e5QoUKjqXJhPxBwBEBatiwoaIxf/78IlR2794t2dnZoXOVKlVSIrV37145++yzJSMjI3QN9+Xm5kqsZ0I3Gl+++uormTJlSuhUq1at5Ic//GHotxe+ZGVleSEbjuQB9YgPRJeBBMIJOCJA4QmGfy9dunSRVgBaPuXKlZPI83gG184666xi1/Qz4fHeeeedgo8OGMyGqHkhZGZmSvXq1eXgwYNy8uRJL2Qp5XlAS6+wsNAzdYAC16hRI+XlZgIlE3BkDChWNvCHiFaNDviOF+Occ86RY8eOSUFBgb6k7qtZs6b64432TOhGfiEBEvANAVcFqGnTprJ69WrZvn27auHMnDlTmjVrJmglNG/eXN577z0FcuHChVKlShX1ifWMb4gzoyRAAiECtrpg33zzjZpGx9R4t27dZNeuXYKBYz3IGIrV4heM+XTv3l3FVbVqVcnJyZFOnTqpp++///7Q9DviHzJkiDpv9ozFZHkbCZCARwhkGH3zQit5Wbt2rbRr10727NkjZ86ckf/85z/y8MMPy+HDh2XatGkJzTJhLATdrWizJIcOHZLKlSsXy6LZM5E3YwzIK+MtegzIS3mK5JXs34899piq35EjRyY76rjj4xhQ3OiS+qDlLhgWCbZo0UINJNaqVUtlYsKECar189ZbbyWUqTJlykQVH0QaTXxw3uwZXGcgARLwPgFLAoR1PEuXLpXhw4cLukA6YH1O7969i6zt0dd4JAESIIGSCFgSIHQbMA6DmanIsGbNmrjHgCLj4m8SIIH0ImBJgMqWLSvXXXed9O3bVz777DNFCK2iN998U8aOHStt27ZNL2osLQmQQFIIWJ4Fw36tW2+9VU2TY1VrmzZt1MDu3XffLb169UpKZhgJCZBAehGwLEDYS7VkyRK1eXTdunWCVtGll16qPumFjKUlARJIFgHLAoT9XJhG7dChg2BvFQMJkAAJJErA0hgQFiBu3LhRrf9JNEE+TwIkQAKagKUWEDaIovUDw2AwiVGvXj21aVRHUq1aNbn44ov1Tx5JgARIwBIBSwKEmIYNG6ZWPffo0aNYxNh5Pnny5GLneYIESIAEzAhYFiC0fGLt2sA6IQYSIAESsEvAsnIcOXIkpgChiwZbPQwkQAIkYIeAZQHCGA82nkYL7IJFo8JzJEACJRGwLECffPJJEeuFMDC/bNkyGTNmjDz11FMlpcPrJEACJFCMgGUB0nadw2No2bKlYEvGk08+KS+99FL4JX4nARIggRIJWFoHZBYL7KqsWrXK7BZeIwESIIGoBCy3gGCOAwbgdcB3GCd74oknpHXr1vo0jyRAAiRgmYBlAcJu+GiD0OiGPf7445YT5I0kQAIkoAlYFiC4TI5cBwTfXVwDpFHySAIkYJeA5TGg22+/XfnkgoNA/YH4wCyrNiRvN3HeTwIkkN4ETFtAMDw/d+5cRQjT8NgPFr7gEA7n3n33XbU3LL0xsvQkQALxEDAVoLp168oDDzygPBrAqwT8tsNrqQ4wDF+nTh3lHUOf45EESIAErBIwFSB4KF28eLGKC3aAYII1muscq4nxPhIgARIIJ2AqQOE3wksp7ALt2LEjNB2PLhhmxuDjCrNkDCRAAiRgh4BlAXrjjTfk3nvvjeoZ4/e//z0FyA513ksCJKAIWJ4F69+/v9xxxx3y4YcfSsWKFZV96Oeee06wEvpPf/oTcZIACZCAbQKWWkDoZmHV84gRI+TCCy+U6tWrK4+lPXv2VAPU2Av2l7/8xXbifIAESCC9CVhqAZUvX165QtZT8D/60Y9Cg9PNmzdXfuLTGyNLTwIkEA8BSy0gTLdfcskl8thjj6m1QHDH884778hvfvMb5Zb5Bz/4QTxpp+UzBQUFArdGmE0MX9LgNgxsKB5quN4+feZM0rOCyYvCQpFbjMWsyQ7w2Dv4oYfksssuS3bUjM8BApYECPl4/vnnpX379mrjKQadUeH4I8JMGGbIGKwRwJaWLl26KI+yXjLkv337djmwb58UXNVaCktbfi2sFfq7u47YurvkmzMMVcua95Eg7xSgknl58Q7Lb1rTpk0FfzwnTpxQwrN8+XKZM2eOXHvttZKTk+PFsjFPcRDIv/4GkaysOJ504ZHvBMiFlJlkkghYFiCkB2+o+CDUrl1bunXrpr7zHxIgARKIh4ClQWgdMfrykyZNkuHGWMGuXbvk888/p7NCDYdHEiAB2wQst4DWrl0r7dq1U9PxZ4yBSowHPfroo2ol9LRp0wS+4xlIgARIwA4Byy2grl27SosWLWTv3r1Sq1YtlQZMcWAW4q233rKTJu8lARIgAUXAkgDB8DxMsqLrValSpRC68847T3r37q2m4kMn+YUESIAELBKwJEAwPIaWzrFjx4pFu2bNGnWt2AWeIIE0JfDnP/9ZuTLfvHlzVAJ///vf1XW4trIa4Bg0kfDxxx/LK6+8YiuKRNO0kpglAcLMF3a79+3bVz777DMVL1pFMM8xduxYadu2rZW0eA8JpAWBUaNGydChQ2X8+PHFygtrEr169VLXrf6Bz549O+G/MQhQtPwUy+B3J3r06CHY65nqYEmAkIlx48YpH2DNmjUTKHubNm3kV7/6lVx99dUKaKozyvhJwE8E6tevr3YLROb57bfflpo1a0aeNv29evXqqL0P04cSvAhrqE4Ey7NggLZkyRJZtGiR2kqAVhG2ZODDQAIkUJQALEfAhPHKlSulUaNGoYtYxnLXXXfJX//619A5fFmxYoXyMrx161b58Y9/LAMGDFBCtWDBApk6daqyw4UdCE8//bRUqVJFNm3apFoo2NaDvZqYIMJ4rF6nh+svv/yyiveqq64q4tUY6WFL0LPPPqt6NGiJNWjQQPr166cWFWNj+ZYtW2TGjBlqu9CgQYNM70d88QZTAYIdaKxy1rNeGRkZ0qpVK/WJN0E3nsMmWq9YcsQsIgJeGrxIXgnwcOLX4DWW4IhlKfjDnzx5ckiANm7cKF999ZUSpnABQvfoxhtvlFtuuUXuvPNO1VWCaEG8zj33XGWBAuvu0PuAwKAHguu33Xab2o+J++Aaa5+xlQYClZubK9dcc41gkuiee+6R1157TcXVpEmTUBVff/31AuHp3r275Ofnyz/+8Q9lchn5w2ZzvA8XXHCBEkM8ZHY/xofjDaYCBANk6AviiIANi+iPDhw4MN70XHkOgGHT2gtBD+RjDO3gwYNeyJLKg86XZzJkIyPxsIQdq1QH3dKBGRsEtH4gMtqqhE7/wQcfVGvs9HIWtHQgFjBzgwFrWJz43//+JziPsH79etWKgjt0/PFjKATigx4KwujRo9V/uJ9++qmg0XDffffJFVdcoa7hnwMHDihhw/it3o+IFtANN9yg4oEYYgzr8ssvl5tvvrnE+yF08QZTAYqM9Msvv1TNPr8JUGQ5+JsEnCAAV1b4D/yLL75Q1iQgQJghCw/oCuE6BPEhY1e/DrCUgJ0G0QImfVq3bi1oOWGBMP4uYShQLwZGdw7XIT464JmPPvpI/YStd7TMcB/W8kHQFi5cqK5ht0NksHt/5PNmv+NvO5nFymskQAJSrVo1NUkD0zVYrvL111/LL37xiyJk0A3CzgIMEaA1oz/Y5A0BixbQ5YIJHCwOxmAxumPo7ulw6NChYmM+MKmjA3oEECQMp0AU0YVFKypWsHt/rHiinbfVAooWAc+RAAnEJoBuGMZl0KLB+E6kJ2FYF8XiXkzyoMulAyxNaNEIb8ngOrpHGKdBq0fblMJ4LUzjIKD7hqGS8KBbPzgHX374jYFqPb6LcwgQw8hg9/7I581+swVkRofXSCBBArfeeqv6Q4c9rbvvvjtqbBijef3115VdLYgIukMYe4G3GYSqVavK7t27ZcOGDcojDbpaGGj+1tBboZqtmjJlipqpwv0dO3ZUpnMwy4V70N3S7rVwHc8jHbTIEDDzNnjwYPUdrR0EdLvQvUO6Vu5XD8XxT4ktICw2RB8VAQNhaN4BWHiA4uoBsvDz/E4C6U4AM51YxItuU8uWLaPigKXRo0ePKqcPaCFhUBdOINBiQkBXCS0dDBT/+9//Vs5CsTYI92FAu3HjxoLFjxibRZcOA85Y9YzfmM7Xs2GYskfA+NDvfvc71R1E9wstLTyPySbY+cIyAAyWY1ofYgiDbyXdryKO4x/DqByMZUYPaD6imVZSwJSflz1j4H8Sr8yCYQrVixYR0WRHV+Hwk0/7yiBZ9oN9pE+fPuoPpqT3NPy6E7Ng4elZ+Y53FK0SOH6IFvCff+XKlUOXMJuFaXl4qYkVdu7cqbp3kd043A/jgvjbiLUwEtdPnTqlxois3B8rD2bnTVtAWLXJQAIk4AwBtERiiQ9yEC4++I1uUkkBa3liBYhXLPHBM7iOjw4l3a/vs3PkGJAdWryXBEggqQQoQEnFychIgATsEKAA2aHFe0mABJJKgAKUVJyMjARIwA4BWwJEo/R20PJeEiCBkgiYzoKFP0yj9OE0+J0ESCAZBCwLkDZK/+KLL6qFT0h8grGRDftVsIsX1hIZSCDdCWCtTqpC5DR8qtJxMl5LXTAapXeySphWuhDAamY7dqGDyMVSC4hG6YNY9dHLVHHkEyLfW3GIfhPPJoUAbEOjZ2G2Ez0pCXk4EksChBWQ2ig9lusjhBulD7dj4uGyMmsWCJQ6ctjCXbwlGQSwIVTvYE9GfH6Mw5IAoWAwSo+dvTALiX0lMEqPvSvY4QslZwgGgUJjO4BRwb4pTIaxX4nBvwQsCxBMAtAovX8r2mrOjwwzumBZWVZvd/c+Yx81NqN6McCQ/AS45Ym911tOGdYQXzN2rb9jGAWLFSoatoLGG5M9kWZcY93vt/OWBahevXqqG9a5c2dlyNpvBWV+ScBJAjDufuLwYelQzWTD6PnmtpS3G7Z55hnWE2BiI+0F6G9/+5u8+uqrytp+7dq1BUL029/+1nT3rpMVzrRIwGsEzskqK71zopvWsJLXebkHZd5Be9P6MCAGa4ew6RPuBcNKem7cY2kaHhmDgaLp06cL3INgzc8///lPqVOnjnLXMXfuXDfyzjRJgATCCMyfP1/56Vu2bJly8wOvF14PlgVIFwQ2SO6//3554403lAU1iA8coDGQAAm4SwCTQTDNCp9jMFYPS4vwuuHlYHkMCIWA9TQYKYP4LF26VFniR7cslvV+LxeceSOBVBKAcfeTxmdXfvwCkGvDlx0sF2LcSZt9xTAJjN3DjPJPfvKTVBY1obgtCxDs08I6PiysYewHtqLhGoSBBEigOAHYX952/Btpv2JV8Ys2z0Tz1RUZBew2Z2dnF/EFht4KTLwGQoBQmH/961/KkHU0+7KRQPz+GwstMfuQ7KA9HcCe7549e5IdvbIP7Gc3y0kH4lKEF110kRzeukUeyqkVdw5W5h2V8bv2SLly5UqMA7sV0AoKD1in5/V3wbQFhH4kHKZBQeHqAwWEN8bIAKv7DRs2jDzt6989jHGuTVu2pKwMjzzySErivsDwsDnR2BzM4C6BLGMt1dmGKFxZ5Xsj8nZzdMpkDVFkXHCdg/8w4VZHT9njPziv91JMBQj7VOB1Ed0t+Ck6bKxriBbQPYPvoSCFY4ablDbGy9Ph3Gq+Kdb7+3NlSV6eb/LLjCaPAAzaw9spdixgMBrDJWgYnHvuuclLJAUxmQoQfATp7tbevXtjJg93skEMF56VJa0S+B/MaSZrjx6TJQXcmuA0d6+kh32aN910k4wZM0b5EZs4caJXshYzH6YCpF3D4ukbb7xRrQNClyw8wCYQ3MiilcRAAiTwPYEjJ0/JrH0Hvj9h89vaY8dsPQF3zZj1wjgj/NL7IZgKEMaA9CJD+J4eOXJkqH+JwmEnL5p62KbBQAIk8D0BeETdZ6zBGbJx8/cn4/iGsSR4L7UT/CI+KJOpANWtW1e5gcViJoyoz5o1SzXtNAy0kLAa+uGHH9aneCQBEjAI/OEPf5Bf//rXxl7UmI6H5YYbbpB77rknps94gIx0Dhg0uKYChKl37dS+Q4cOqpsV2QULGhCWhwSSRaCkKXCMr0Jg0vlvylSAwivivffeC//J7yRAAgkS0BM8CUbj68dNBSh8HdCCBQuKLXTSJQ/iOiBdNh5JIFUE+vfvrzaPpip+P8RrKkDpvA7ID5XHPPqbAGaW0z2YClC6rwNK95eD5SeBVBMwXUGIWS7sMUHAUW/DwMDZc889J1gBjcVO+p5UZ5bxk0CQCGzYsEE5dwhSmeyWxVSAwiMbPHiwWmUJi2tYdKg9YWDZN0x0MJAACdgjcN9996l1dPaeCtbdlgUI1tVmz54tOTk58vrrr6uV0TNmzJDhw4fLJBOj2sHCxdKQQPIIYHN35A725MXuj5gsCVBubq7y4Ni6dWuBbRKYfoSLHgSYHdAmJvxRZOaSBEjAKwRMB6F1JrGsHMvBFy1aJOiCnTB8MWHnLdQb3S/4CEs0wIATzAhgZbUOELb//ve/6lyDBg30aXVcv369bN26VRne9tPS8yKF4I/AEkDvYILRUzAL+cbugtcmvinTTdbYZRtWDV94/vkiW6DM4vTbNUsChAVTGAO6/vrrBaYme/bsKbA/AkP1WCs0cODAhMq92XA90q9fP+nRo0dIgDADN2TIEOUK6HmjAuCFQ7e6Ro8eLWvWrFF70LDzFx47YIKSgQS8QmDt2rWyyzA6V9D4sthZql1HYLB1f4w7ShnPlzHec7rlMQBh0Lldu3bK4BG8oyL07t1beUotacm5ujnGP/if4hXDORvMSYaHZ555RkaMGCGXXHKJ/PKXv5Ru3bqpcSd45UBLDMa3YQYE40+YiRs0aFD44/xOAq4TyMiuLPm33RF3PjJXrZQy6760/Tzsdi1cuFDat29v+1mnH7DUAtKZwgA0ZsDeMizuoQXUuHFjCTfZoe+zc4S5SXjVwLS+XpqOrt2OHTuUMTTEhZXW6ALu3LlT0FqCkTRtgwi+j7BJNjx8+umnRc5hUyDybifo+O0844V7wTBSzK3ky+6OaytxOnUPuu7xlNmp/DmZDkwJw1166dKlgyVAK1euVN0hGLmGAOzbt0+NB2EsCP7CtBlIu7Cvu+660CN65zCMn6FVpQUJN+AFw2A4xqDCXzZY/od95fCA+1avXh06hd38iQplKDKvf8mQuMqq+WQ/PMDrJSyWP+Rd57/YxTQ6gXcewxQYE8VGcj8Eyy2g7t27S4sWLdR4y4UXXqjMDKxYsUJuu+02efbZZxMeBwqHBfWGraHwgFYRRC7yGs5HGu2GVTh8dMBgtt2ZOox1GYnpKHxzLDxTaLusKBy6un369FFjfMkuLJZvoJ5gUSHZAS3Vyy+/3HaZaxi2s1MZ8J9pofEOZyRiIteYcbYT8oy04CYL/0ljWMMPwZIAoVn3+eefC1pBEB8EtE7QBXvggQeUl9REB6LDYUG9jxnW4NBygUEmBLRqatasqTxJIB864HyqXyadVpCPMAmBSYVUhC+++ELVJf6zSpeAWV3Z+7VUGpq48wErbnnAFQ0EhKlTp6qjH/6xJED4XwaKDiGKDBAK/O+WzICtHc2bNxeYAMF2DwyoYSkAPk2bNlUtLvhBgvDMnDlTDYQnM33GRQKJEoA3inV7vpbj7W+OO6pS27bKWQvmFWvhxx2hBx+0JEDo+rQ2FiGilQOzrGjyonUyb948NXiM7RjJDnD/PGDAADW+BAHElDwCxnzQHcSsWNWqVdXgcqdOnZKdPOMjgYQI4G8mw5hgOXlp47jjycQQgCFAQQ6WBAgAXnjhBTXAhSn46tWry1HDbQ2ahpjqwxqeRMOwYcOKRIFZKyxyPHTokFSuXNS3EsZ3sCYJIpjO1uSKAOMPEvAhAcsCBMPzWJX84Ycfil61fOmll8rPf/7zlBY7Unx0Ypz50CR4JAH/ErAkQDBIf/DgQeXkDEaUaEjJvxXOnDtIwOghZK5cEXeCmdu2xfXs7bffLvj4IZgKEIQHU7Pjx49XK6Dr16+vvqe61eMHcMwjCZgRwFhl4aGDcvariU2HZxprnCKXmZil67drpgL04osvKlevGHzG9DvGgbDKEhtB/bxy1m+VxPz6j8C9994rd911l2nGsWiwY8eOaqtRrBuxDCXI45ymAoRpcPg3wp4shJYtW8pPf/pTNRZ05ZVXxmLG8ySQ9gQwc1vSamSspcOK/5LuCzJMU3tAWD2MbpcODRs2VDNgMIPBQAIkkBiB8K1GicXk36dNW0Cw+wP7z+EBah1tQWL4PfxOAiRQMgGsn8OaunQOpgKUzmBYdhJINYF02poSi2WJAgR7Pdj2oAN2wb/77rvKGqE+h3EhDE4zkEC6E4i1bi3ducQqv6kAYeYLM1746IDpRSxEVJvtvjsJA0gUIE2IRxIgAasETAXo/ffftxoP7yMBEiAB2wRMZ8Fsx8YHSIAESMAGAQqQDVi8lQRIILkEKEDJ5cnYSIAEbBCgANmAxVtJgASSS8CWAMH+D9zgwB0z3OPATKuynZzcPDE2EiCBNCFgOgsWzgCO1uAXbM+ePUp0YIjs0UcfFUzBT5s2TbnpCb+f30mABEigJAKWW0Bdu3ZVRq/hMqdWrVoq3gkTJij/XPATxkACJEACdglYEiDs/Vq6dKnqemEhog5wGAjvqHC7wkACJEACdglY6oLBSwXMC8ADRmSAj3ZcC1qAp49JhsPD6Xtjee72XolPGL7MyhrG0BlIwC8ELAkQdsTDg2nfvn3l6aefVmVDqwhumseOHav8xvulwHbyedJwRXQywkGinefduLeMG4kyTRKIk4AlAULc48aNC3nFgB2TNm3aCEy2Yg9YKtzyxFme5D1mlPEso2VXvrR/WnffnD4jhmdmBhLwDQHLAgSvpEuWLJFFixapjahoFcErBj5BDPDJdOd51aVPzrcD7n4o44vbd8rbh/P8kFXmkQQUAcsCBH/T8I5at25d9dH8sB4IRrPhtZSBBEiABOwQsCxAF198sVrzEy1yuE+ePHlytEs8RwIkQAIxCVgWoE8++UROhw3I5uXlybJly2TMmDHy1FNPxUyAF0iABEggFgHLAgSD9JEBXjIwG/bkk0/KSy+9FHmZv0mABEjAlEDCUzw1atSQVatWmSbCiyRAAiQQjYDlFhBWQmNxng74jn1hTzzxhLRu3Vqf5pEESIAELBOwLEBYiIiNp5EB3bDHH3888jR/kwAJkECJBCwL0JYtWyTSkRq8OmKbBgMJkAAJxEPA8hjQz372M1mwYIFkZ2eHPhSfeJDzGRIgAU3AkgDBENnGjRtpfExT45EESCApBCz1n7DSeeTIkTJgwADZuXOn1KtXT61+1jmoVq2aYKEiAwmQAAnYIWBJgBDhsGHD1CB0jx49isXPldDFkPAECZCABQKWBQgtH+wFixY4FhSNCs+RAAmURMBUgPr16ydt27aVa6+9VjDjxUACJEACySRgOgj9wQcfqMHnZCbIuEiABEhAEzAVIH0TjyRAAiSQCgKmXTAkuH37dlm+fLlp2rAFVKdOHdN7eJEESIAEIgmUKEDY6Y6PWfD6LNhZhqH2ChUqmBWh2DW/GtrHanWvGYcDSy/mq1il84TjBEoUoEGDBskdd9xhmjGvvfCRmc3Pz1f2qyPPm/1WHl8Ns6x+C5ipPHjwoKeyDZZeyxesODC4T6BEAapdu7Y0adLE/ZwyByRAAoEjwEHowFUpC0QC/iFgKkDwBQ8j9AwkQAIkkAoCpl2wUaNGpSJN38T5Rd5Rgasbv4TPj9Alj1/qivn8loCpAKUzpLrGhtvVhqnZrUn2s3XGGCQuMAbFs7KyUuLSum79+p6rtvLly0tpHw7oew5kADOUYcxORN/gFaDC7t+/3/YsWKqKv3nzZunSpYtyaZ0uFgQwS4pp+Nzc3FRhtR0vZ8FsI0vJA6ZjQClJkZGSAAmQwHcEKEB8FUiABFwjQAFyDT0TJgESoADxHSABEnCNAAXINfRMmARIgALEd4AESMA1AhQg19AzYRIgAQoQ3wESIAHXCFCAXEPPhEmABChAfAdIgARcI0ABcg09EyYBEqAA8R0gARJwjQAFyDX0TJgESIACxHeABEjANQIUINfQM2ESIAEKEN8BEiAB1whQgFxDz4RJgAQoQHwHSIAEXCNAAXINPRMmARKgAPEdIAEScI0ABcg19EyYBEiAAsR3gARIwDUCFCDX0DNhEiABChDfARIgAdcIUIBcQ8+ESYAEKEB8B0iABFwjQAFyDT0TJgESoADxHSABEnCNAAXINfRMmARIgALEd4AESMA1AhQg19AzYRIgAQoQ3wESIAHXCFCAXEPPhEmABChAfAdIgARcI0ABcg09EyYBEvC8AB09elQWL15crKbWr18vc+bMkf379xe75uUTOTk5Mnv2bGnQoIGXs8m8kYAjBDwtQPn5+TJ06FCZMWNGERijR4+WUaNGyfLly6Vr166ybdu2Ite9/KNs2bJy0UUXSVZWlpezybyRgCMEPCtAmzZtks6dO0teXl4REFu2bJFFixbJuHHjZODAgdKxY0eZOHFikXv4gwRIwB8EMr2azePHj8vgwYPlwIEDqsui8wlhatSokZQq9a12NmnSRGbNmqUvq+PHH38skyZNCp3r1auX1K1bN/TbzS8ZGRkq+YoVK0phYaGbWXEs7TJlyqi0qlSp4liaTMgfBDwrQA0bNlQE58+fX4Tk7t27JTs7O3SuUqVKSqRCJ4wvZ86ckdOnT4dO4Q9d/+GHTrr0RedDH13KhuPJorzpVmbHIfswQc8KUCyWpUuXLiIup06dknLlyhW5/ZprrhF8dMBAdW5urv7p6jEzM1OqV68uR44ckZMnT7qaF6cSR8sH4uOVOkC5a9So4VTxmY4JAc+OAcXKM/54w19kfOfLFIsWz5OAtwn4ToCaNm0qq1evlu3btwtaPzNnzpRmzZp5mzJzRwIkEJWA77pgGPPp3r27dOvWTapWrSpYV9OpU6eoheNJEiABbxPIMAZofTkVg/GTgoICqVChQomEMQbklfEWPQbkpTyVCDDBGzgGlCDAAD/uuxaQrgtM7erpXX2ORxIgAX8R8K0A2cFcrVo1O7en9N5169ZJq1at1OJJrGFKh/DHP/5RsKr95ZdfTofisow2CPhuENpG2Tx5K3q8WKfk055vXEwj12XFFQkfCiQBClAgq5WFIgF/EEiLLpiXqgJbMNq2bSvptC0BXU2vTAJ46V1gXkR8OwvGyiMBEvA/AXbB/F+HLAEJ+JYABcjhqvOrIbVEMc2dO1etXE80Hj4fLAIUIAfr08+G1BLBNHXqVBk+fHiRTcSJxMdng0OAg9AO1aU2pDZlyhRlywj2imBIbdCgQQ7lwPlksFdvyJAhxcylOJ8TpuhVAmwBOVQz0Qypffnllw6l7k4ysMl0xRVXyJgxY9zJAFP1PAG2gByqIiuG1BzKimPJwO51hw4dHEuPCfmPAFtADtWZFUNqDmWFyZCAZwhQgByqChpScwg0k/EVAQqQQ9VFQ2oOgWYyviLAMSCHqouG1BwCzWR8RYBbMRyuLjuG1BzOGpMjAccJUIAcR84ESYAENAGOAWkSPJIACThOgALkOHImSAIkoAlQgDQJHkmABBwnQAFyHDkTJAES0AQoQJoEjyRAAo4ToAA5jpwJkgAJaAIUIE2CRxIgAccJUIAcR84ESYAENAEKkCbBIwmQgOMEKECOI2eCJEACmgAFSJPgkQRIwHECFCDHkTNBEiABTYACpEnwSAIk4DgBCpDjyJkgCZCAJkAB0iR4JAEScJwABchx5EyQBEhAE6AAaRI8kgAJOE6AAuQ4ciZIAiSgCVCANAkeSYAEHCdAAXIcORMkARLQBChAmgSPJEACjhOgADmOnAmSAAloAhQgTYJHEiABxwlQgBxHzgRJgAQ0AQqQJsEjCZCA4wT+D6zybbJZLO4HAAAAAElFTkSuQmCC" /><!-- --></p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="co"># TN (We do not expect significant group differences here)</span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3">plot_data_TN <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Metadata, <span class="dt">y =</span> Feature, <span class="dt">fill =</span> Metadata)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_boxplot</span>(<span class="dt">outlier.shape =</span> <span class="ot">NA</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_log10</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">'True Negative Feature'</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">''</span>)</a></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAYAAAAUg66AAAAEDmlDQ1BrQ0dDb2xvclNwYWNlR2VuZXJpY1JHQgAAOI2NVV1oHFUUPpu5syskzoPUpqaSDv41lLRsUtGE2uj+ZbNt3CyTbLRBkMns3Z1pJjPj/KRpKT4UQRDBqOCT4P9bwSchaqvtiy2itFCiBIMo+ND6R6HSFwnruTOzu5O4a73L3PnmnO9+595z7t4LkLgsW5beJQIsGq4t5dPis8fmxMQ6dMF90A190C0rjpUqlSYBG+PCv9rt7yDG3tf2t/f/Z+uuUEcBiN2F2Kw4yiLiZQD+FcWyXYAEQfvICddi+AnEO2ycIOISw7UAVxieD/Cyz5mRMohfRSwoqoz+xNuIB+cj9loEB3Pw2448NaitKSLLRck2q5pOI9O9g/t/tkXda8Tbg0+PszB9FN8DuPaXKnKW4YcQn1Xk3HSIry5ps8UQ/2W5aQnxIwBdu7yFcgrxPsRjVXu8HOh0qao30cArp9SZZxDfg3h1wTzKxu5E/LUxX5wKdX5SnAzmDx4A4OIqLbB69yMesE1pKojLjVdoNsfyiPi45hZmAn3uLWdpOtfQOaVmikEs7ovj8hFWpz7EV6mel0L9Xy23FMYlPYZenAx0yDB1/PX6dledmQjikjkXCxqMJS9WtfFCyH9XtSekEF+2dH+P4tzITduTygGfv58a5VCTH5PtXD7EFZiNyUDBhHnsFTBgE0SQIA9pfFtgo6cKGuhooeilaKH41eDs38Ip+f4At1Rq/sjr6NEwQqb/I/DQqsLvaFUjvAx+eWirddAJZnAj1DFJL0mSg/gcIpPkMBkhoyCSJ8lTZIxk0TpKDjXHliJzZPO50dR5ASNSnzeLvIvod0HG/mdkmOC0z8VKnzcQ2M/Yz2vKldduXjp9bleLu0ZWn7vWc+l0JGcaai10yNrUnXLP/8Jf59ewX+c3Wgz+B34Df+vbVrc16zTMVgp9um9bxEfzPU5kPqUtVWxhs6OiWTVW+gIfywB9uXi7CGcGW/zk98k/kmvJ95IfJn/j3uQ+4c5zn3Kfcd+AyF3gLnJfcl9xH3OfR2rUee80a+6vo7EK5mmXUdyfQlrYLTwoZIU9wsPCZEtP6BWGhAlhL3p2N6sTjRdduwbHsG9kq32sgBepc+xurLPW4T9URpYGJ3ym4+8zA05u44QjST8ZIoVtu3qE7fWmdn5LPdqvgcZz8Ww8BWJ8X3w0PhQ/wnCDGd+LvlHs8dRy6bLLDuKMaZ20tZrqisPJ5ONiCq8yKhYM5cCgKOu66Lsc0aYOtZdo5QCwezI4wm9J/v0X23mlZXOfBjj8Jzv3WrY5D+CsA9D7aMs2gGfjve8ArD6mePZSeCfEYt8CONWDw8FXTxrPqx/r9Vt4biXeANh8vV7/+/16ffMD1N8AuKD/A/8leAvFY9bLAAAAOGVYSWZNTQAqAAAACAABh2kABAAAAAEAAAAaAAAAAAACoAIABAAAAAEAAAEgoAMABAAAAAEAAAEgAAAAAKtAJY0AACpQSURBVHgB7Z0JlBTV1cfvwLAz7Itg2E5AQ4KgEECQKEpkEcGgbCJJICBRAkEkQYGAiIgYNCDGoHIU/BQhRCADgQTQBEE8Bgio7EQWGTaH3QHCsH71f1htT093zeuerq7tfz1td1fVW+7v9Vzeem/aVUOEQgIkQAIOECjiQJkskgRIgAQUARog/hBIgAQcI0AD5Bh6FkwCJEADxN8ACZCAYwRogBxDz4JJgARogPgbIAEScIxAumMlu6TgEydOOF6TokWLSrFixeT8+fOO18XOCpQoUUIuX74sly5dsrMYrbwrVaqk9RwfspdA4A1Qbm6uvYQ1ci9TpoyUKlVKTp8+rfG0dx8pX768nDt3TtzA3LsU/VVzDsH81Z7UhgQ8RYAGyFPNxcqSgL8I0AD5qz2pDQl4ikBKDdCZM2dk7dq1+QDt3LlTVqxYIceOHctzD9+XL18uuB8piaSJzIPfSYAEnCWQMgOEFZ7x48dLZmZmHo2nTp0qU6ZMkU2bNsmAAQNk//796j6+9+/fX3bt2iUjR46URYsWhdIlkiaUmB9IgARcQyAlq2B79uyR0aNHC1ZB8DJl3759smbNGnnvvfekSJEiMm/ePJkzZ46MGjVKpk2bJhMnTpQmTZpIz549ZeDAgdK5c2c5dOhQ3GmKFy9uFsl3EiABFxFIiQHC0uuYMWPk+PHjsmzZspD6MEyNGzdWxgcXmzZtKkuXLlX7RA4cOKDu4Xr16tWldOnScvDgQdm7d2/caerVq4dslHTv3l3MvT8wbuh9OS1paWmqClWrVnW6KraWj39kzC0HthbEzD1DICUGqFGjRgrIqlWr8oA5fPhwnh5RuXLllJHKzs5WP1TzDxOJ0HOC4UgkTbgBatu2rWAuClKnTh1XbP7DJkT00vy+ERH/iGAT4oULFxR/J/9XtmxZJ4tn2d8QSIkBikUbO4CxM9YU/DixIS/yOu7jXsmSJfPd00lj5o/3IUOGhH9VBi3PBQe+oFcAA5STk+NA6akrEm2LTYjmPwCpKzl/STRA+Zk4cSVlk9DRlMOQwxwO4T4+16hRQypXrixnz57Ns2MW92rWrCmJpIlWNq+RAAk4T8BRA9S8eXPZsmWLZGVlqR7OkiVLpEWLFpKeni4tW7aUxYsXK0KrV6+WihUrqlciaZzHzBqQAAlEI+DoEAxzPoMGDVIrXDgciDmZPn36qHoOHjw4tPyOyctx48ap64mkiaY4r5EACThPIM0NTukvXryohlvRxuWnTp2SChUq5COVSJp8mRgXMKnttGAOKCMjQ44cOeJ0VWwtv1q1auowqhvmgDDUpzhPwNEekKk+VoHwiibRjA+eSyRNtPx5jQRIwDkCjs4BOac2SyYBEnADARogN7QC60ACASVAAxTQhqfaJOAGAjRAbmgF1oEEAkqABiigDU+1ScANBGiA3NAKrAMJBJQADVBAG55qk4AbCNAAuaEVWAcSCCgBGqCANjzVJgE3EKABckMrsA4kEFACNEABbXiqTQJuIEAD5IZWYB1IIKAEaIAC2vBUmwTcQIAGyA2twDqQQEAJ0AAFtOGpNgm4gQANkBtagXUggYASoAEKaMNTbRJwAwEaIDe0AutAAgElQAMU0Ian2iTgBgI0QG5oBdaBBAJKgAYooA1PtUnADQRogNzQCqwDCQSUAA1QQBueapOAGwjQALmhFVgHEggoARqggDY81SYBNxCgAXJDK7AOJBBQAjRAAW14qk0CbiBAA+SGVmAdSCCgBNIDqndI7bJly4Y+O/WhWLFiqmg31MVOBmlpaVK8eHHxu552MvRb3oE3QLm5uY63Kf4wS5QoIW6oi50wypQpI5cvX3aFnhkZGXaqyrw1CQTeAF28eFETlX2PoVcAcUNd7NNS5OrVq8oA+V1POxn6Le+45oD+97//ybx582TChAly6NAh2bBhg1y5csVvTKgPCZBAigho94C2b98unTp1kiNHjiij06VLFxk7dqycPn1aFi5cKNddd12KqsxiSIAE/EJAuwc0YMAAadWqlWRnZ0utWrWU/rNnz5YiRYrI3Llz/cKDepAACaSQgJYBOnfunKxbt04NvcqVKxeqXvXq1WXYsGGybNmy0DV+IAESIAFdAloGKD09XfV0zp49my/frVu3qnv5bvACCZAACRRAQMsAYZWmffv2Mnz4cFm/fr3KEr2id999V2bMmCEdO3YsoBjeJgESIIH8BNKMpdGr+S/nv4JVr27duqmhGPatoFeE5dTevXvLO++8I0WLFs2fyANXDh8+7HgtsT8G+1Iwwe9nqVatmuAfrjNnzjiuZo0aNRyvAysgor0KhlWuTz75RNasWSM7duxQO1pvvvlmwYtCAiRAAokQ0DZAjRo1ksmTJ0vXrl3l9ttvT6QspiEBEiCBPAS05oCwAXH37t3cdJgHHb+QAAkUloBWD6hUqVKq9zNy5Eg5ePCgNGjQQHDNlCpVqkjDhg3Nr3wnARIgAS0C2pPQFSpUULueo+Xao0cPmT9/frRbrr/GSejUNREnoVPH2islafWAoAx6PrEWzLAiRiEBEiCBeAloW46vv/46pgHCcKxkyZLxls3nSYAEAk5A2wBhjgcHT6OJl4dg0fThNRIggdQQ0DZAH330kfLlYlYrJydHNm7cKK+88oo8//zz5mW+kwAJkIA2AW0DhH1AkdKmTRu1s3XSpEkyc+bMyNv8TgIkQAKWBLT2AVnlgC3tmzdvtnqE90iABEggKgHtHhDccVy6dCmUCT7j7NKzzz4rbdu2DV3nBxKIReCxxx6T1q1b8/cSC1AAr2sbIJyGjzYJjWHYM888E0B0VDleAnDhW7t27XiT8XkfE9A2QPv378+3DI9T3NwD5ONfB1UjAZsJaM8BPfDAA8rlRvny5cV8wfjALWufPn1sriazJwES8CMByx7Qv//9b1m5cqXSG8vwOA0fvuEQMZ7++te/qrNhfoRDnUiABOwlYGmA6tevL48//rgKJAfnY0uXLs3jeAwRPevWrSujR4+2t5bMnQRIwJcELA1Q5cqVZe3atUpx+AGCC1aG1fXl74BKkYAjBCwNUHiNFi9eLPALdODAgdByPIZgWBk7duyY8hkd/jw/kwAJkEBBBLQnoeH3uWrVqiomWL169QQvDNGaNWsm7733XkHl8L4FgczMTOnQoYPFE7xFAv4koG2Afvvb30r37t3l/fffVw7U4R96+vTpgp3Qzz33nD/ppEgrhDuCuxMKCQSNgNYQDMMs7HqeOHGifOc731E9ITgoGzp0qJqgxlmwF198MWjsqC8JkEAhCWj1gEqXLi1Y8TKX4L/3ve+FJqdbtmwpWK6nkAAJkEC8BLQMEIxPkyZN5KmnnhK44UAonr/85S8qLhjCMmM+qLCCUD/79u3Lkw0mt5cvXy47d+7Mcx1fcG3FihVqAjz8plWa8Of4mQRIwHkCWgYI1fzTn/4kCxYskH/84x/y8MMPqwCFWJKfMmWKPPjgg4XSZO/evTJixAjZvn17KJ9NmzZJ//79ZdeuXQJn+IsWLQrdmzp1qioXzwwYMEBwTARilSaUmB9IgARcQ0BrDgi1bd68ufpDv3DhgtoLhD929EDuvvtuqVOnTsIKYQVo1qxZ6nhHeCbTpk1Tc07oefXs2VMGDhwonTt3FkRoRXBErLwVKVJE5s2bJ3PmzJFRo0ZJrDQILU0hARJwHwFtA4SqY9/P3/72N9UrgUHAUKxWrVqF0gr+pN944w21ooaQzxC4+sB+o8aNG6vv1atXF8xDYaUIvSVch/GBNG3aVO3QtkoTPkSEMTtx4oRKi/rjeInTYhpIbHPwu2A4HwQ9/d6OydJP2wBheNSpUye1GnblyhXp0qWLjB07Vm1EXLhwoSB0cyICNx+mmFE3srOzBSftTYOE+zgAC8OBMDr4bEq5cuXk+PHjYpUm3ADBfYgZmxyuIXJzc82sHHuHYYe4oS52Q8Bvxw160ouD3S2tl7+2AcJcS6tWreS1116TW265ReU+2zgJj1Pyc+fOleHDh+uVqPFU0aJF8/ifRhL0cLAKF3kP19GLirwenia8yF//+tfhX5VBy3PBgS+mAULkEb8LdHWDnvgHjuI8Aa1J6HPnzqlJ5wkTJgh6HKZgaDRs2DDBSlgyBWfQsDkv/F9K9H5q1qypuu/mEApl4jM2Q1qlSWbdmBcJkEDyCGgZIHRXMecCoxApW7duDc3HRN5L9DvKw/4inD+DrF69WipWrKhemAzfsmWLZGVlqV7RkiVLpEWLFsoxWqw0idaD6UiABOwloDUEwyQp5mowzPr973+vaoReEU7Hz5gxQ5588smk13Lw4MGh5XcYv3Hjxqky0AMbNGiQWhWrVKmSWoEzHaLFSpP0yjFDEiCBpBDQjg2P5e9u3bqpoRgmh9FLgY+g3r17Cw6qYg7GDjl16pTg2EekoGwM0aK5B4mVJjIPfHdDbHhsRXj55ZfVObtodfTLtfvvv1/w6tu3r+MqYdhOcZ6AVg8I1cT8Cw6gYg8Odi2jV4RlbLzslGjGB+VhORevaBIrTbRneY0ESMA5ApYGCG5YscnQ3OuDns/tt9+uXs5VmSWTgLsJvPDCC2q+9Gc/+1nUY0p//OMf1dYReBvNyMjQUgYrh+ELQFqJwh765z//KV9++aU6XRB22fJjYcu0zPybm5aT0I888oja5GdmhACEDMNs0uA7CUQngONJ48ePlzfffDPfA9hgi60guK+7HQGrzB07dsyXVzwXYICi1SdWHkOGDFGbg2PdT9Z1SwMUWci2bdtSUqnIcvmdBLxG4IYbblAHtiPr/ec//1lNZ0Ret/qOVd9oK9BWaQp7L1UeLuIyQIVViulJICgE4Lzvv//9r3z++ed5VMbZxV69euW5hi+ffvqpOuSN1WZEkMWiD+TDDz9Uh8DRc8Ih8JMnT6rre/bsUc+hZ4SJffS6cE7TFNwfM2aMOj+JlWtzs6t5Hws4uN6jRw/ljRO9MgzRIPDtBc8UWBwxnQ1aPW/mmcg7DVAi1JiGBAoggKNJd9xxh8yfPz/05O7du5VRuueee0LX8AHDI5wywBEhGAT0PnDeEUaoWrVqygkgzkJivxsWf8zzkNiE+9Of/lSFxUJ04t/97ncqX1z/8Y9/rPK99957lScJeC8NF7gAhjFs166dOmKFOtx1112CozLw94Wd4tdff718//vfV8msng/PN97PlpPQ8WbG50mABL4lgJ7OH/7wB+XVAVfxB/+Tn/wk5NjPfPI3v/mNMgI40gRBTweHrOFpFBPW2GD7xRdfqOu4D19YyHvmzJlqE/BDDz0kR48eVavUuA93Ndie8vHHH6vzlI8++qjceuutuKUEZydh2LCHr2HDhurajTfeKDCMyAdeJzBH9cMf/lDuu+8+NWFu9TxORCQqBRogbDb87LPPVP6AgD02UChcAAvQKCRAAt8SwDlJTObi7wduZWCAsEIWLhja4D72JYVv6MW+ug0bNoQ/GvqMYVfbtm1VDweHxDE3C1/t5oFwDOdwP/wwN9J88MEHKg8cW0LPDM/hPCcMGk4bQBD5JlLifT4yvdV3SwP0gx/8QDCWDAeBLln4d2TOfTdWiHkvqASqVKmihjXwHoqNu1999ZUa8oRP8GIlDMMe9FhMFzPgBT9bOH4UTTCvhCERhmM/+tGP5LbbblPDN9NzKDoJkXM+4Xvmzp8/r3pi6CG1bt1apUcvCt+jSbzPR8sj1jVLA4QZewoJkEDiBDBUwmQvejSY34EhChf4RsL+Hmz0xZDLFDj7M41GeE8G9zE8wjwNej3mCQTs2TONDkYkkQfEzd4P0iOcOr6jc2Hu8cM1CIxhpMT7fGR6q++chLaiw3skUEgCOL6EP3S4NMaxpWiCKY23335bHb6GEcFwCHMv8G8OwZlHHBmCe2K4n8FQCxPNGC7BhxZWq+AhFMM5CFwkw03xSy+9pJ7BcMuMcIz7SI9y0CODYPULK2YQ9HYgGHZheIdydZ5XiRL4Hw1QAtCYhAR0CWAYhaV1+KyCM7xogmAPME5YusfO6J///OeCOHzoMUFw+gA9HUwUY/oDO6jhlA+Tvxjm4RwhluGxyoYhHSac4eYY802YHsEkN3Zlm4L5oV/84hdqOIi5JwzjcNgbz8LVMgST5TBqWHnTed7MO9537cOo8Wbsled5GDV1LcXDqNasccAavRLE3osmkYessZqFeSCr4xxwY4zhXeQwDvlj3xB6WbgfTXAfPS5sAYAU9Hy0PAq6lndAWtDTvE8CJGAbAcz5xDI+KDRysQfDpIIEe3liCYxXLOODNLiPlykFPW8+F887h2Dx0OKzJEACSSUQlwHCpBf2MsA1K3ZpYjwabdY8qTVkZiRAAr4loD0Esysqhm/JUjESIIECCWj3gMyoGAh/Y+4dwC5KbJ4yt5AXWBofIAESIIEwAloGKNVRMcLqx48kQAI+JqBlgLB7M5VRMXzMm6qRAAmEEdCaA8LyW6qjYoTVkR9JwDMEsFfHLolchrernFTmq9UDQoVef/11wVAMOyPhj+TOO+8UHGCDD5HIaKOpVIBlkYBXCcDTYU5Ojlern5R6a/WAUBLOozgRFSMpWjITEnAhAfzDjcUd/EMeVNE2QA0aNFDDsH79+qnAgEEEhoN/vx46VI4ZK4HJFBwiTDcijjzYs2cys1Xb7/sZP3AMnynuI4ADoeYJdvfVLjU10jZAOPD21ltvKVePtWvXFhgiHJqz2jqeGhVSVwp+LJuNbnPzchnS4JvzMUkpvVRJEbySLJnHjisXoPEaIOiJM0QwuMkU5Aufxqbv4WTlDfehOJRJ8R6BuA+j4gAc/ATBfcD69evViVqctoUDJS9KPIdRcTAPvnZH16sjD1Sv6np1u3y2VW4zfAL/6le/iquucKD1yiuvxJXGyYexQrtkyRLlx1i3HnZFRjUnoRcsWCCzEZbHwojDpQZOyeMVSzIMX0FvGvvtSpYsme8sWKw0Xrqu3QMylcIBOMRgh0c2+J6FnxO4HPCqATL14vu3BHDkJq1ECcnpN+Dbiy79lL5nt5RcuVyFCXdTFRER48Lp09K1isWB0eusfSlnGb55/mUs+MDFBgyQHyUuA4Sj++j9IBb8unXrlNd/DMvg+5biMwJGr+LyDTe6XqkixsqsW6VyieIyrE501xo6df7XiZPyr5PxLeujRw9vh3CdDM+IbhftZXg4R0K3FY6PMKcAB/UI5YGwIKa/ELcry/qRgJ8JrFq1Sm6++WbZuHGjimyBqBduF+0eEIZef//739WcTzTnRm5XNFb94AxcVzAH5ClJu+bTJR4doR82nhpJPSWYiI5XT08pqFFZLOvDiyE8HCK4IcLqwPNhCWM47VaxNEDw3o9GRXQM+JnFHyB6PZEC15CNGjWKvOyJ76YfXZ3Kes4AGUph5SkeHcEBabwm0DEePa28CCZDd7ipuWi8Dp2/5qc5kTxPGB4SdQW/Tcw7mW5fsVINZ/cYqeDv161iaYCwSQoRGhEbDE6yTxuTatEEw7PwCJDRnnHrNbjB1BXPGSBjFR3GJB4dwQJpkrsAr0s48efQNvHqmXhpBafcsWOH7D/3P+ny6eaCHy7giWixuiKTZGVlKT/R4aMTjFrg4tWzBggOqk2F4IYjloTHM4r1DK+TQJAIfPe735XTX+6TJ+vUSljtz3POyJuHjlgu05uZ48B45D+QMMgYmrpZLHtAZlwiKIBwrYsWLco3zoZPIMQwQi+JQgIkcI0A5l3KGEbhRxUrJIzkksUeoshMEToHy/UIq2Mu2R85ckTq1asX+airvlsaIMwBrVy5UlUYgc8mT54cUg4X0VVH0DIc06CQAAk4RwCdBYRfxqFxTEbj7xJzs4jp7maxNED169dXMYgwuYfu3NKlS0ORGKEUlK5bt66MHj3azTqybiQQCAKIwHqvsfMdu9gRR2zOnDmu19vSAGESy4yo2LVrVzXMCvpSp+tblBV0DYGvL16SpUePJ1yf7WfPxpUW4Zqx6oUNw145G2dpgMK1X7x4sQrzeuDAgdBkF4ZgWBmDwvEeeAzPm59JwG8EcDzpqDFyGLd7b6FUw1xSvBt9vWJ8AEbbAOH4xSOPPCJno1jlhx9+mAaoUD8zJvYbgV/+8pfSt29fS48C99xzjwqZHCtmPJhgUyhefhVtA4RY1YhdjaMX3bp1U5PTOA/23HPPqZdfAVEvEkiUQEFL4NjiAuMS5GkNLQOEYRaW9CZOnKj8/1StWlW5BhhqOOfCBPWkSZPkxRdfTLSdmI4EAknA3GMXSOW/UVrrMCrGoFjxMvcXYLLLnJxu2bKlYLmeQgIkEB8BjCratWsXXyKfPa1lgGB8mjRpIk899ZRyoo0Tt3BahaX5ZcuWuX6zk8/ajOr4hAA2915//fU+0SYxNbSGYMgajse6dOkibdu2FUw6N2vWTI1dsRKGFTIKCZAACcRLQNsANW/eXPbv3y8XLlxQhgfnxHAEA54Q69SpE2+5fJ4EAk9g165dak413mV2P4HTGoKZCofP2OO4/8CBA2l8TDh8J4E4CTz66KPqyEScyXz1uHYPaNu2bVH9xOAkPJYRcRjOzY6PfNVqVMYXBHB6PfIEuy8Ui0MJbQPUunXrmP6AUB48+2OS+oknnoijeD5KAiQQZALaBujpp59WJ22HDRumImLgSAZctL766quSmZkpu3fvVgdXb7jhBrVRMchQqTsJ4G9ithG6ykrOG6vI/zfnXVlksYhT3vBq+KqxAGRugbHKz4v3tAwQ3Euid7Nw4UIVCx6KYuL5tttuk61bt6rJaBgouIR8//33aYC8+EtgnZNKYPv27XLIiKGXe0uz2PnWritw2HosxhNFjPTFjL+vwIflgZMjuIXExHOk1KxZU8zgfrgPV5QUEiABkbTyFeT8/d0TRpG++XMptmNb3OlxcmH16tVq20zciVOcQGsVDMuEmANCBNSdO3eGqojwH7MNj4hwhA2lETMs6Ds7Q3D4gQQcIHDOiJOGw62vvfaaA6XHX6SWAUK2M2fOVEMsHMOAEzL0dhD2A9EycEB11KhRyjjBbxCFBEgg9QS2bNmiTiyY4aFTX4P4S9SaA0K28I6IWPAff/yxYBMiJsXQ87nllltUqY8//rgKWljQCeD4q8gUJOA9AlcNf85XjVMCaTk5iVfemPaIR3KMshCpGFMis2bNiiepY89qGyDUEKd34XwM80EPPfSQHDp0SDBBjb1AMFAUEiCBawTUXGj2V1Ju/O8KjUQnLA8KadWqlSprwYIFhS4zVRloGyDM6nfq1Em55YDRwbmwsWPHqrkfrI5hIyKFBEjgGgFEo9hx5Cs51+W+hJEU2f+llPzwX1pheRIuxOGE2gYIQQphYTG5ZQ67MAH9wAMPyNy5c2X48OEOq8LiScA9BDBFkWZszr1487UpikRqlm44lhfDAPlZtAwQZtbh/RDjS4R7NQVhP7AxEaFAaIBMKv54v4pDx5MmuF6ZtNwLrq8jKxibgJYBQtRFzPNE8weNjYhBi4yamX1UNn5diMnF2O2R1DunjZ22CYsxiVrU2AhHIQE7CWgZIJyCR9QL9HIQewiCXhGioc6YMUOefPJJO+voury3nj0neLld0o1Fg8LIVeMfHdeLYSjT4oggmlJ9jMWa9M8/TbjIdMP9TSKCaRG8vCBaBgiKYJgFZ/QtWrRQq2F33nmn8oiITU+IxEjxF4E0IxzM6YmTXa9UsU83Sem3Z7uunpiquHrqpJR5q3DL4emGN1Ic9ParaBsgHLn45JNPZM2aNeq4BXpFcM2KV9DkFzWvk7sqV3S92sN37XF9Hf1aQYSw6tWrl6V6+AcdG3l79uwZ8zm4uPFz1AxLA4Tl9ZMnT+aDg7CvcMX6n//8R72w5HjXXXfley4ZF86cOSOfffaZOvganh+OhHz55ZfStGnTPFEgsU8J9cJu7RtvvDE8SdI+X2f8KBqWKZO0/OzKqJgXhlB2Ke9wvpgXRWRhK8G+OmzcLeg5qzy8fs/SAGG+B/t/Ygk2SMExfY8ePWwxQDgEO378eDXJjZP3pkydOlWdwm/QoIGKg/3yyy+royHYoT1u3Dg1XwUf1v369ePJfBMa311HgGF5RCxnGTHkwiHTyBd6RXDPgZ4QnNPjjz7ZsmfPHmVAsL08XPbt26eGgZiTgvMzdGHnzJmjHpk2bZqKXYZ4Zbj/5ptvKh/W4en5mQTcQgBzp23btnVLdRyph2UPKFqN4POnf//+6lwYDA+MAJbpky1YZRszZowcN5aCEfrHFBimxo0bh5b+MQRbunSpcm0JJ2m4B8EeJZziP3jwYJ6wQZg0P3HihHoGoYYQ2VVXPOc+01gEAwMEkoxHMCxIM/7zkmAYU6lSJS9VWe6//35P1deOympbDhy/mD59uowePVoaNmwoGzZskJtuusmOOqk8GzVqpN5XrVqVpwwctCtfvnzoGlYbYKSys7PVeDq8W4vnYGwwR2XKrbfeKphXgmCeCJFddcVzBshQDHN18egIFtf0vKqLxRXPIVpLPHra8Y8mQFSoUMEVPLxSCS0D9MUXX6heDyKg4vwXXG/Y1YAFgTMnwM3n8MeCZcrI67iPe5GuLB977DEzqXo3nanluRjji+cMkGFD8EcJj3rxCNJ4y/yICpiJ34Cu0GuDLil7n7OcA4JLAfR6MFTBLmj0emCAnDI+QIHhhDmEwnd8rlGjhlpJQB3D/xXEPWwfoJAACbiTgGUPCKtbONpfq1YttVdh5cqVglekYLn73nvvjbxsy3cESHzppZckKytLGZ4lS5aozZEwiohTjyitqDdcUlasWFG9bKkIMyUBEig0AUsDhNUv0w80jlzEEsS4TpUBwpzPoEGDVFBETDrCOX6fPn1U1QYPHiwjR46URYsWqUlqO1bnYjHgdRIggfgJWBqgaL2d+IsoXAosU0YuVcLYdejQQQ23wneJwhjBLzVcUnIysHDcmZoEUkHA0gClogKJllHMOCODVzSh8YlGhddIwH0ELCeh3Vdd1ogESMBPBGiA/NSa1IUEPEaABshjDcbqkoCfCNAA+ak1qQsJeIxAXAYIp9/nzZsnEyZMUCF5sDERRzQoJEACJJAIAe1VMIblSQQv05AACVgR0O4BmWF5cOgTO6Mhs42wPHC8hLA8FBIgARKIl4CWATLD8mDoFS0sT7i7jHgrwOdJgASCS0DLAOGcFcPyBPdHQs1JwC4CWgYoPCzP+vXrVV3Cw/J07NjRrvoxXxIgAR8T0J6EZlgeH/8KqBoJOERA2wAxLI9DLeRQsVeNgAAVJ01wqHT9Yq8aoZm95jxNXzv/P6ltgOA5EA7K6tevr14mmkOHDimPhPC9Q/EHgTvuuEO5rUV7J1PgqwmRTODSN5kC17vhbnqTmTfzspeAtgHCjwb+gaIJHIDNnz8/2i1fXlty9Jh8lnPNr7SbFTxp+ElORODWBL6Vki0ffPCBwCd33759k5018/MoAW0D9NFHHykH56aeCJezceNGFZfr+eefNy/7+h2rgW1at5ZjR4/K7iRqCp/NcKwf7jw/GdnXqVTZ1sAByagj8wg2AW0DZEapCMfVpk0bwWrYpEmTZObMmeG3fPt5oqFrsiUzM1MQXPHVgDBMNj/m510CWsvwVurBIfzmzZutHuE9EiABEohKQLsHtG7dum/iRV3LByFqjhw5Is8++2w+l6lRS+JFEiABEoggoG2A2rdvH3USGsOwZ555JiJbfiUBEiCBggloGyDEZA+POoqsEdzNyRhhBavHJ0iABNxMQHsOqLWx+vPhhx+q/RbmvgsaHzc3LetGAu4noGWA4Ihs9+7ddD7m/vZkDUnAUwS0hmCIvT558mQV9O/gwYNqNyuumVKlSpWk72418+Y7CZCAfwloGSCo//TTT6tJ6CFDhuSjEbSd0PkA8AIJkEBCBLQNEHo+sc4GcS4oIfZMRAKBJ2BpgEaMGCHw9XP33XerFa/A0yIAEiCBpBKwnIRevny5mnxOaonMjARIgAS+IWBpgEiJBEiABOwkYDkEQ8FZWVmyadMmyzrAF1DdunUtn3HrzbJlyzpetaJFi6o6uKEudsOAb/Eg6Gk3R7/kX6ABwkl3vKzEy6tgubm5Vqql5J4Z3NENdbFbYSxkuEHPjIwMu1Vl/hoECjRAo0aNku7du1tm5WVviBcvXrTULRU3zdVFN9TFbn2haxD0tJujX/Iv0ADVrl1bmjZt6hd9qQcJkICLCHAS2kWNwaqQQNAIWBqgTp065XFAHzQ41JcESMBeApZDsClTpthbOnMnARIINAHLHlCgyVB5EiAB2wnQANmOmAWQAAnEIkADFIsMr5MACdhOgAbIdsQsgARIIBYBGqBYZHidBEjAdgI0QLYjZgEkQAKxCNAAxSLD6yRAArYToAGyHTELIAESiEWABigWGV4nARKwnQANkO2IWQAJkEAsAjRAscjwOgmQgO0EaIBsR8wCSIAEYhGgAYpFhtdJgARsJ0ADZDtiFkACJBCLAA1QLDK8TgIkYDsBGiDbEbMAEiCBWARogGKR4XUSIAHbCdAA2Y6YBZAACcQiQAMUiwyvkwAJ2E6ABsh2xCyABEggFgEaoFhkeJ0ESMB2AjRAtiNmASRAArEI0ADFIsPrJEACthOgAbIdMQsgARKIRYAGKBYZXicBErCdAA2Q7YhZAAmQQCwCnjZAO3bskH379uXR7dixY7J8+XLZuXNnnuv8QgIk4D4CnjVAe/fulREjRsj27dtDVDdt2iT9+/eXXbt2yciRI2XRokWhe/xAAiTgPgLp7qtSwTXKzMyUWbNmSfny5fM8PG3aNJk4caI0adJEevbsKQMHDpTOnTtL8eLF8zzHLyRAAu4g4EkDVKpUKXnjjTdk+vTpkpaWpkheunRJDhw4II0bN1bfq1evLqVLl5aDBw9KvXr1QrT79OkjJ06cUN/x7KRJk0L3nPpQuXJlqVWrllStWtWpKqSs3GLFigVCz5QB9XhBnjRA7du3D2G/evWq+pydnS1lypQJGSRcRA8JxibcADVr1kzOnj2r0tStW1dyc3PVZyf/17VrV+nevbucO3fOyWqkpOwrV664gnl6uid/+ilpo1QW4ptWKFq0qFy+fDkPO/SKSpYsmeca5o3C5fDhw+FfHfkMw4lh4tdff+1I+aksFG3kBj3BnOI8Ac9OQkeiwzAGPZvwHg16PzVr1ox8lN9JgARcQsA3Bghd6pYtW8rixYsV2tWrV0vFihXVyyWsA1+Nd955R3r37h14DgTwLQHfDMGg0uDBg0PL70WKFJFx48Z9qyk/OU4Ac26Y5zpz5ozjdWEF3EEgzZjEvTaL6476JKUWp06dkgoVKmjl5ZY5oIyMDDly5IhWnb36ULVq1VxjgGrUqOFVjL6qt2+GYOGtomt8wtPwMwmQQOoJ+NIApR4jSyQBEkiEAA1QItSYhgRIICkEaICSgpGZkAAJJEKABigRakxDAiSQFAI0QEnByExIgAQSIUADlAg1piEBEkgKARqgpGBkJiRAAokQ8OVGxERAOJnmrbfekhdeeEE2b97sZDVsL7tNmzbSq1cvGTp0qO1lsQBvEGAPyAXtBBcVePldcBI+CHr6vR2TqR8NUDJpMi8SIIG4CNAAxYXLnofhMK1Dhw72ZO6iXNu1aycNGjRwUY1YFacJcA7I6RZg+SQQYALsAQW48ak6CThNgAbI6RYwykcMsxUrVghimvlZ4Ado7dq1flaRusVJgAYoTmDJfnzq1KkyZcoUQUyzAQMGyP79+5NdhCvyO3/+vIwfP14QUolCAiYBGiCThAPviOq6Zs0aef311+WJJ56QBx98UObMmeNATewtcs+ePdKvXz/JycmxtyDm7jkCNEAONhn+MBGbDO5jIU2bNpVt27Y5WCN7ioYb1jFjxigDa08JzNWrBGiAHGw5uIMNj+5arlw5OX78uIM1sqfoRo0ayU033WRP5szV0wRogBxsvshYZohjhqivFBIICgEaIAdbGqGYzTDRqAY+01m6gw3ColNOgAYo5ci/LbB58+ayZcsWycrKEvR+lixZIi1atPj2AX4iAZ8T8FVcMK+1FeZ8Bg0aJAMHDpRKlSpJnTp1pE+fPl5Tg/UlgYQJ8ChGwuiSl/DixYsqpHTZsmWTlylzIgEPEKAB8kAjsYok4FcCnAPya8tSLxLwAAEaIA80EqtIAn4lQAPk15alXiTgAQI0QB5oJFaRBPxKgAbIry1LvUjAAwRogDzQSKwiCfiVAA2QX1uWepGABwjQAHmgkVhFEvArARogv7Ys9SIBDxCgAfJAI7GKJOBXAjRAfm1Z6kUCHiBAA+SBRmIVScCvBGiA/Nqy1IsEPECABsgDjcQqkoBfCdAA+bVlqRcJeIAADZAHGolVJAG/EqAB8mvLUi8S8AABGiAPNBKrSAJ+JUAD5NeWpV4k4AECNEAeaCRWkQT8SoAGyK8tS71IwAMEaIA80EisIgn4lQANkF9blnqRgAcI0AB5oJFYRRLwKwEaIL+2LPUiAQ8Q+H/c2n5wKwJteAAAAABJRU5ErkJggg==" /><!-- --></p>
<p>As expected, we see significant group differences for the true positive features (<code>Feature46_TP</code>, <code>Feature58_TP</code>, <code>Feature165_TP</code>, <code>Feature172_TP</code>) but not for the rest. Now that we have validated SparseDOSSA, letâ€™s run Tweedieverse on this <code>'fake'</code> dataset (assuming that the absolute path of the output directory exists).</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co"># Run Tweedieverse using the CPLM model</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="cf">if</span> (<span class="op">!</span><span class="kw">dir.exists</span>(<span class="st">'demo_output'</span>)) <span class="kw">dir.create</span>(<span class="st">'demo_output'</span>)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">SimData&lt;-<span class="kw">Tweedieverse</span>(features, </a>
<a class="sourceLine" id="cb5-4" data-line-number="4">                   metadata, </a>
<a class="sourceLine" id="cb5-5" data-line-number="5">                   <span class="dt">output =</span> <span class="st">'./demo_output/SimData'</span>, <span class="co"># Assuming demo_output exists</span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6">                   <span class="dt">base_model =</span> <span class="st">'CPLM'</span>,</a>
<a class="sourceLine" id="cb5-7" data-line-number="7">                   <span class="dt">standardize =</span> <span class="ot">FALSE</span>) </a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="co">#&gt; [1] &quot;Warning: Deleting existing log file: ./demo_output/SimData/Tweedieverse.log&quot;</span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Writing function arguments to log file</span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Determining format of input files</span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Input format is data samples as rows and metadata samples as rows</span></a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Filter data based on min abundance and min prevalence</span></a>
<a class="sourceLine" id="cb5-13" data-line-number="13"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Total samples in data: 100</span></a>
<a class="sourceLine" id="cb5-14" data-line-number="14"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Min samples required with min abundance for a feature not to be filtered: 10.000000</span></a>
<a class="sourceLine" id="cb5-15" data-line-number="15"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Total filtered features with prevalence-abundance filtering: 0</span></a>
<a class="sourceLine" id="cb5-16" data-line-number="16"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Filtered feature names:</span></a>
<a class="sourceLine" id="cb5-17" data-line-number="17"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Total filtered features with variance filtering: 0</span></a>
<a class="sourceLine" id="cb5-18" data-line-number="18"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Filtered feature names:</span></a>
<a class="sourceLine" id="cb5-19" data-line-number="19"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Excluded metadata with entropy less or equal to 0:</span></a>
<a class="sourceLine" id="cb5-20" data-line-number="20"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Formula for fixed effects: expr ~  Metadata1</span></a>
<a class="sourceLine" id="cb5-21" data-line-number="21"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Bypass z-score application to metadata</span></a>
<a class="sourceLine" id="cb5-22" data-line-number="22"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Running selected analysis method: CPLM</span></a>
<a class="sourceLine" id="cb5-23" data-line-number="23"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Fitting model to feature number 1, Feature1</span></a>
<a class="sourceLine" id="cb5-24" data-line-number="24"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Fitting model to feature number 2, Feature2</span></a>
<a class="sourceLine" id="cb5-25" data-line-number="25"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Fitting model to feature number 3, Feature3</span></a>
<a class="sourceLine" id="cb5-26" data-line-number="26"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Fitting model to feature number 4, Feature4</span></a>
<a class="sourceLine" id="cb5-27" data-line-number="27"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Fitting model to feature number 5, Feature5</span></a>
<a class="sourceLine" id="cb5-28" data-line-number="28"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Fitting model to feature number 6, Feature6</span></a>
<a class="sourceLine" id="cb5-29" data-line-number="29"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Fitting model to feature number 7, Feature7</span></a>
<a class="sourceLine" id="cb5-30" data-line-number="30"><span class="co">#&gt; 2022-02-26 14:29:11 INFO::Fitting model to feature number 8, Feature8</span></a>
<a class="sourceLine" id="cb5-31" data-line-number="31"><span class="co">#&gt; 2022-02-26 14:29:12 INFO::Fitting model to feature number 9, Feature9</span></a>
<a class="sourceLine" id="cb5-32" data-line-number="32"><span class="co">#&gt; 2022-02-26 14:29:12 INFO::Fitting model to feature number 10, Feature10</span></a>
<a class="sourceLine" id="cb5-33" data-line-number="33"><span class="co">#&gt; 2022-02-26 14:29:12 INFO::Fitting model to feature number 11, Feature11</span></a>
<a class="sourceLine" id="cb5-34" data-line-number="34"><span class="co">#&gt; 2022-02-26 14:29:12 INFO::Fitting model to feature number 12, Feature12</span></a>
<a class="sourceLine" id="cb5-35" data-line-number="35"><span class="co">#&gt; 2022-02-26 14:29:12 INFO::Fitting model to feature number 13, Feature13</span></a>
<a class="sourceLine" id="cb5-36" data-line-number="36"><span class="co">#&gt; 2022-02-26 14:29:12 INFO::Fitting model to feature number 14, Feature14</span></a>
<a class="sourceLine" id="cb5-37" data-line-number="37"><span class="co">#&gt; 2022-02-26 14:29:12 INFO::Fitting model to feature number 15, Feature15</span></a>
<a class="sourceLine" id="cb5-38" data-line-number="38"><span class="co">#&gt; 2022-02-26 14:29:12 INFO::Fitting model to feature number 16, Feature16</span></a>
<a class="sourceLine" id="cb5-39" data-line-number="39"><span class="co">#&gt; 2022-02-26 14:29:12 INFO::Fitting model to feature number 17, Feature17</span></a>
<a class="sourceLine" id="cb5-40" data-line-number="40"><span class="co">#&gt; 2022-02-26 14:29:12 INFO::Fitting model to feature number 18, Feature18</span></a>
<a class="sourceLine" id="cb5-41" data-line-number="41"><span class="co">#&gt; 2022-02-26 14:29:12 INFO::Fitting model to feature number 19, Feature19</span></a>
<a class="sourceLine" id="cb5-42" data-line-number="42"><span class="co">#&gt; 2022-02-26 14:29:12 INFO::Fitting model to feature number 20, Feature20</span></a>
<a class="sourceLine" id="cb5-43" data-line-number="43"><span class="co">#&gt; 2022-02-26 14:29:12 INFO::Fitting model to feature number 21, Feature21</span></a>
<a class="sourceLine" id="cb5-44" data-line-number="44"><span class="co">#&gt; 2022-02-26 14:29:12 INFO::Fitting model to feature number 22, Feature22</span></a>
<a class="sourceLine" id="cb5-45" data-line-number="45"><span class="co">#&gt; 2022-02-26 14:29:12 INFO::Fitting model to feature number 23, Feature23</span></a>
<a class="sourceLine" id="cb5-46" data-line-number="46"><span class="co">#&gt; 2022-02-26 14:29:13 INFO::Fitting model to feature number 24, Feature24</span></a>
<a class="sourceLine" id="cb5-47" data-line-number="47"><span class="co">#&gt; 2022-02-26 14:29:13 INFO::Fitting model to feature number 25, Feature25</span></a>
<a class="sourceLine" id="cb5-48" data-line-number="48"><span class="co">#&gt; 2022-02-26 14:29:13 INFO::Fitting model to feature number 26, Feature26</span></a>
<a class="sourceLine" id="cb5-49" data-line-number="49"><span class="co">#&gt; 2022-02-26 14:29:13 INFO::Fitting model to feature number 27, Feature27</span></a>
<a class="sourceLine" id="cb5-50" data-line-number="50"><span class="co">#&gt; 2022-02-26 14:29:13 INFO::Fitting model to feature number 28, Feature28</span></a>
<a class="sourceLine" id="cb5-51" data-line-number="51"><span class="co">#&gt; 2022-02-26 14:29:13 INFO::Fitting model to feature number 29, Feature29</span></a>
<a class="sourceLine" id="cb5-52" data-line-number="52"><span class="co">#&gt; 2022-02-26 14:29:13 INFO::Fitting model to feature number 30, Feature30</span></a>
<a class="sourceLine" id="cb5-53" data-line-number="53"><span class="co">#&gt; 2022-02-26 14:29:13 INFO::Fitting model to feature number 31, Feature31</span></a>
<a class="sourceLine" id="cb5-54" data-line-number="54"><span class="co">#&gt; 2022-02-26 14:29:13 INFO::Fitting model to feature number 32, Feature32</span></a>
<a class="sourceLine" id="cb5-55" data-line-number="55"><span class="co">#&gt; 2022-02-26 14:29:13 INFO::Fitting model to feature number 33, Feature33</span></a>
<a class="sourceLine" id="cb5-56" data-line-number="56"><span class="co">#&gt; 2022-02-26 14:29:13 INFO::Fitting model to feature number 34, Feature34</span></a>
<a class="sourceLine" id="cb5-57" data-line-number="57"><span class="co">#&gt; 2022-02-26 14:29:13 INFO::Fitting model to feature number 35, Feature35</span></a>
<a class="sourceLine" id="cb5-58" data-line-number="58"><span class="co">#&gt; 2022-02-26 14:29:14 INFO::Fitting model to feature number 36, Feature36</span></a>
<a class="sourceLine" id="cb5-59" data-line-number="59"><span class="co">#&gt; 2022-02-26 14:29:14 INFO::Fitting model to feature number 37, Feature37</span></a>
<a class="sourceLine" id="cb5-60" data-line-number="60"><span class="co">#&gt; 2022-02-26 14:29:14 INFO::Fitting model to feature number 38, Feature38</span></a>
<a class="sourceLine" id="cb5-61" data-line-number="61"><span class="co">#&gt; 2022-02-26 14:29:14 INFO::Fitting model to feature number 39, Feature39</span></a>
<a class="sourceLine" id="cb5-62" data-line-number="62"><span class="co">#&gt; 2022-02-26 14:29:14 INFO::Fitting model to feature number 40, Feature40</span></a>
<a class="sourceLine" id="cb5-63" data-line-number="63"><span class="co">#&gt; 2022-02-26 14:29:14 INFO::Fitting model to feature number 41, Feature41</span></a>
<a class="sourceLine" id="cb5-64" data-line-number="64"><span class="co">#&gt; 2022-02-26 14:29:14 INFO::Fitting model to feature number 42, Feature42</span></a>
<a class="sourceLine" id="cb5-65" data-line-number="65"><span class="co">#&gt; 2022-02-26 14:29:14 INFO::Fitting model to feature number 43, Feature43</span></a>
<a class="sourceLine" id="cb5-66" data-line-number="66"><span class="co">#&gt; 2022-02-26 14:29:14 INFO::Fitting model to feature number 44, Feature44</span></a>
<a class="sourceLine" id="cb5-67" data-line-number="67"><span class="co">#&gt; 2022-02-26 14:29:14 INFO::Fitting model to feature number 45, Feature45</span></a>
<a class="sourceLine" id="cb5-68" data-line-number="68"><span class="co">#&gt; 2022-02-26 14:29:14 INFO::Fitting model to feature number 46, Feature46_TP</span></a>
<a class="sourceLine" id="cb5-69" data-line-number="69"><span class="co">#&gt; 2022-02-26 14:29:15 INFO::Fitting model to feature number 47, Feature47</span></a>
<a class="sourceLine" id="cb5-70" data-line-number="70"><span class="co">#&gt; 2022-02-26 14:29:15 INFO::Fitting model to feature number 48, Feature48</span></a>
<a class="sourceLine" id="cb5-71" data-line-number="71"><span class="co">#&gt; 2022-02-26 14:29:15 INFO::Fitting model to feature number 49, Feature49</span></a>
<a class="sourceLine" id="cb5-72" data-line-number="72"><span class="co">#&gt; 2022-02-26 14:29:15 INFO::Fitting model to feature number 50, Feature50</span></a>
<a class="sourceLine" id="cb5-73" data-line-number="73"><span class="co">#&gt; 2022-02-26 14:29:15 INFO::Fitting model to feature number 51, Feature51</span></a>
<a class="sourceLine" id="cb5-74" data-line-number="74"><span class="co">#&gt; 2022-02-26 14:29:16 INFO::Fitting model to feature number 52, Feature52</span></a>
<a class="sourceLine" id="cb5-75" data-line-number="75"><span class="co">#&gt; 2022-02-26 14:29:16 INFO::Fitting model to feature number 53, Feature53</span></a>
<a class="sourceLine" id="cb5-76" data-line-number="76"><span class="co">#&gt; 2022-02-26 14:29:16 INFO::Fitting model to feature number 54, Feature54</span></a>
<a class="sourceLine" id="cb5-77" data-line-number="77"><span class="co">#&gt; 2022-02-26 14:29:16 INFO::Fitting model to feature number 55, Feature55</span></a>
<a class="sourceLine" id="cb5-78" data-line-number="78"><span class="co">#&gt; 2022-02-26 14:29:16 INFO::Fitting model to feature number 56, Feature56</span></a>
<a class="sourceLine" id="cb5-79" data-line-number="79"><span class="co">#&gt; 2022-02-26 14:29:16 INFO::Fitting model to feature number 57, Feature57</span></a>
<a class="sourceLine" id="cb5-80" data-line-number="80"><span class="co">#&gt; 2022-02-26 14:29:16 INFO::Fitting model to feature number 58, Feature58_TP</span></a>
<a class="sourceLine" id="cb5-81" data-line-number="81"><span class="co">#&gt; 2022-02-26 14:29:16 INFO::Fitting model to feature number 59, Feature59</span></a>
<a class="sourceLine" id="cb5-82" data-line-number="82"><span class="co">#&gt; 2022-02-26 14:29:16 INFO::Fitting model to feature number 60, Feature60</span></a>
<a class="sourceLine" id="cb5-83" data-line-number="83"><span class="co">#&gt; 2022-02-26 14:29:16 INFO::Fitting model to feature number 61, Feature61</span></a>
<a class="sourceLine" id="cb5-84" data-line-number="84"><span class="co">#&gt; 2022-02-26 14:29:16 INFO::Fitting model to feature number 62, Feature62</span></a>
<a class="sourceLine" id="cb5-85" data-line-number="85"><span class="co">#&gt; 2022-02-26 14:29:16 INFO::Fitting model to feature number 63, Feature63</span></a>
<a class="sourceLine" id="cb5-86" data-line-number="86"><span class="co">#&gt; 2022-02-26 14:29:16 INFO::Fitting model to feature number 64, Feature64</span></a>
<a class="sourceLine" id="cb5-87" data-line-number="87"><span class="co">#&gt; 2022-02-26 14:29:16 INFO::Fitting model to feature number 65, Feature65</span></a>
<a class="sourceLine" id="cb5-88" data-line-number="88"><span class="co">#&gt; 2022-02-26 14:29:16 INFO::Fitting model to feature number 66, Feature66</span></a>
<a class="sourceLine" id="cb5-89" data-line-number="89"><span class="co">#&gt; 2022-02-26 14:29:16 INFO::Fitting model to feature number 67, Feature67</span></a>
<a class="sourceLine" id="cb5-90" data-line-number="90"><span class="co">#&gt; 2022-02-26 14:29:17 INFO::Fitting model to feature number 68, Feature68</span></a>
<a class="sourceLine" id="cb5-91" data-line-number="91"><span class="co">#&gt; 2022-02-26 14:29:17 INFO::Fitting model to feature number 69, Feature69</span></a>
<a class="sourceLine" id="cb5-92" data-line-number="92"><span class="co">#&gt; 2022-02-26 14:29:17 INFO::Fitting model to feature number 70, Feature70</span></a>
<a class="sourceLine" id="cb5-93" data-line-number="93"><span class="co">#&gt; 2022-02-26 14:29:17 INFO::Fitting model to feature number 71, Feature71</span></a>
<a class="sourceLine" id="cb5-94" data-line-number="94"><span class="co">#&gt; 2022-02-26 14:29:17 INFO::Fitting model to feature number 72, Feature72</span></a>
<a class="sourceLine" id="cb5-95" data-line-number="95"><span class="co">#&gt; 2022-02-26 14:29:17 INFO::Fitting model to feature number 73, Feature73</span></a>
<a class="sourceLine" id="cb5-96" data-line-number="96"><span class="co">#&gt; 2022-02-26 14:29:17 INFO::Fitting model to feature number 74, Feature74</span></a>
<a class="sourceLine" id="cb5-97" data-line-number="97"><span class="co">#&gt; 2022-02-26 14:29:17 INFO::Fitting model to feature number 75, Feature75</span></a>
<a class="sourceLine" id="cb5-98" data-line-number="98"><span class="co">#&gt; 2022-02-26 14:29:17 INFO::Fitting model to feature number 76, Feature76</span></a>
<a class="sourceLine" id="cb5-99" data-line-number="99"><span class="co">#&gt; 2022-02-26 14:29:17 INFO::Fitting model to feature number 77, Feature77</span></a>
<a class="sourceLine" id="cb5-100" data-line-number="100"><span class="co">#&gt; 2022-02-26 14:29:17 INFO::Fitting model to feature number 78, Feature78</span></a>
<a class="sourceLine" id="cb5-101" data-line-number="101"><span class="co">#&gt; 2022-02-26 14:29:17 INFO::Fitting model to feature number 79, Feature79</span></a>
<a class="sourceLine" id="cb5-102" data-line-number="102"><span class="co">#&gt; 2022-02-26 14:29:17 INFO::Fitting model to feature number 80, Feature80</span></a>
<a class="sourceLine" id="cb5-103" data-line-number="103"><span class="co">#&gt; 2022-02-26 14:29:17 INFO::Fitting model to feature number 81, Feature81</span></a>
<a class="sourceLine" id="cb5-104" data-line-number="104"><span class="co">#&gt; 2022-02-26 14:29:17 INFO::Fitting model to feature number 82, Feature82</span></a>
<a class="sourceLine" id="cb5-105" data-line-number="105"><span class="co">#&gt; 2022-02-26 14:29:17 INFO::Fitting model to feature number 83, Feature83</span></a>
<a class="sourceLine" id="cb5-106" data-line-number="106"><span class="co">#&gt; 2022-02-26 14:29:17 INFO::Fitting model to feature number 84, Feature84</span></a>
<a class="sourceLine" id="cb5-107" data-line-number="107"><span class="co">#&gt; 2022-02-26 14:29:17 INFO::Fitting model to feature number 85, Feature85</span></a>
<a class="sourceLine" id="cb5-108" data-line-number="108"><span class="co">#&gt; 2022-02-26 14:29:17 INFO::Fitting model to feature number 86, Feature86</span></a>
<a class="sourceLine" id="cb5-109" data-line-number="109"><span class="co">#&gt; 2022-02-26 14:29:17 INFO::Fitting model to feature number 87, Feature87</span></a>
<a class="sourceLine" id="cb5-110" data-line-number="110"><span class="co">#&gt; 2022-02-26 14:29:18 INFO::Fitting model to feature number 88, Feature88</span></a>
<a class="sourceLine" id="cb5-111" data-line-number="111"><span class="co">#&gt; 2022-02-26 14:29:18 INFO::Fitting model to feature number 89, Feature89</span></a>
<a class="sourceLine" id="cb5-112" data-line-number="112"><span class="co">#&gt; 2022-02-26 14:29:18 INFO::Fitting model to feature number 90, Feature90</span></a>
<a class="sourceLine" id="cb5-113" data-line-number="113"><span class="co">#&gt; 2022-02-26 14:29:18 INFO::Fitting model to feature number 91, Feature91</span></a>
<a class="sourceLine" id="cb5-114" data-line-number="114"><span class="co">#&gt; 2022-02-26 14:29:18 INFO::Fitting model to feature number 92, Feature92</span></a>
<a class="sourceLine" id="cb5-115" data-line-number="115"><span class="co">#&gt; 2022-02-26 14:29:18 INFO::Fitting model to feature number 93, Feature93</span></a>
<a class="sourceLine" id="cb5-116" data-line-number="116"><span class="co">#&gt; 2022-02-26 14:29:18 INFO::Fitting model to feature number 94, Feature94</span></a>
<a class="sourceLine" id="cb5-117" data-line-number="117"><span class="co">#&gt; 2022-02-26 14:29:18 INFO::Fitting model to feature number 95, Feature95</span></a>
<a class="sourceLine" id="cb5-118" data-line-number="118"><span class="co">#&gt; 2022-02-26 14:29:18 INFO::Fitting model to feature number 96, Feature96</span></a>
<a class="sourceLine" id="cb5-119" data-line-number="119"><span class="co">#&gt; 2022-02-26 14:29:18 INFO::Fitting model to feature number 97, Feature97</span></a>
<a class="sourceLine" id="cb5-120" data-line-number="120"><span class="co">#&gt; 2022-02-26 14:29:18 INFO::Fitting model to feature number 98, Feature98</span></a>
<a class="sourceLine" id="cb5-121" data-line-number="121"><span class="co">#&gt; 2022-02-26 14:29:18 INFO::Fitting model to feature number 99, Feature99</span></a>
<a class="sourceLine" id="cb5-122" data-line-number="122"><span class="co">#&gt; 2022-02-26 14:29:18 INFO::Fitting model to feature number 100, Feature100</span></a>
<a class="sourceLine" id="cb5-123" data-line-number="123"><span class="co">#&gt; 2022-02-26 14:29:18 INFO::Fitting model to feature number 101, Feature101</span></a>
<a class="sourceLine" id="cb5-124" data-line-number="124"><span class="co">#&gt; 2022-02-26 14:29:18 INFO::Fitting model to feature number 102, Feature102</span></a>
<a class="sourceLine" id="cb5-125" data-line-number="125"><span class="co">#&gt; 2022-02-26 14:29:18 INFO::Fitting model to feature number 103, Feature103</span></a>
<a class="sourceLine" id="cb5-126" data-line-number="126"><span class="co">#&gt; 2022-02-26 14:29:19 INFO::Fitting model to feature number 104, Feature104</span></a>
<a class="sourceLine" id="cb5-127" data-line-number="127"><span class="co">#&gt; 2022-02-26 14:29:19 INFO::Fitting model to feature number 105, Feature105</span></a>
<a class="sourceLine" id="cb5-128" data-line-number="128"><span class="co">#&gt; 2022-02-26 14:29:19 INFO::Fitting model to feature number 106, Feature106</span></a>
<a class="sourceLine" id="cb5-129" data-line-number="129"><span class="co">#&gt; 2022-02-26 14:29:19 INFO::Fitting model to feature number 107, Feature107</span></a>
<a class="sourceLine" id="cb5-130" data-line-number="130"><span class="co">#&gt; 2022-02-26 14:29:19 INFO::Fitting model to feature number 108, Feature108</span></a>
<a class="sourceLine" id="cb5-131" data-line-number="131"><span class="co">#&gt; 2022-02-26 14:29:19 INFO::Fitting model to feature number 109, Feature109</span></a>
<a class="sourceLine" id="cb5-132" data-line-number="132"><span class="co">#&gt; 2022-02-26 14:29:19 INFO::Fitting model to feature number 110, Feature110</span></a>
<a class="sourceLine" id="cb5-133" data-line-number="133"><span class="co">#&gt; 2022-02-26 14:29:19 INFO::Fitting model to feature number 111, Feature111</span></a>
<a class="sourceLine" id="cb5-134" data-line-number="134"><span class="co">#&gt; 2022-02-26 14:29:19 INFO::Fitting model to feature number 112, Feature112</span></a>
<a class="sourceLine" id="cb5-135" data-line-number="135"><span class="co">#&gt; 2022-02-26 14:29:19 INFO::Fitting model to feature number 113, Feature113</span></a>
<a class="sourceLine" id="cb5-136" data-line-number="136"><span class="co">#&gt; 2022-02-26 14:29:19 INFO::Fitting model to feature number 114, Feature114</span></a>
<a class="sourceLine" id="cb5-137" data-line-number="137"><span class="co">#&gt; 2022-02-26 14:29:19 INFO::Fitting model to feature number 115, Feature115</span></a>
<a class="sourceLine" id="cb5-138" data-line-number="138"><span class="co">#&gt; 2022-02-26 14:29:19 INFO::Fitting model to feature number 116, Feature116</span></a>
<a class="sourceLine" id="cb5-139" data-line-number="139"><span class="co">#&gt; 2022-02-26 14:29:19 INFO::Fitting model to feature number 117, Feature117</span></a>
<a class="sourceLine" id="cb5-140" data-line-number="140"><span class="co">#&gt; 2022-02-26 14:29:19 INFO::Fitting model to feature number 118, Feature118</span></a>
<a class="sourceLine" id="cb5-141" data-line-number="141"><span class="co">#&gt; 2022-02-26 14:29:19 INFO::Fitting model to feature number 119, Feature119</span></a>
<a class="sourceLine" id="cb5-142" data-line-number="142"><span class="co">#&gt; 2022-02-26 14:29:19 INFO::Fitting model to feature number 120, Feature120</span></a>
<a class="sourceLine" id="cb5-143" data-line-number="143"><span class="co">#&gt; 2022-02-26 14:29:19 INFO::Fitting model to feature number 121, Feature121</span></a>
<a class="sourceLine" id="cb5-144" data-line-number="144"><span class="co">#&gt; 2022-02-26 14:29:20 INFO::Fitting model to feature number 122, Feature122</span></a>
<a class="sourceLine" id="cb5-145" data-line-number="145"><span class="co">#&gt; 2022-02-26 14:29:20 INFO::Fitting model to feature number 123, Feature123</span></a>
<a class="sourceLine" id="cb5-146" data-line-number="146"><span class="co">#&gt; 2022-02-26 14:29:20 INFO::Fitting model to feature number 124, Feature124</span></a>
<a class="sourceLine" id="cb5-147" data-line-number="147"><span class="co">#&gt; 2022-02-26 14:29:20 INFO::Fitting model to feature number 125, Feature125</span></a>
<a class="sourceLine" id="cb5-148" data-line-number="148"><span class="co">#&gt; 2022-02-26 14:29:20 INFO::Fitting model to feature number 126, Feature126</span></a>
<a class="sourceLine" id="cb5-149" data-line-number="149"><span class="co">#&gt; 2022-02-26 14:29:20 INFO::Fitting model to feature number 127, Feature127</span></a>
<a class="sourceLine" id="cb5-150" data-line-number="150"><span class="co">#&gt; 2022-02-26 14:29:20 INFO::Fitting model to feature number 128, Feature128</span></a>
<a class="sourceLine" id="cb5-151" data-line-number="151"><span class="co">#&gt; 2022-02-26 14:29:20 INFO::Fitting model to feature number 129, Feature129</span></a>
<a class="sourceLine" id="cb5-152" data-line-number="152"><span class="co">#&gt; 2022-02-26 14:29:20 INFO::Fitting model to feature number 130, Feature130</span></a>
<a class="sourceLine" id="cb5-153" data-line-number="153"><span class="co">#&gt; 2022-02-26 14:29:20 INFO::Fitting model to feature number 131, Feature131</span></a>
<a class="sourceLine" id="cb5-154" data-line-number="154"><span class="co">#&gt; 2022-02-26 14:29:20 INFO::Fitting model to feature number 132, Feature132</span></a>
<a class="sourceLine" id="cb5-155" data-line-number="155"><span class="co">#&gt; 2022-02-26 14:29:21 INFO::Fitting model to feature number 133, Feature133</span></a>
<a class="sourceLine" id="cb5-156" data-line-number="156"><span class="co">#&gt; 2022-02-26 14:29:21 INFO::Fitting model to feature number 134, Feature134</span></a>
<a class="sourceLine" id="cb5-157" data-line-number="157"><span class="co">#&gt; 2022-02-26 14:29:21 INFO::Fitting model to feature number 135, Feature135</span></a>
<a class="sourceLine" id="cb5-158" data-line-number="158"><span class="co">#&gt; 2022-02-26 14:29:21 INFO::Fitting model to feature number 136, Feature136</span></a>
<a class="sourceLine" id="cb5-159" data-line-number="159"><span class="co">#&gt; 2022-02-26 14:29:21 INFO::Fitting model to feature number 137, Feature137</span></a>
<a class="sourceLine" id="cb5-160" data-line-number="160"><span class="co">#&gt; 2022-02-26 14:29:21 INFO::Fitting model to feature number 138, Feature138</span></a>
<a class="sourceLine" id="cb5-161" data-line-number="161"><span class="co">#&gt; 2022-02-26 14:29:21 INFO::Fitting model to feature number 139, Feature139</span></a>
<a class="sourceLine" id="cb5-162" data-line-number="162"><span class="co">#&gt; 2022-02-26 14:29:21 INFO::Fitting model to feature number 140, Feature140</span></a>
<a class="sourceLine" id="cb5-163" data-line-number="163"><span class="co">#&gt; 2022-02-26 14:29:22 INFO::Fitting model to feature number 141, Feature141</span></a>
<a class="sourceLine" id="cb5-164" data-line-number="164"><span class="co">#&gt; 2022-02-26 14:29:22 INFO::Fitting model to feature number 142, Feature142</span></a>
<a class="sourceLine" id="cb5-165" data-line-number="165"><span class="co">#&gt; 2022-02-26 14:29:22 INFO::Fitting model to feature number 143, Feature143</span></a>
<a class="sourceLine" id="cb5-166" data-line-number="166"><span class="co">#&gt; 2022-02-26 14:29:22 INFO::Fitting model to feature number 144, Feature144</span></a>
<a class="sourceLine" id="cb5-167" data-line-number="167"><span class="co">#&gt; 2022-02-26 14:29:22 INFO::Fitting model to feature number 145, Feature145</span></a>
<a class="sourceLine" id="cb5-168" data-line-number="168"><span class="co">#&gt; 2022-02-26 14:29:22 INFO::Fitting model to feature number 146, Feature146</span></a>
<a class="sourceLine" id="cb5-169" data-line-number="169"><span class="co">#&gt; 2022-02-26 14:29:22 INFO::Fitting model to feature number 147, Feature147</span></a>
<a class="sourceLine" id="cb5-170" data-line-number="170"><span class="co">#&gt; 2022-02-26 14:29:22 INFO::Fitting model to feature number 148, Feature148</span></a>
<a class="sourceLine" id="cb5-171" data-line-number="171"><span class="co">#&gt; 2022-02-26 14:29:22 INFO::Fitting model to feature number 149, Feature149</span></a>
<a class="sourceLine" id="cb5-172" data-line-number="172"><span class="co">#&gt; 2022-02-26 14:29:22 INFO::Fitting model to feature number 150, Feature150</span></a>
<a class="sourceLine" id="cb5-173" data-line-number="173"><span class="co">#&gt; 2022-02-26 14:29:22 INFO::Fitting model to feature number 151, Feature151</span></a>
<a class="sourceLine" id="cb5-174" data-line-number="174"><span class="co">#&gt; 2022-02-26 14:29:22 INFO::Fitting model to feature number 152, Feature152</span></a>
<a class="sourceLine" id="cb5-175" data-line-number="175"><span class="co">#&gt; 2022-02-26 14:29:22 INFO::Fitting model to feature number 153, Feature153</span></a>
<a class="sourceLine" id="cb5-176" data-line-number="176"><span class="co">#&gt; 2022-02-26 14:29:22 INFO::Fitting model to feature number 154, Feature154</span></a>
<a class="sourceLine" id="cb5-177" data-line-number="177"><span class="co">#&gt; 2022-02-26 14:29:23 INFO::Fitting model to feature number 155, Feature155</span></a>
<a class="sourceLine" id="cb5-178" data-line-number="178"><span class="co">#&gt; 2022-02-26 14:29:23 INFO::Fitting model to feature number 156, Feature156</span></a>
<a class="sourceLine" id="cb5-179" data-line-number="179"><span class="co">#&gt; 2022-02-26 14:29:23 INFO::Fitting model to feature number 157, Feature157</span></a>
<a class="sourceLine" id="cb5-180" data-line-number="180"><span class="co">#&gt; 2022-02-26 14:29:23 INFO::Fitting model to feature number 158, Feature158</span></a>
<a class="sourceLine" id="cb5-181" data-line-number="181"><span class="co">#&gt; 2022-02-26 14:29:23 INFO::Fitting model to feature number 159, Feature159</span></a>
<a class="sourceLine" id="cb5-182" data-line-number="182"><span class="co">#&gt; 2022-02-26 14:29:23 INFO::Fitting model to feature number 160, Feature160</span></a>
<a class="sourceLine" id="cb5-183" data-line-number="183"><span class="co">#&gt; 2022-02-26 14:29:23 INFO::Fitting model to feature number 161, Feature161</span></a>
<a class="sourceLine" id="cb5-184" data-line-number="184"><span class="co">#&gt; 2022-02-26 14:29:23 INFO::Fitting model to feature number 162, Feature162</span></a>
<a class="sourceLine" id="cb5-185" data-line-number="185"><span class="co">#&gt; 2022-02-26 14:29:23 INFO::Fitting model to feature number 163, Feature163</span></a>
<a class="sourceLine" id="cb5-186" data-line-number="186"><span class="co">#&gt; 2022-02-26 14:29:23 INFO::Fitting model to feature number 164, Feature164</span></a>
<a class="sourceLine" id="cb5-187" data-line-number="187"><span class="co">#&gt; 2022-02-26 14:29:23 INFO::Fitting model to feature number 165, Feature165_TP</span></a>
<a class="sourceLine" id="cb5-188" data-line-number="188"><span class="co">#&gt; 2022-02-26 14:29:23 INFO::Fitting model to feature number 166, Feature166</span></a>
<a class="sourceLine" id="cb5-189" data-line-number="189"><span class="co">#&gt; 2022-02-26 14:29:23 INFO::Fitting model to feature number 167, Feature167</span></a>
<a class="sourceLine" id="cb5-190" data-line-number="190"><span class="co">#&gt; 2022-02-26 14:29:23 INFO::Fitting model to feature number 168, Feature168</span></a>
<a class="sourceLine" id="cb5-191" data-line-number="191"><span class="co">#&gt; 2022-02-26 14:29:23 INFO::Fitting model to feature number 169, Feature169</span></a>
<a class="sourceLine" id="cb5-192" data-line-number="192"><span class="co">#&gt; 2022-02-26 14:29:23 INFO::Fitting model to feature number 170, Feature170</span></a>
<a class="sourceLine" id="cb5-193" data-line-number="193"><span class="co">#&gt; 2022-02-26 14:29:24 INFO::Fitting model to feature number 171, Feature171</span></a>
<a class="sourceLine" id="cb5-194" data-line-number="194"><span class="co">#&gt; 2022-02-26 14:29:24 INFO::Fitting model to feature number 172, Feature172_TP</span></a>
<a class="sourceLine" id="cb5-195" data-line-number="195"><span class="co">#&gt; 2022-02-26 14:29:24 INFO::Fitting model to feature number 173, Feature173</span></a>
<a class="sourceLine" id="cb5-196" data-line-number="196"><span class="co">#&gt; 2022-02-26 14:29:24 INFO::Fitting model to feature number 174, Feature174</span></a>
<a class="sourceLine" id="cb5-197" data-line-number="197"><span class="co">#&gt; 2022-02-26 14:29:24 INFO::Fitting model to feature number 175, Feature175</span></a>
<a class="sourceLine" id="cb5-198" data-line-number="198"><span class="co">#&gt; 2022-02-26 14:29:24 INFO::Fitting model to feature number 176, Feature176</span></a>
<a class="sourceLine" id="cb5-199" data-line-number="199"><span class="co">#&gt; 2022-02-26 14:29:24 INFO::Fitting model to feature number 177, Feature177</span></a>
<a class="sourceLine" id="cb5-200" data-line-number="200"><span class="co">#&gt; 2022-02-26 14:29:24 INFO::Fitting model to feature number 178, Feature178</span></a>
<a class="sourceLine" id="cb5-201" data-line-number="201"><span class="co">#&gt; 2022-02-26 14:29:24 INFO::Fitting model to feature number 179, Feature179</span></a>
<a class="sourceLine" id="cb5-202" data-line-number="202"><span class="co">#&gt; 2022-02-26 14:29:24 INFO::Fitting model to feature number 180, Feature180</span></a>
<a class="sourceLine" id="cb5-203" data-line-number="203"><span class="co">#&gt; 2022-02-26 14:29:24 INFO::Fitting model to feature number 181, Feature181</span></a>
<a class="sourceLine" id="cb5-204" data-line-number="204"><span class="co">#&gt; 2022-02-26 14:29:24 INFO::Fitting model to feature number 182, Feature182</span></a>
<a class="sourceLine" id="cb5-205" data-line-number="205"><span class="co">#&gt; 2022-02-26 14:29:24 INFO::Fitting model to feature number 183, Feature183</span></a>
<a class="sourceLine" id="cb5-206" data-line-number="206"><span class="co">#&gt; 2022-02-26 14:29:25 INFO::Fitting model to feature number 184, Feature184</span></a>
<a class="sourceLine" id="cb5-207" data-line-number="207"><span class="co">#&gt; 2022-02-26 14:29:25 INFO::Fitting model to feature number 185, Feature185</span></a>
<a class="sourceLine" id="cb5-208" data-line-number="208"><span class="co">#&gt; 2022-02-26 14:29:25 INFO::Fitting model to feature number 186, Feature186</span></a>
<a class="sourceLine" id="cb5-209" data-line-number="209"><span class="co">#&gt; 2022-02-26 14:29:25 INFO::Fitting model to feature number 187, Feature187</span></a>
<a class="sourceLine" id="cb5-210" data-line-number="210"><span class="co">#&gt; 2022-02-26 14:29:25 INFO::Fitting model to feature number 188, Feature188</span></a>
<a class="sourceLine" id="cb5-211" data-line-number="211"><span class="co">#&gt; 2022-02-26 14:29:25 INFO::Fitting model to feature number 189, Feature189</span></a>
<a class="sourceLine" id="cb5-212" data-line-number="212"><span class="co">#&gt; 2022-02-26 14:29:25 INFO::Fitting model to feature number 190, Feature190</span></a>
<a class="sourceLine" id="cb5-213" data-line-number="213"><span class="co">#&gt; 2022-02-26 14:29:25 INFO::Fitting model to feature number 191, Feature191</span></a>
<a class="sourceLine" id="cb5-214" data-line-number="214"><span class="co">#&gt; 2022-02-26 14:29:25 INFO::Fitting model to feature number 192, Feature192</span></a>
<a class="sourceLine" id="cb5-215" data-line-number="215"><span class="co">#&gt; 2022-02-26 14:29:25 INFO::Fitting model to feature number 193, Feature193</span></a>
<a class="sourceLine" id="cb5-216" data-line-number="216"><span class="co">#&gt; 2022-02-26 14:29:25 INFO::Fitting model to feature number 194, Feature194</span></a>
<a class="sourceLine" id="cb5-217" data-line-number="217"><span class="co">#&gt; 2022-02-26 14:29:25 INFO::Fitting model to feature number 195, Feature195</span></a>
<a class="sourceLine" id="cb5-218" data-line-number="218"><span class="co">#&gt; 2022-02-26 14:29:25 INFO::Fitting model to feature number 196, Feature196</span></a>
<a class="sourceLine" id="cb5-219" data-line-number="219"><span class="co">#&gt; 2022-02-26 14:29:25 INFO::Fitting model to feature number 197, Feature197</span></a>
<a class="sourceLine" id="cb5-220" data-line-number="220"><span class="co">#&gt; 2022-02-26 14:29:25 INFO::Fitting model to feature number 198, Feature198</span></a>
<a class="sourceLine" id="cb5-221" data-line-number="221"><span class="co">#&gt; 2022-02-26 14:29:25 INFO::Fitting model to feature number 199, Feature199</span></a>
<a class="sourceLine" id="cb5-222" data-line-number="222"><span class="co">#&gt; 2022-02-26 14:29:25 INFO::Fitting model to feature number 200, Feature200</span></a>
<a class="sourceLine" id="cb5-223" data-line-number="223"><span class="co">#&gt; 2022-02-26 14:29:26 INFO::Counting prevalence for each feature</span></a>
<a class="sourceLine" id="cb5-224" data-line-number="224"><span class="co">#&gt; 2022-02-26 14:29:26 INFO::Writing all results to file (ordered by increasing q-values): ./demo_output/SimData/all_results.tsv</span></a>
<a class="sourceLine" id="cb5-225" data-line-number="225"><span class="co">#&gt; 2022-02-26 14:29:26 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.050000 ) to file (ordered by increasing q-values): ./demo_output/SimData/significant_results.tsv</span></a></code></pre></div>
<p>After computation, <code>SimData</code> is a data frame containing coefficient estimates, p-values, and q-values (multiplicity-adjusted p-values) along with other parameter estimates from the fitted per-feature models. By default, p-values are adjusted with the Benjamini-Hochberg method.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co"># Table with DA analysis results - significant features at 5% FDR</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2">SimData[SimData<span class="op">$</span>qval<span class="op">&lt;</span><span class="fl">0.05</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">7</span>]</a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="co">#&gt;         feature  metadata value      coef     stderr         pval         qval</span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co">#&gt; 1 Feature172_TP Metadata1     1 1.0330517 0.01733483 8.550106e-79 1.710021e-76</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="co">#&gt; 2 Feature165_TP Metadata1     1 1.0742294 0.02012774 3.069442e-74 3.069442e-72</span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co">#&gt; 3  Feature58_TP Metadata1     1 1.0637617 0.02265985 5.535149e-69 3.690099e-67</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">#&gt; 4  Feature46_TP Metadata1     1 0.8285178 0.01958388 9.219455e-65 4.609728e-63</span></a></code></pre></div>
<p>Using the adjusted p-values, there are 4 statistically significant differentially abundant features at 5% FDR, which matches exactly with the <code>ground truth</code> information from SparseDOSSA indicating that Tweedieverse is able to call out the potential true positive features while also maintaining a low false positive rate.</p>
</div>
<div id="example-2---multivariable-association-on-hmp2-baseline-microbiomes" class="section level3">
<h3>Example 2 - Multivariable Association on HMP2 Baseline Microbiomes</h3>
<p>For the second example, we will analyze the microbiome taxonomic data from the <a href="https://www.hmpdacc.org/ihmp/">integrative Human Microbiome Project</a>, available from the <a href="https://github.com/biobakery/Maaslin2">R/Bioconductor package MaAsLin2</a>. Check out the <a href="https://www.nature.com/articles/s41586-019-1237-9">original paper</a> for details on the related statistical modeling. Compared to Example 1, Example 2 is arguably complex with multiple covariates and repeated measures. Focusing on the HMP2 baseline samples, we will fit a zero-inflated model unlike Example 1. Following original publication, we specify age, antibiotic usage, and disease status (diagnosis) as fixed effects.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="co"># Load data </span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="kw">library</span>(data.table)</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">input_features &lt;-<span class="st"> </span><span class="kw">fread</span>(<span class="st">&quot;https://raw.githubusercontent.com/biobakery/Maaslin2/master/inst/extdata/HMP2_taxonomy.tsv&quot;</span>, <span class="dt">sep =</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</a>
<a class="sourceLine" id="cb7-4" data-line-number="4">input_metadata &lt;-<span class="kw">fread</span>(<span class="st">&quot;https://raw.githubusercontent.com/biobakery/Maaslin2/master/inst/extdata/HMP2_metadata.tsv&quot;</span>, <span class="dt">sep =</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</a>
<a class="sourceLine" id="cb7-5" data-line-number="5"></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="co"># Format data </span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="kw">library</span>(tibble)</a>
<a class="sourceLine" id="cb7-8" data-line-number="8">input_features&lt;-<span class="st"> </span><span class="kw">column_to_rownames</span>(input_features, <span class="st">'ID'</span>)</a>
<a class="sourceLine" id="cb7-9" data-line-number="9">input_metadata&lt;-<span class="st"> </span><span class="kw">column_to_rownames</span>(input_metadata, <span class="st">'ID'</span>)</a>
<a class="sourceLine" id="cb7-10" data-line-number="10"></a>
<a class="sourceLine" id="cb7-11" data-line-number="11"><span class="co"># Extract baseline samples</span></a>
<a class="sourceLine" id="cb7-12" data-line-number="12">input_metadata&lt;-input_metadata[<span class="kw">order</span>(input_metadata<span class="op">$</span>collection),] <span class="co"># Order to extract first sample</span></a>
<a class="sourceLine" id="cb7-13" data-line-number="13"></a>
<a class="sourceLine" id="cb7-14" data-line-number="14"><span class="co"># Create baseline data </span></a>
<a class="sourceLine" id="cb7-15" data-line-number="15">metadata_baseline&lt;-<span class="st"> </span>input_metadata[<span class="op">!</span><span class="kw">duplicated</span>(input_metadata<span class="op">$</span>subject),]</a>
<a class="sourceLine" id="cb7-16" data-line-number="16">metadata_baseline&lt;-<span class="st"> </span>metadata_baseline[metadata_baseline<span class="op">$</span>collection <span class="op">&lt;</span><span class="st"> </span><span class="dv">3</span>,]</a>
<a class="sourceLine" id="cb7-17" data-line-number="17"></a>
<a class="sourceLine" id="cb7-18" data-line-number="18"><span class="co"># Remove irrelevant columns from both </span></a>
<a class="sourceLine" id="cb7-19" data-line-number="19">metadata_baseline&lt;-metadata_baseline <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(age, antibiotics, diagnosis)</a>
<a class="sourceLine" id="cb7-20" data-line-number="20">metadata_final&lt;-metadata_baseline</a>
<a class="sourceLine" id="cb7-21" data-line-number="21"></a>
<a class="sourceLine" id="cb7-22" data-line-number="22"><span class="co"># Remove missing data and set reference</span></a>
<a class="sourceLine" id="cb7-23" data-line-number="23">metadata_final&lt;-<span class="st"> </span>metadata_final[<span class="kw">complete.cases</span>(metadata_final), ]</a>
<a class="sourceLine" id="cb7-24" data-line-number="24">metadata_final<span class="op">$</span>diagnosis&lt;-<span class="kw">as.factor</span>(metadata_final<span class="op">$</span>diagnosis)</a>
<a class="sourceLine" id="cb7-25" data-line-number="25">metadata_final<span class="op">$</span>diagnosis&lt;-<span class="kw">relevel</span>(metadata_final<span class="op">$</span>diagnosis, <span class="dt">ref =</span> <span class="st">&quot;nonIBD&quot;</span>) <span class="co">#NonIBD as reference</span></a>
<a class="sourceLine" id="cb7-26" data-line-number="26"></a>
<a class="sourceLine" id="cb7-27" data-line-number="27"><span class="co"># Adjust features accordingly </span></a>
<a class="sourceLine" id="cb7-28" data-line-number="28">features_final&lt;-input_features[<span class="kw">rownames</span>(metadata_final),]</a>
<a class="sourceLine" id="cb7-29" data-line-number="29"></a>
<a class="sourceLine" id="cb7-30" data-line-number="30"><span class="co"># Fit a ZICP model </span></a>
<a class="sourceLine" id="cb7-31" data-line-number="31"><span class="cf">if</span> (<span class="op">!</span><span class="kw">dir.exists</span>(<span class="st">'demo_output'</span>)) <span class="kw">dir.create</span>(<span class="st">'demo_output'</span>)</a>
<a class="sourceLine" id="cb7-32" data-line-number="32">HMP2_base &lt;-<span class="st"> </span><span class="kw">Tweedieverse</span>(features_final, </a>
<a class="sourceLine" id="cb7-33" data-line-number="33">                     metadata_final,</a>
<a class="sourceLine" id="cb7-34" data-line-number="34">                     <span class="dt">output =</span> <span class="st">'./demo_output/HMP2_base'</span>, <span class="co"># Assuming demo_output exists</span></a>
<a class="sourceLine" id="cb7-35" data-line-number="35">                     <span class="dt">fixed_effects =</span> <span class="kw">c</span>(<span class="st">'age'</span>, <span class="st">'antibiotics'</span>,<span class="st">'diagnosis'</span>),</a>
<a class="sourceLine" id="cb7-36" data-line-number="36">                     <span class="dt">base_model =</span> <span class="st">'ZICP'</span>,</a>
<a class="sourceLine" id="cb7-37" data-line-number="37">                     <span class="dt">max_significance =</span> <span class="fl">0.25</span>, <span class="co"># To match original study</span></a>
<a class="sourceLine" id="cb7-38" data-line-number="38">                     <span class="dt">adjust_offset =</span> <span class="ot">FALSE</span>) <span class="co"># No offset as the values are relative abundances)</span></a>
<a class="sourceLine" id="cb7-39" data-line-number="39"><span class="co">#&gt; [1] &quot;Warning: Deleting existing log file: ./demo_output/HMP2_base/Tweedieverse.log&quot;</span></a>
<a class="sourceLine" id="cb7-40" data-line-number="40"><span class="co">#&gt; 2022-02-26 14:29:26 INFO::Writing function arguments to log file</span></a>
<a class="sourceLine" id="cb7-41" data-line-number="41"><span class="co">#&gt; 2022-02-26 14:29:26 INFO::Determining format of input files</span></a>
<a class="sourceLine" id="cb7-42" data-line-number="42"><span class="co">#&gt; 2022-02-26 14:29:26 INFO::Input format is data samples as rows and metadata samples as rows</span></a>
<a class="sourceLine" id="cb7-43" data-line-number="43"><span class="co">#&gt; 2022-02-26 14:29:26 INFO::Factor detected for categorial metadata 'diagnosis'. Provide a reference argument or manually set factor ordering to change reference level.</span></a>
<a class="sourceLine" id="cb7-44" data-line-number="44"><span class="co">#&gt; 2022-02-26 14:29:26 INFO::Filter data based on min abundance and min prevalence</span></a>
<a class="sourceLine" id="cb7-45" data-line-number="45"><span class="co">#&gt; 2022-02-26 14:29:26 INFO::Total samples in data: 111</span></a>
<a class="sourceLine" id="cb7-46" data-line-number="46"><span class="co">#&gt; 2022-02-26 14:29:26 INFO::Min samples required with min abundance for a feature not to be filtered: 11.100000</span></a>
<a class="sourceLine" id="cb7-47" data-line-number="47"><span class="co">#&gt; 2022-02-26 14:29:26 INFO::Total filtered features with prevalence-abundance filtering: 0</span></a>
<a class="sourceLine" id="cb7-48" data-line-number="48"><span class="co">#&gt; 2022-02-26 14:29:26 INFO::Filtered feature names:</span></a>
<a class="sourceLine" id="cb7-49" data-line-number="49"><span class="co">#&gt; 2022-02-26 14:29:26 INFO::Total filtered features with variance filtering: 0</span></a>
<a class="sourceLine" id="cb7-50" data-line-number="50"><span class="co">#&gt; 2022-02-26 14:29:26 INFO::Filtered feature names:</span></a>
<a class="sourceLine" id="cb7-51" data-line-number="51"><span class="co">#&gt; 2022-02-26 14:29:26 INFO::Excluded metadata with entropy less or equal to 0:</span></a>
<a class="sourceLine" id="cb7-52" data-line-number="52"><span class="co">#&gt; 2022-02-26 14:29:26 INFO::Formula for fixed effects: expr ~  age + antibiotics + diagnosis</span></a>
<a class="sourceLine" id="cb7-53" data-line-number="53"><span class="co">#&gt; 2022-02-26 14:29:26 INFO::Applying z-score to standardize continuous metadata</span></a>
<a class="sourceLine" id="cb7-54" data-line-number="54"><span class="co">#&gt; 2022-02-26 14:29:26 INFO::Running selected analysis method: ZICP</span></a>
<a class="sourceLine" id="cb7-55" data-line-number="55"><span class="co">#&gt; 2022-02-26 14:29:26 INFO::Fitting model to feature number 1, Bifidobacterium adolescentis</span></a>
<a class="sourceLine" id="cb7-56" data-line-number="56"><span class="co">#&gt; 2022-02-26 14:29:26 INFO::Fitting model to feature number 2, Bifidobacterium bifidum</span></a>
<a class="sourceLine" id="cb7-57" data-line-number="57"><span class="co">#&gt; 2022-02-26 14:29:27 INFO::Fitting model to feature number 3, Bifidobacterium longum</span></a>
<a class="sourceLine" id="cb7-58" data-line-number="58"><span class="co">#&gt; 2022-02-26 14:29:27 INFO::Fitting model to feature number 4, Bifidobacterium pseudocatenulatum</span></a>
<a class="sourceLine" id="cb7-59" data-line-number="59"><span class="co">#&gt; 2022-02-26 14:29:27 INFO::Fitting model to feature number 5, Collinsella aerofaciens</span></a>
<a class="sourceLine" id="cb7-60" data-line-number="60"><span class="co">#&gt; 2022-02-26 14:29:27 INFO::Fitting model to feature number 6, Bacteroides caccae</span></a>
<a class="sourceLine" id="cb7-61" data-line-number="61"><span class="co">#&gt; 2022-02-26 14:29:28 INFO::Fitting model to feature number 7, Bacteroides cellulosilyticus</span></a>
<a class="sourceLine" id="cb7-62" data-line-number="62"><span class="co">#&gt; 2022-02-26 14:29:28 INFO::Fitting model to feature number 8, Bacteroides dorei</span></a>
<a class="sourceLine" id="cb7-63" data-line-number="63"><span class="co">#&gt; 2022-02-26 14:29:28 INFO::Fitting model to feature number 9, Bacteroides eggerthii</span></a>
<a class="sourceLine" id="cb7-64" data-line-number="64"><span class="co">#&gt; 2022-02-26 14:29:28 INFO::Fitting model to feature number 10, Bacteroides faecis</span></a>
<a class="sourceLine" id="cb7-65" data-line-number="65"><span class="co">#&gt; 2022-02-26 14:29:29 INFO::Fitting model to feature number 11, Bacteroides finegoldii</span></a>
<a class="sourceLine" id="cb7-66" data-line-number="66"><span class="co">#&gt; 2022-02-26 14:29:29 INFO::Fitting model to feature number 12, Bacteroides fragilis</span></a>
<a class="sourceLine" id="cb7-67" data-line-number="67"><span class="co">#&gt; 2022-02-26 14:29:29 INFO::Fitting model to feature number 13, Bacteroides intestinalis</span></a>
<a class="sourceLine" id="cb7-68" data-line-number="68"><span class="co">#&gt; 2022-02-26 14:29:29 INFO::Fitting model to feature number 14, Bacteroides massiliensis</span></a>
<a class="sourceLine" id="cb7-69" data-line-number="69"><span class="co">#&gt; 2022-02-26 14:29:30 INFO::Fitting model to feature number 15, Bacteroides ovatus</span></a>
<a class="sourceLine" id="cb7-70" data-line-number="70"><span class="co">#&gt; 2022-02-26 14:29:30 INFO::Fitting model to feature number 16, Bacteroides salyersiae</span></a>
<a class="sourceLine" id="cb7-71" data-line-number="71"><span class="co">#&gt; 2022-02-26 14:29:30 INFO::Fitting model to feature number 17, Bacteroides stercoris</span></a>
<a class="sourceLine" id="cb7-72" data-line-number="72"><span class="co">#&gt; 2022-02-26 14:29:30 INFO::Fitting model to feature number 18, Bacteroides thetaiotaomicron</span></a>
<a class="sourceLine" id="cb7-73" data-line-number="73"><span class="co">#&gt; 2022-02-26 14:29:31 INFO::Fitting model to feature number 19, Bacteroides uniformis</span></a>
<a class="sourceLine" id="cb7-74" data-line-number="74"><span class="co">#&gt; 2022-02-26 14:29:31 INFO::Fitting model to feature number 20, Bacteroides vulgatus</span></a>
<a class="sourceLine" id="cb7-75" data-line-number="75"><span class="co">#&gt; 2022-02-26 14:29:31 INFO::Fitting model to feature number 21, Bacteroides xylanisolvens</span></a>
<a class="sourceLine" id="cb7-76" data-line-number="76"><span class="co">#&gt; 2022-02-26 14:29:31 INFO::Fitting model to feature number 22, Bacteroidales bacterium ph8</span></a>
<a class="sourceLine" id="cb7-77" data-line-number="77"><span class="co">#&gt; 2022-02-26 14:29:32 INFO::Fitting model to feature number 23, Barnesiella intestinihominis</span></a>
<a class="sourceLine" id="cb7-78" data-line-number="78"><span class="co">#&gt; 2022-02-26 14:29:32 INFO::Fitting model to feature number 24, Coprobacter fastidiosus</span></a>
<a class="sourceLine" id="cb7-79" data-line-number="79"><span class="co">#&gt; 2022-02-26 14:29:32 INFO::Fitting model to feature number 25, Odoribacter splanchnicus</span></a>
<a class="sourceLine" id="cb7-80" data-line-number="80"><span class="co">#&gt; 2022-02-26 14:29:32 INFO::Fitting model to feature number 26, Parabacteroides distasonis</span></a>
<a class="sourceLine" id="cb7-81" data-line-number="81"><span class="co">#&gt; 2022-02-26 14:29:33 INFO::Fitting model to feature number 27, Parabacteroides goldsteinii</span></a>
<a class="sourceLine" id="cb7-82" data-line-number="82"><span class="co">#&gt; Error in solve.default(hn) : </span></a>
<a class="sourceLine" id="cb7-83" data-line-number="83"><span class="co">#&gt;   Lapack routine dgesv: system is exactly singular: U[4,4] = 0</span></a>
<a class="sourceLine" id="cb7-84" data-line-number="84"><span class="co">#&gt; 2022-02-26 14:29:33 </span><span class="al">WARNING</span><span class="co">::Fitting problem for feature 27 returning NA</span></a>
<a class="sourceLine" id="cb7-85" data-line-number="85"><span class="co">#&gt; 2022-02-26 14:29:33 INFO::Fitting model to feature number 28, Parabacteroides merdae</span></a>
<a class="sourceLine" id="cb7-86" data-line-number="86"><span class="co">#&gt; 2022-02-26 14:29:33 INFO::Fitting model to feature number 29, Parabacteroides unclassified</span></a>
<a class="sourceLine" id="cb7-87" data-line-number="87"><span class="co">#&gt; 2022-02-26 14:29:34 INFO::Fitting model to feature number 30, Paraprevotella clara</span></a>
<a class="sourceLine" id="cb7-88" data-line-number="88"><span class="co">#&gt; 2022-02-26 14:29:34 INFO::Fitting model to feature number 31, Paraprevotella unclassified</span></a>
<a class="sourceLine" id="cb7-89" data-line-number="89"><span class="co">#&gt; 2022-02-26 14:29:34 INFO::Fitting model to feature number 32, Prevotella copri</span></a>
<a class="sourceLine" id="cb7-90" data-line-number="90"><span class="co">#&gt; 2022-02-26 14:29:35 INFO::Fitting model to feature number 33, Alistipes finegoldii</span></a>
<a class="sourceLine" id="cb7-91" data-line-number="91"><span class="co">#&gt; 2022-02-26 14:29:35 INFO::Fitting model to feature number 34, Alistipes onderdonkii</span></a>
<a class="sourceLine" id="cb7-92" data-line-number="92"><span class="co">#&gt; 2022-02-26 14:29:35 INFO::Fitting model to feature number 35, Alistipes putredinis</span></a>
<a class="sourceLine" id="cb7-93" data-line-number="93"><span class="co">#&gt; 2022-02-26 14:29:35 INFO::Fitting model to feature number 36, Alistipes shahii</span></a>
<a class="sourceLine" id="cb7-94" data-line-number="94"><span class="co">#&gt; 2022-02-26 14:29:36 INFO::Fitting model to feature number 37, Alistipes unclassified</span></a>
<a class="sourceLine" id="cb7-95" data-line-number="95"><span class="co">#&gt; 2022-02-26 14:29:36 INFO::Fitting model to feature number 38, Streptococcus salivarius</span></a>
<a class="sourceLine" id="cb7-96" data-line-number="96"><span class="co">#&gt; 2022-02-26 14:29:36 INFO::Fitting model to feature number 39, Clostridium bolteae</span></a>
<a class="sourceLine" id="cb7-97" data-line-number="97"><span class="co">#&gt; 2022-02-26 14:29:36 INFO::Fitting model to feature number 40, Clostridium citroniae</span></a>
<a class="sourceLine" id="cb7-98" data-line-number="98"><span class="co">#&gt; 2022-02-26 14:29:37 INFO::Fitting model to feature number 41, Clostridium clostridioforme</span></a>
<a class="sourceLine" id="cb7-99" data-line-number="99"><span class="co">#&gt; 2022-02-26 14:29:37 INFO::Fitting model to feature number 42, Clostridium hathewayi</span></a>
<a class="sourceLine" id="cb7-100" data-line-number="100"><span class="co">#&gt; 2022-02-26 14:29:37 INFO::Fitting model to feature number 43, Clostridium leptum</span></a>
<a class="sourceLine" id="cb7-101" data-line-number="101"><span class="co">#&gt; 2022-02-26 14:29:37 INFO::Fitting model to feature number 44, Clostridium nexile</span></a>
<a class="sourceLine" id="cb7-102" data-line-number="102"><span class="co">#&gt; 2022-02-26 14:29:37 INFO::Fitting model to feature number 45, Clostridium symbiosum</span></a>
<a class="sourceLine" id="cb7-103" data-line-number="103"><span class="co">#&gt; 2022-02-26 14:29:38 INFO::Fitting model to feature number 46, Flavonifractor plautii</span></a>
<a class="sourceLine" id="cb7-104" data-line-number="104"><span class="co">#&gt; 2022-02-26 14:29:38 INFO::Fitting model to feature number 47, Eubacterium eligens</span></a>
<a class="sourceLine" id="cb7-105" data-line-number="105"><span class="co">#&gt; 2022-02-26 14:29:38 INFO::Fitting model to feature number 48, Eubacterium hallii</span></a>
<a class="sourceLine" id="cb7-106" data-line-number="106"><span class="co">#&gt; 2022-02-26 14:29:39 INFO::Fitting model to feature number 49, Eubacterium rectale</span></a>
<a class="sourceLine" id="cb7-107" data-line-number="107"><span class="co">#&gt; 2022-02-26 14:29:39 INFO::Fitting model to feature number 50, Eubacterium siraeum</span></a>
<a class="sourceLine" id="cb7-108" data-line-number="108"><span class="co">#&gt; 2022-02-26 14:29:39 INFO::Fitting model to feature number 51, Eubacterium sp 3 1 31</span></a>
<a class="sourceLine" id="cb7-109" data-line-number="109"><span class="co">#&gt; 2022-02-26 14:29:40 INFO::Fitting model to feature number 52, Eubacterium ventriosum</span></a>
<a class="sourceLine" id="cb7-110" data-line-number="110"><span class="co">#&gt; 2022-02-26 14:29:40 INFO::Fitting model to feature number 53, Ruminococcus gnavus</span></a>
<a class="sourceLine" id="cb7-111" data-line-number="111"><span class="co">#&gt; 2022-02-26 14:29:40 INFO::Fitting model to feature number 54, Ruminococcus obeum</span></a>
<a class="sourceLine" id="cb7-112" data-line-number="112"><span class="co">#&gt; 2022-02-26 14:29:40 INFO::Fitting model to feature number 55, Ruminococcus torques</span></a>
<a class="sourceLine" id="cb7-113" data-line-number="113"><span class="co">#&gt; 2022-02-26 14:29:41 INFO::Fitting model to feature number 56, Coprococcus comes</span></a>
<a class="sourceLine" id="cb7-114" data-line-number="114"><span class="co">#&gt; 2022-02-26 14:29:41 INFO::Fitting model to feature number 57, Dorea longicatena</span></a>
<a class="sourceLine" id="cb7-115" data-line-number="115"><span class="co">#&gt; 2022-02-26 14:29:41 INFO::Fitting model to feature number 58, Lachnospiraceae bacterium 1 1 57FAA</span></a>
<a class="sourceLine" id="cb7-116" data-line-number="116"><span class="co">#&gt; 2022-02-26 14:29:41 INFO::Fitting model to feature number 59, Lachnospiraceae bacterium 3 1 46FAA</span></a>
<a class="sourceLine" id="cb7-117" data-line-number="117"><span class="co">#&gt; 2022-02-26 14:29:42 INFO::Fitting model to feature number 60, Roseburia hominis</span></a>
<a class="sourceLine" id="cb7-118" data-line-number="118"><span class="co">#&gt; 2022-02-26 14:29:42 INFO::Fitting model to feature number 61, Roseburia intestinalis</span></a>
<a class="sourceLine" id="cb7-119" data-line-number="119"><span class="co">#&gt; 2022-02-26 14:29:42 INFO::Fitting model to feature number 62, Roseburia inulinivorans</span></a>
<a class="sourceLine" id="cb7-120" data-line-number="120"><span class="co">#&gt; 2022-02-26 14:29:42 INFO::Fitting model to feature number 63, Roseburia unclassified</span></a>
<a class="sourceLine" id="cb7-121" data-line-number="121"><span class="co">#&gt; 2022-02-26 14:29:43 INFO::Fitting model to feature number 64, Oscillibacter unclassified</span></a>
<a class="sourceLine" id="cb7-122" data-line-number="122"><span class="co">#&gt; 2022-02-26 14:29:43 INFO::Fitting model to feature number 65, Peptostreptococcaceae noname unclassified</span></a>
<a class="sourceLine" id="cb7-123" data-line-number="123"><span class="co">#&gt; 2022-02-26 14:29:43 INFO::Fitting model to feature number 66, Faecalibacterium prausnitzii</span></a>
<a class="sourceLine" id="cb7-124" data-line-number="124"><span class="co">#&gt; 2022-02-26 14:29:43 INFO::Fitting model to feature number 67, Ruminococcus bromii</span></a>
<a class="sourceLine" id="cb7-125" data-line-number="125"><span class="co">#&gt; 2022-02-26 14:29:43 INFO::Fitting model to feature number 68, Ruminococcus callidus</span></a>
<a class="sourceLine" id="cb7-126" data-line-number="126"><span class="co">#&gt; 2022-02-26 14:29:44 INFO::Fitting model to feature number 69, Ruminococcus lactaris</span></a>
<a class="sourceLine" id="cb7-127" data-line-number="127"><span class="co">#&gt; 2022-02-26 14:29:44 INFO::Fitting model to feature number 70, Subdoligranulum unclassified</span></a>
<a class="sourceLine" id="cb7-128" data-line-number="128"><span class="co">#&gt; 2022-02-26 14:29:44 INFO::Fitting model to feature number 71, Coprobacillus unclassified</span></a>
<a class="sourceLine" id="cb7-129" data-line-number="129"><span class="co">#&gt; 2022-02-26 14:29:44 INFO::Fitting model to feature number 72, Acidaminococcus unclassified</span></a>
<a class="sourceLine" id="cb7-130" data-line-number="130"><span class="co">#&gt; 2022-02-26 14:29:44 INFO::Fitting model to feature number 73, Dialister invisus</span></a>
<a class="sourceLine" id="cb7-131" data-line-number="131"><span class="co">#&gt; 2022-02-26 14:29:45 INFO::Fitting model to feature number 74, Veillonella atypica</span></a>
<a class="sourceLine" id="cb7-132" data-line-number="132"><span class="co">#&gt; 2022-02-26 14:29:45 INFO::Fitting model to feature number 75, Veillonella dispar</span></a>
<a class="sourceLine" id="cb7-133" data-line-number="133"><span class="co">#&gt; 2022-02-26 14:29:45 INFO::Fitting model to feature number 76, Veillonella parvula</span></a>
<a class="sourceLine" id="cb7-134" data-line-number="134"><span class="co">#&gt; 2022-02-26 14:29:45 INFO::Fitting model to feature number 77, Veillonella unclassified</span></a>
<a class="sourceLine" id="cb7-135" data-line-number="135"><span class="co">#&gt; 2022-02-26 14:29:45 INFO::Fitting model to feature number 78, Burkholderiales bacterium 1 1 47</span></a>
<a class="sourceLine" id="cb7-136" data-line-number="136"><span class="co">#&gt; 2022-02-26 14:29:46 INFO::Fitting model to feature number 79, Parasutterella excrementihominis</span></a>
<a class="sourceLine" id="cb7-137" data-line-number="137"><span class="co">#&gt; 2022-02-26 14:29:46 INFO::Fitting model to feature number 80, Sutterella wadsworthensis</span></a>
<a class="sourceLine" id="cb7-138" data-line-number="138"><span class="co">#&gt; 2022-02-26 14:29:46 INFO::Fitting model to feature number 81, Bilophila unclassified</span></a>
<a class="sourceLine" id="cb7-139" data-line-number="139"><span class="co">#&gt; 2022-02-26 14:29:46 INFO::Fitting model to feature number 82, Escherichia coli</span></a>
<a class="sourceLine" id="cb7-140" data-line-number="140"><span class="co">#&gt; 2022-02-26 14:29:46 INFO::Fitting model to feature number 83, Escherichia unclassified</span></a>
<a class="sourceLine" id="cb7-141" data-line-number="141"><span class="co">#&gt; 2022-02-26 14:29:47 INFO::Fitting model to feature number 84, Klebsiella pneumoniae</span></a>
<a class="sourceLine" id="cb7-142" data-line-number="142"><span class="co">#&gt; 2022-02-26 14:29:47 INFO::Fitting model to feature number 85, Haemophilus parainfluenzae</span></a>
<a class="sourceLine" id="cb7-143" data-line-number="143"><span class="co">#&gt; 2022-02-26 14:29:47 INFO::Fitting model to feature number 86, Akkermansia muciniphila</span></a>
<a class="sourceLine" id="cb7-144" data-line-number="144"><span class="co">#&gt; 2022-02-26 14:29:47 INFO::Fitting model to feature number 87, C2likevirus unclassified</span></a>
<a class="sourceLine" id="cb7-145" data-line-number="145"><span class="co">#&gt; 2022-02-26 14:29:47 INFO::Counting prevalence for each feature</span></a>
<a class="sourceLine" id="cb7-146" data-line-number="146"><span class="co">#&gt; 2022-02-26 14:29:47 INFO::Writing all results to file (ordered by increasing q-values): ./demo_output/HMP2_base/all_results.tsv</span></a>
<a class="sourceLine" id="cb7-147" data-line-number="147"><span class="co">#&gt; 2022-02-26 14:29:47 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.250000 ) to file (ordered by increasing q-values): ./demo_output/HMP2_base/significant_results.tsv</span></a>
<a class="sourceLine" id="cb7-148" data-line-number="148"></a>
<a class="sourceLine" id="cb7-149" data-line-number="149"><span class="co"># Table with DA analysis results - significant features at 25% FDR with respect to diagnosis</span></a>
<a class="sourceLine" id="cb7-150" data-line-number="150"><span class="kw">nrow</span>(HMP2_base[HMP2_base<span class="op">$</span>metadata <span class="op">==</span><span class="st"> 'diagnosis'</span>,])</a>
<a class="sourceLine" id="cb7-151" data-line-number="151"><span class="co">#&gt; [1] 37</span></a></code></pre></div>
<p>The results, as stored in the data frame <code>HMP2_base</code>, indicate that there are 37 statistically significant differentially abundant features at 25% FDR including <em>Subdoligranulum unclassified</em> and <em>Bacteroides vulgatus</em>, both detected as significant in the original publication. Notably, the original publication failed to find significant taxonomic associations at the baseline level, which can be attributed to (i) low sample size, (ii) small effect size, and (iii) lack of detection power using linear models.</p>
<p>It is very satisfactory to observe that even at the baseline level (i.e.Â using a smaller sample size), Tweedieverse is able to detect several dysbiosis-associated species identified at the longitudinal level (i.e.Â using a larger sample size) in the original study.</p>
</div>
<div id="example-3---multivariable-association-on-hmp2-longitudinal-microbiomes" class="section level3">
<h3>Example 3 - Multivariable Association on HMP2 Longitudinal Microbiomes</h3>
<p>As a final illustration, we will analyze the same iHMP dataset but without restricting ourselves to baseline samples. Following the original publication, we will include subjects and recruitment sites as random effects to account for the correlations in the repeated measures and model each feature as a function of diagnosis (a categorical variable with non-IBD as the reference group) and dysbiosis state as a nested binary variable (with non-dysbiotic as reference) within each IBD phenotype (UC, CD, and non-IBD), while adjusting for consent age as a continuous covariate, and antibiotics as as binary covariate.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"></a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="co">#############</span></a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="co"># Load data #</span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="co">#############</span></a>
<a class="sourceLine" id="cb8-5" data-line-number="5"></a>
<a class="sourceLine" id="cb8-6" data-line-number="6"><span class="kw">library</span>(data.table)</a>
<a class="sourceLine" id="cb8-7" data-line-number="7">input_features &lt;-<span class="st"> </span><span class="kw">fread</span>(<span class="st">&quot;https://raw.githubusercontent.com/biobakery/Maaslin2/master/inst/extdata/HMP2_taxonomy.tsv&quot;</span>, <span class="dt">sep =</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</a>
<a class="sourceLine" id="cb8-8" data-line-number="8">input_metadata &lt;-<span class="kw">fread</span>(<span class="st">&quot;https://raw.githubusercontent.com/biobakery/Maaslin2/master/inst/extdata/HMP2_metadata.tsv&quot;</span>, <span class="dt">sep =</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</a>
<a class="sourceLine" id="cb8-9" data-line-number="9"></a>
<a class="sourceLine" id="cb8-10" data-line-number="10"><span class="co">###############</span></a>
<a class="sourceLine" id="cb8-11" data-line-number="11"><span class="co"># Format data #</span></a>
<a class="sourceLine" id="cb8-12" data-line-number="12"><span class="co">###############</span></a>
<a class="sourceLine" id="cb8-13" data-line-number="13"></a>
<a class="sourceLine" id="cb8-14" data-line-number="14"><span class="kw">library</span>(tibble)</a>
<a class="sourceLine" id="cb8-15" data-line-number="15">input_features&lt;-<span class="st"> </span><span class="kw">column_to_rownames</span>(input_features, <span class="st">'ID'</span>)</a>
<a class="sourceLine" id="cb8-16" data-line-number="16">input_metadata&lt;-<span class="st"> </span><span class="kw">column_to_rownames</span>(input_metadata, <span class="st">'ID'</span>)</a>
<a class="sourceLine" id="cb8-17" data-line-number="17"></a>
<a class="sourceLine" id="cb8-18" data-line-number="18"><span class="co"># Remove missing data and set reference</span></a>
<a class="sourceLine" id="cb8-19" data-line-number="19">metadata_final&lt;-<span class="st"> </span>input_metadata[<span class="kw">complete.cases</span>(input_metadata), ]</a>
<a class="sourceLine" id="cb8-20" data-line-number="20">metadata_final<span class="op">$</span>diagnosis&lt;-<span class="kw">as.factor</span>(metadata_final<span class="op">$</span>diagnosis)</a>
<a class="sourceLine" id="cb8-21" data-line-number="21">metadata_final<span class="op">$</span>diagnosis&lt;-<span class="kw">relevel</span>(metadata_final<span class="op">$</span>diagnosis, <span class="dt">ref =</span> <span class="st">&quot;nonIBD&quot;</span>) <span class="co">#NonIBD as reference</span></a>
<a class="sourceLine" id="cb8-22" data-line-number="22"></a>
<a class="sourceLine" id="cb8-23" data-line-number="23"><span class="co"># Adjust features accordingly </span></a>
<a class="sourceLine" id="cb8-24" data-line-number="24">features_final&lt;-input_features[<span class="kw">rownames</span>(metadata_final),]</a>
<a class="sourceLine" id="cb8-25" data-line-number="25"></a>
<a class="sourceLine" id="cb8-26" data-line-number="26"><span class="co">#################################</span></a>
<a class="sourceLine" id="cb8-27" data-line-number="27"><span class="co"># Fit a CPLM mixed effect model #</span></a>
<a class="sourceLine" id="cb8-28" data-line-number="28"><span class="co">#################################</span></a>
<a class="sourceLine" id="cb8-29" data-line-number="29"></a>
<a class="sourceLine" id="cb8-30" data-line-number="30"><span class="cf">if</span> (<span class="op">!</span><span class="kw">dir.exists</span>(<span class="st">'demo_output'</span>)) <span class="kw">dir.create</span>(<span class="st">'demo_output'</span>)</a>
<a class="sourceLine" id="cb8-31" data-line-number="31">HMP2_long &lt;-<span class="st"> </span><span class="kw">Tweedieverse</span>(features_final, </a>
<a class="sourceLine" id="cb8-32" data-line-number="32">                          metadata_final,</a>
<a class="sourceLine" id="cb8-33" data-line-number="33">                          <span class="dt">output =</span> <span class="st">'./demo_output/HMP2_long'</span>, <span class="co"># Assuming demo_output exists</span></a>
<a class="sourceLine" id="cb8-34" data-line-number="34">                          <span class="dt">fixed_effects =</span> <span class="kw">c</span>(<span class="st">'diagnosis'</span>, <span class="st">'dysbiosisnonIBD'</span>,<span class="st">'dysbiosisUC'</span>,<span class="st">'dysbiosisCD'</span>, <span class="st">'antibiotics'</span>, <span class="st">'age'</span>),</a>
<a class="sourceLine" id="cb8-35" data-line-number="35">                          <span class="dt">random_effects =</span> <span class="kw">c</span>(<span class="st">'subject'</span>, <span class="st">'site'</span>),</a>
<a class="sourceLine" id="cb8-36" data-line-number="36">                          <span class="dt">base_model =</span> <span class="st">'CPLM'</span>,</a>
<a class="sourceLine" id="cb8-37" data-line-number="37">                          <span class="dt">adjust_offset =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb8-38" data-line-number="38">                          <span class="dt">reference =</span> <span class="kw">c</span>(<span class="st">'diagnosis,nonIBD'</span>)) </a>
<a class="sourceLine" id="cb8-39" data-line-number="39"></a></code></pre></div>
<p>The results from this analysis (<code>HMP2_long</code>) indicate that the primary conclusion of the original study is largely recapitulated which can be broadly summarized as <strong>increase in facultative anaerobes including <em>E. coli</em> at the expense of anaerobic SCFA producers including <em>F. prausnitzi</em></strong>.</p>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">sessionInfo</span>()</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="co">#&gt; R version 3.6.3 (2020-02-29)</span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="co">#&gt; Platform: x86_64-apple-darwin15.6.0 (64-bit)</span></a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="co">#&gt; Running under: macOS  10.16</span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="co">#&gt; Matrix products: default</span></a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib</span></a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib</span></a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-10" data-line-number="10"><span class="co">#&gt; locale:</span></a>
<a class="sourceLine" id="cb9-11" data-line-number="11"><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></a>
<a class="sourceLine" id="cb9-12" data-line-number="12"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-13" data-line-number="13"><span class="co">#&gt; attached base packages:</span></a>
<a class="sourceLine" id="cb9-14" data-line-number="14"><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></a>
<a class="sourceLine" id="cb9-15" data-line-number="15"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-16" data-line-number="16"><span class="co">#&gt; other attached packages:</span></a>
<a class="sourceLine" id="cb9-17" data-line-number="17"><span class="co">#&gt;  [1] data.table_1.13.6  stringi_1.7.3      sparseDOSSA_0.99.6 devtools_2.4.3    </span></a>
<a class="sourceLine" id="cb9-18" data-line-number="18"><span class="co">#&gt;  [5] usethis_2.0.1      forcats_0.5.1      stringr_1.4.0      dplyr_1.0.7       </span></a>
<a class="sourceLine" id="cb9-19" data-line-number="19"><span class="co">#&gt;  [9] purrr_0.3.4        readr_1.4.0        tidyr_1.1.3        tibble_3.1.3      </span></a>
<a class="sourceLine" id="cb9-20" data-line-number="20"><span class="co">#&gt; [13] ggplot2_3.3.5      tidyverse_1.3.1    Tweedieverse_0.0.1</span></a>
<a class="sourceLine" id="cb9-21" data-line-number="21"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-22" data-line-number="22"><span class="co">#&gt; loaded via a namespace (and not attached):</span></a>
<a class="sourceLine" id="cb9-23" data-line-number="23"><span class="co">#&gt;  [1] minqa_1.2.4        colorspace_2.0-2   ellipsis_0.3.2     rprojroot_2.0.2   </span></a>
<a class="sourceLine" id="cb9-24" data-line-number="24"><span class="co">#&gt;  [5] fs_1.5.0           rstudioapi_0.13    farver_2.1.0       MatrixModels_0.4-1</span></a>
<a class="sourceLine" id="cb9-25" data-line-number="25"><span class="co">#&gt;  [9] remotes_2.4.0      getopt_1.20.3      fansi_0.5.0        mvtnorm_1.1-1     </span></a>
<a class="sourceLine" id="cb9-26" data-line-number="26"><span class="co">#&gt; [13] lubridate_1.7.10   xml2_1.3.2         splines_3.6.3      logging_0.10-108  </span></a>
<a class="sourceLine" id="cb9-27" data-line-number="27"><span class="co">#&gt; [17] cachem_1.0.3       knitr_1.31         pkgload_1.2.1      jsonlite_1.7.2    </span></a>
<a class="sourceLine" id="cb9-28" data-line-number="28"><span class="co">#&gt; [21] mcmc_0.9-7         broom_0.7.7        dbplyr_2.1.1       compiler_3.6.3    </span></a>
<a class="sourceLine" id="cb9-29" data-line-number="29"><span class="co">#&gt; [25] httr_1.4.2         backports_1.2.1    assertthat_0.2.1   Matrix_1.2-18     </span></a>
<a class="sourceLine" id="cb9-30" data-line-number="30"><span class="co">#&gt; [29] fastmap_1.1.0      gmm_1.6-5          cli_3.0.1          htmltools_0.5.1.1 </span></a>
<a class="sourceLine" id="cb9-31" data-line-number="31"><span class="co">#&gt; [33] quantreg_5.83      prettyunits_1.1.1  tools_3.6.3        coda_0.19-4       </span></a>
<a class="sourceLine" id="cb9-32" data-line-number="32"><span class="co">#&gt; [37] gtable_0.3.0       glue_1.4.2         reshape2_1.4.4     Rcpp_1.0.7        </span></a>
<a class="sourceLine" id="cb9-33" data-line-number="33"><span class="co">#&gt; [41] biglm_0.9-2.1      cellranger_1.1.0   vctrs_0.3.8        nlme_3.1-144      </span></a>
<a class="sourceLine" id="cb9-34" data-line-number="34"><span class="co">#&gt; [45] conquer_1.0.2      optparse_1.6.6     tmvtnorm_1.4-10    xfun_0.20         </span></a>
<a class="sourceLine" id="cb9-35" data-line-number="35"><span class="co">#&gt; [49] ps_1.5.0           testthat_3.0.3     rvest_1.0.0        lifecycle_1.0.0   </span></a>
<a class="sourceLine" id="cb9-36" data-line-number="36"><span class="co">#&gt; [53] statmod_1.4.33     MASS_7.3-51.5      zoo_1.8-8          scales_1.1.1      </span></a>
<a class="sourceLine" id="cb9-37" data-line-number="37"><span class="co">#&gt; [57] hms_1.0.0          parallel_3.6.3     sandwich_3.0-0     SparseM_1.78      </span></a>
<a class="sourceLine" id="cb9-38" data-line-number="38"><span class="co">#&gt; [61] yaml_2.2.1         curl_4.3           memoise_2.0.0      pbapply_1.4-3     </span></a>
<a class="sourceLine" id="cb9-39" data-line-number="39"><span class="co">#&gt; [65] highr_0.8          desc_1.3.0         pkgbuild_1.2.0     truncnorm_1.0-8   </span></a>
<a class="sourceLine" id="cb9-40" data-line-number="40"><span class="co">#&gt; [69] rlang_0.4.11       pkgconfig_2.0.3    matrixStats_0.58.0 cplm_0.7-8        </span></a>
<a class="sourceLine" id="cb9-41" data-line-number="41"><span class="co">#&gt; [73] evaluate_0.14      lattice_0.20-38    processx_3.5.2     tidyselect_1.1.0  </span></a>
<a class="sourceLine" id="cb9-42" data-line-number="42"><span class="co">#&gt; [77] plyr_1.8.6         magrittr_2.0.1     R6_2.5.1           generics_0.1.0    </span></a>
<a class="sourceLine" id="cb9-43" data-line-number="43"><span class="co">#&gt; [81] DBI_1.1.1          pillar_1.6.2       haven_2.3.1        withr_2.4.2       </span></a>
<a class="sourceLine" id="cb9-44" data-line-number="44"><span class="co">#&gt; [85] modelr_0.1.8       crayon_1.4.1       utf8_1.2.2         rmarkdown_2.6     </span></a>
<a class="sourceLine" id="cb9-45" data-line-number="45"><span class="co">#&gt; [89] grid_3.6.3         readxl_1.3.1       callr_3.7.0        reprex_2.0.0      </span></a>
<a class="sourceLine" id="cb9-46" data-line-number="46"><span class="co">#&gt; [93] digest_0.6.27      MCMCpack_1.5-0     stats4_3.6.3       munsell_0.5.0     </span></a>
<a class="sourceLine" id="cb9-47" data-line-number="47"><span class="co">#&gt; [97] tweedie_2.3.3      sessioninfo_1.1.1</span></a></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
