<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Himel Mallick" />

<meta name="date" content="2022-02-26" />

<title>Using Tweedieverse for differential expression of scRNA-Seq non-UMI counts</title>


<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Using Tweedieverse for differential expression of scRNA-Seq non-UMI counts</h1>
<h4 class="author">Himel Mallick</h4>
<h4 class="date">2022-02-26</h4>



<p>This vignette highlights some example workflows for performing differential expression using <code>Tweedieverse</code> on single-cell RNA-seq (scRNA-seq) non-UMI counts (check out the related tutorial for <a href="http://htmlpreview.github.io/?https://github.com/himelmallick/TweedieLabs/blob/main/R/Tweedieverse_scRNASeq_UMI.html">scRNA-seq UMI counts</a>). It should be noted that <code>Tweedieverse</code> is a sister package of <a href="https://huttenhower.sph.harvard.edu/maaslin/">MaAsLin</a> which is developed by many of the same authors as this package and likewise, they share certain similarities but also important differences as described in the in-depth <a href="https://github.com/himelmallick/TweedieLabs">tutorials</a>.</p>
<div id="install-and-load-packages" class="section level2">
<h2>Install and load packages</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co"># Load the Tweedieverse package</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">library</span>(Tweedieverse)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="co"># Load other essential packages</span></a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">library</span>(tidyverse)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw">library</span>(SC2P) <span class="co"># library(devtools); devtools::install_github('haowulab/SC2P')</span></a></code></pre></div>
</div>
<div id="input" class="section level2">
<h2>Input</h2>
<p><code>Tweedieverse</code> requires two tab-delimited input files, one for expression counts (<code>input_features</code>) and one for metadata or covariates (<code>input_metadata</code>). The rows of the <code>input_features</code> correspond to genes and the columns correspond to cells (and vice versa). Similarly, the columns of the <code>input_metadata</code> correspond to cell-specific covariates (e.g., cell types) and the rows correspond to cells.</p>
<p>The <code>input_features</code> file can contain cells not included in the <code>input_metadata</code> file (along with the reverse case) although it is expected that they have matching cells. For both cases, cells not included in either of the files will be removed from the analysis. Also cells do not need to be in the same order in the two files as <code>Tweedieverse</code> will automatically detect the right order based on the cell names (assuming common cell IDs across files).</p>
<p>For the purpose of this vignette, it is assumed that these two input data have already been quality-controlled with necessary preprocessing steps, e.g.,, as described in the <a href="https://satijalab.org/seurat/articles/essential_commands.html">Seurat tutorial</a>. For demonstration purposes, we will be using a cleaned version of the Brain dataset (available from the R package <a href="https://github.com/haowulab/SC2P">SC2P</a>). The original dataset is available from the Gene Expression Omnibus database under accession number <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE67835">GSE67835</a>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># Load data</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw">data</span>(brain_scRNAseq)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="kw">colnames</span>(Y) &lt;-<span class="st"> </span><span class="kw">rownames</span>(design)</a>
<a class="sourceLine" id="cb2-4" data-line-number="4"></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="co"># Prepare data for Tweedieverse</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6">input_features&lt;-<span class="kw">as.data.frame</span>(<span class="kw">t</span>(Y))</a>
<a class="sourceLine" id="cb2-7" data-line-number="7">input_metadata&lt;-design</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">cellIDs&lt;-<span class="kw">paste</span>(<span class="st">'Cell'</span>, <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(input_features), <span class="dt">sep =</span><span class="st">''</span>)</a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="kw">rownames</span>(input_metadata)&lt;-<span class="kw">rownames</span>(input_features)&lt;-cellIDs <span class="co"># Arbitrary Names</span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10"></a>
<a class="sourceLine" id="cb2-11" data-line-number="11"><span class="co"># Take a quick look at the data</span></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="kw">head</span>(input_features[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>])</a>
<a class="sourceLine" id="cb2-13" data-line-number="13"><span class="co">#&gt;       A2M  A2ML1  A2MP1  AAAS  AACS </span></a>
<a class="sourceLine" id="cb2-14" data-line-number="14"><span class="co">#&gt; Cell1    0      0      0     0     0</span></a>
<a class="sourceLine" id="cb2-15" data-line-number="15"><span class="co">#&gt; Cell2    0      0      0     0     0</span></a>
<a class="sourceLine" id="cb2-16" data-line-number="16"><span class="co">#&gt; Cell3   34     25      0   111     0</span></a>
<a class="sourceLine" id="cb2-17" data-line-number="17"><span class="co">#&gt; Cell4    0      0      0     0     0</span></a>
<a class="sourceLine" id="cb2-18" data-line-number="18"><span class="co">#&gt; Cell5    0   1290      0     0     0</span></a>
<a class="sourceLine" id="cb2-19" data-line-number="19"><span class="kw">head</span>(input_metadata[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>])</a>
<a class="sourceLine" id="cb2-20" data-line-number="20"><span class="co">#&gt;               tissue   celltype                     age                   chip</span></a>
<a class="sourceLine" id="cb2-21" data-line-number="21"><span class="co">#&gt; Cell1 tissue: cortex astrocytes age: postnatal 54 years c1 chip id: 1772078217</span></a>
<a class="sourceLine" id="cb2-22" data-line-number="22"><span class="co">#&gt; Cell2 tissue: cortex astrocytes age: postnatal 37 years c1 chip id: 1772078236</span></a>
<a class="sourceLine" id="cb2-23" data-line-number="23"><span class="co">#&gt; Cell3 tissue: cortex astrocytes age: postnatal 37 years c1 chip id: 1772078236</span></a>
<a class="sourceLine" id="cb2-24" data-line-number="24"><span class="co">#&gt; Cell4 tissue: cortex astrocytes age: postnatal 37 years c1 chip id: 1772078237</span></a>
<a class="sourceLine" id="cb2-25" data-line-number="25"><span class="co">#&gt; Cell5 tissue: cortex astrocytes age: postnatal 37 years c1 chip id: 1772078237</span></a>
<a class="sourceLine" id="cb2-26" data-line-number="26"><span class="co">#&gt;       experiment</span></a>
<a class="sourceLine" id="cb2-27" data-line-number="27"><span class="co">#&gt; Cell1      AB_S8</span></a>
<a class="sourceLine" id="cb2-28" data-line-number="28"><span class="co">#&gt; Cell2     AB_S11</span></a>
<a class="sourceLine" id="cb2-29" data-line-number="29"><span class="co">#&gt; Cell3     AB_S11</span></a>
<a class="sourceLine" id="cb2-30" data-line-number="30"><span class="co">#&gt; Cell4     AB_S11</span></a>
<a class="sourceLine" id="cb2-31" data-line-number="31"><span class="co">#&gt; Cell5     AB_S11</span></a>
<a class="sourceLine" id="cb2-32" data-line-number="32"></a>
<a class="sourceLine" id="cb2-33" data-line-number="33"><span class="co"># Calculate sparsity (percentage of zeroes)</span></a>
<a class="sourceLine" id="cb2-34" data-line-number="34"><span class="kw">round</span>(<span class="kw">sum</span>(input_features<span class="op">==</span><span class="dv">0</span>)<span class="op">/</span>(<span class="kw">nrow</span>(input_features)<span class="op">*</span><span class="kw">ncol</span>(input_features))<span class="op">*</span><span class="dv">100</span>, <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb2-35" data-line-number="35"><span class="co">#&gt; [1] 64.9</span></a></code></pre></div>
<p>For brevity, we will consider top 100 most variable genes.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">sds&lt;-<span class="kw">apply</span>(input_features, <span class="dv">2</span>, <span class="cf">function</span>(x){<span class="kw">sd</span>(x, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)})</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">top_n_IDs&lt;-<span class="kw">order</span>(sds, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)[<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>]</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">input_features&lt;-input_features[, top_n_IDs]</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="kw">dim</span>(input_features)</a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co">#&gt; [1] 100 100</span></a></code></pre></div>
</div>
<div id="normalization" class="section level2">
<h2>Normalization</h2>
<p>For normalization, Tweedieverse expects a variable named <code>scale_factor</code> in the <code>input_metadata</code> file which is included as an offset in the base model when prompted (i.e. <code>adjust_offset = TRUE</code>). If not found in metadata, Tweedieverse, by default, includes the library size as an offset in the model (unless <code>adjust_offset = FALSE</code>). Here we use the ZICP base model with <a href="https://bioconductor.org/packages/release/bioc/html/scran.html"><strong><code>scran</code></strong></a> normalization (see the related tutorial for <a href="http://htmlpreview.github.io/?https://github.com/himelmallick/TweedieLabs/blob/main/R/Tweedieverse_scRNASeq_UMI.html">UMI counts</a> for using library size normalization).</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="co"># Load library</span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="kw">library</span>(scran) </a>
<a class="sourceLine" id="cb4-4" data-line-number="4"></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co"># SCRAN normalization</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6">sce&lt;-<span class="kw">t</span>(input_features)</a>
<a class="sourceLine" id="cb4-7" data-line-number="7">scale_factor &lt;-<span class="st"> </span>scran<span class="op">::</span><span class="kw">calculateSumFactors</span>(sce)</a>
<a class="sourceLine" id="cb4-8" data-line-number="8">input_metadata<span class="op">$</span>scale_factor&lt;-scale_factor</a></code></pre></div>
</div>
<div id="differential-expression" class="section level2">
<h2>Differential Expression</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="cf">if</span> (<span class="op">!</span><span class="kw">dir.exists</span>(<span class="st">'demo_output'</span>)) <span class="kw">dir.create</span>(<span class="st">'demo_output'</span>)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">Brain &lt;-<span class="st"> </span><span class="kw">Tweedieverse</span>(</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">  input_features,</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">  input_metadata, </a>
<a class="sourceLine" id="cb5-5" data-line-number="5">  <span class="dt">output =</span> <span class="st">'demo_output/Brain'</span>, <span class="co"># Assuming demo_output exists</span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6">  <span class="dt">fixed_effects =</span> <span class="kw">c</span>(<span class="st">'celltype'</span>),</a>
<a class="sourceLine" id="cb5-7" data-line-number="7">  <span class="dt">base_model =</span> <span class="st">'ZICP'</span>,</a>
<a class="sourceLine" id="cb5-8" data-line-number="8">  <span class="dt">standardize =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb5-9" data-line-number="9">  <span class="dt">adjust_offset =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="co">#&gt; [1] &quot;Creating output folder&quot;</span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="co">#&gt; 2022-02-26 14:31:07 INFO::Writing function arguments to log file</span></a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="co">#&gt; 2022-02-26 14:31:07 INFO::Determining format of input files</span></a>
<a class="sourceLine" id="cb5-13" data-line-number="13"><span class="co">#&gt; 2022-02-26 14:31:07 INFO::Input format is data samples as rows and metadata samples as rows</span></a>
<a class="sourceLine" id="cb5-14" data-line-number="14"><span class="co">#&gt; 2022-02-26 14:31:07 INFO::Factor detected for categorial metadata 'celltype'. Provide a reference argument or manually set factor ordering to change reference level.</span></a>
<a class="sourceLine" id="cb5-15" data-line-number="15"><span class="co">#&gt; 2022-02-26 14:31:07 INFO::Filter data based on min abundance and min prevalence</span></a>
<a class="sourceLine" id="cb5-16" data-line-number="16"><span class="co">#&gt; 2022-02-26 14:31:07 INFO::Total samples in data: 100</span></a>
<a class="sourceLine" id="cb5-17" data-line-number="17"><span class="co">#&gt; 2022-02-26 14:31:07 INFO::Min samples required with min abundance for a feature not to be filtered: 10.000000</span></a>
<a class="sourceLine" id="cb5-18" data-line-number="18"><span class="co">#&gt; 2022-02-26 14:31:07 INFO::Total filtered features with prevalence-abundance filtering: 0</span></a>
<a class="sourceLine" id="cb5-19" data-line-number="19"><span class="co">#&gt; 2022-02-26 14:31:07 INFO::Filtered feature names:</span></a>
<a class="sourceLine" id="cb5-20" data-line-number="20"><span class="co">#&gt; 2022-02-26 14:31:07 INFO::Total filtered features with variance filtering: 0</span></a>
<a class="sourceLine" id="cb5-21" data-line-number="21"><span class="co">#&gt; 2022-02-26 14:31:07 INFO::Filtered feature names:</span></a>
<a class="sourceLine" id="cb5-22" data-line-number="22"><span class="co">#&gt; 2022-02-26 14:31:07 INFO::Excluded metadata with entropy less or equal to 0: adult</span></a>
<a class="sourceLine" id="cb5-23" data-line-number="23"><span class="co">#&gt; 2022-02-26 14:31:07 INFO::Formula for fixed effects: expr ~  celltype</span></a>
<a class="sourceLine" id="cb5-24" data-line-number="24"><span class="co">#&gt; 2022-02-26 14:31:07 INFO::Bypass z-score application to metadata</span></a>
<a class="sourceLine" id="cb5-25" data-line-number="25"><span class="co">#&gt; 2022-02-26 14:31:07 INFO::Running selected analysis method: ZICP</span></a>
<a class="sourceLine" id="cb5-26" data-line-number="26"><span class="co">#&gt; 2022-02-26 14:31:07 INFO::Fitting model to feature number 1, no_feature</span></a>
<a class="sourceLine" id="cb5-27" data-line-number="27"><span class="co">#&gt; 2022-02-26 14:31:07 INFO::Fitting model to feature number 2, alignment_not_unique</span></a>
<a class="sourceLine" id="cb5-28" data-line-number="28"><span class="co">#&gt; 2022-02-26 14:31:10 INFO::Fitting model to feature number 3, PLP1</span></a>
<a class="sourceLine" id="cb5-29" data-line-number="29"><span class="co">#&gt; 2022-02-26 14:31:10 INFO::Fitting model to feature number 4, SLC1A2</span></a>
<a class="sourceLine" id="cb5-30" data-line-number="30"><span class="co">#&gt; 2022-02-26 14:31:10 INFO::Fitting model to feature number 5, MALAT1</span></a>
<a class="sourceLine" id="cb5-31" data-line-number="31"><span class="co">#&gt; 2022-02-26 14:31:11 INFO::Fitting model to feature number 6, SPARCL1</span></a>
<a class="sourceLine" id="cb5-32" data-line-number="32"><span class="co">#&gt; 2022-02-26 14:31:11 INFO::Fitting model to feature number 7, ambiguous</span></a>
<a class="sourceLine" id="cb5-33" data-line-number="33"><span class="co">#&gt; 2022-02-26 14:31:11 INFO::Fitting model to feature number 8, CLU</span></a>
<a class="sourceLine" id="cb5-34" data-line-number="34"><span class="co">#&gt; 2022-02-26 14:31:11 INFO::Fitting model to feature number 9, PIP4K2A</span></a>
<a class="sourceLine" id="cb5-35" data-line-number="35"><span class="co">#&gt; 2022-02-26 14:31:11 INFO::Fitting model to feature number 10, SAT1</span></a>
<a class="sourceLine" id="cb5-36" data-line-number="36"><span class="co">#&gt; 2022-02-26 14:31:12 INFO::Fitting model to feature number 11, AQP4</span></a>
<a class="sourceLine" id="cb5-37" data-line-number="37"><span class="co">#&gt; 2022-02-26 14:31:12 INFO::Fitting model to feature number 12, DPYSL2</span></a>
<a class="sourceLine" id="cb5-38" data-line-number="38"><span class="co">#&gt; 2022-02-26 14:31:12 INFO::Fitting model to feature number 13, GLUL</span></a>
<a class="sourceLine" id="cb5-39" data-line-number="39"><span class="co">#&gt; 2022-02-26 14:31:12 INFO::Fitting model to feature number 14, CNP</span></a>
<a class="sourceLine" id="cb5-40" data-line-number="40"><span class="co">#&gt; 2022-02-26 14:31:12 INFO::Fitting model to feature number 15, SLC1A3</span></a>
<a class="sourceLine" id="cb5-41" data-line-number="41"><span class="co">#&gt; 2022-02-26 14:31:13 INFO::Fitting model to feature number 16, MBP</span></a>
<a class="sourceLine" id="cb5-42" data-line-number="42"><span class="co">#&gt; 2022-02-26 14:31:13 INFO::Fitting model to feature number 17, CLDND1</span></a>
<a class="sourceLine" id="cb5-43" data-line-number="43"><span class="co">#&gt; 2022-02-26 14:31:13 INFO::Fitting model to feature number 18, HSPA5</span></a>
<a class="sourceLine" id="cb5-44" data-line-number="44"><span class="co">#&gt; 2022-02-26 14:31:14 INFO::Fitting model to feature number 19, FOS</span></a>
<a class="sourceLine" id="cb5-45" data-line-number="45"><span class="co">#&gt; 2022-02-26 14:31:14 INFO::Fitting model to feature number 20, GPM6A</span></a>
<a class="sourceLine" id="cb5-46" data-line-number="46"><span class="co">#&gt; 2022-02-26 14:31:14 INFO::Fitting model to feature number 21, TMEM144</span></a>
<a class="sourceLine" id="cb5-47" data-line-number="47"><span class="co">#&gt; 2022-02-26 14:31:14 INFO::Fitting model to feature number 22, DST</span></a>
<a class="sourceLine" id="cb5-48" data-line-number="48"><span class="co">#&gt; 2022-02-26 14:31:14 INFO::Fitting model to feature number 23, ALDOC</span></a>
<a class="sourceLine" id="cb5-49" data-line-number="49"><span class="co">#&gt; 2022-02-26 14:31:15 INFO::Fitting model to feature number 24, ARRDC3</span></a>
<a class="sourceLine" id="cb5-50" data-line-number="50"><span class="co">#&gt; 2022-02-26 14:31:15 INFO::Fitting model to feature number 25, PCDH9</span></a>
<a class="sourceLine" id="cb5-51" data-line-number="51"><span class="co">#&gt; 2022-02-26 14:31:15 INFO::Fitting model to feature number 26, DDAH1</span></a>
<a class="sourceLine" id="cb5-52" data-line-number="52"><span class="co">#&gt; 2022-02-26 14:31:15 INFO::Fitting model to feature number 27, ATP1A2</span></a>
<a class="sourceLine" id="cb5-53" data-line-number="53"><span class="co">#&gt; 2022-02-26 14:31:15 INFO::Fitting model to feature number 28, PON2</span></a>
<a class="sourceLine" id="cb5-54" data-line-number="54"><span class="co">#&gt; 2022-02-26 14:31:16 INFO::Fitting model to feature number 29, GPR98</span></a>
<a class="sourceLine" id="cb5-55" data-line-number="55"><span class="co">#&gt; 2022-02-26 14:31:16 INFO::Fitting model to feature number 30, ERMN</span></a>
<a class="sourceLine" id="cb5-56" data-line-number="56"><span class="co">#&gt; 2022-02-26 14:31:16 INFO::Fitting model to feature number 31, HSP90AA1</span></a>
<a class="sourceLine" id="cb5-57" data-line-number="57"><span class="co">#&gt; 2022-02-26 14:31:16 INFO::Fitting model to feature number 32, CPE</span></a>
<a class="sourceLine" id="cb5-58" data-line-number="58"><span class="co">#&gt; 2022-02-26 14:31:16 INFO::Fitting model to feature number 33, MAP1B</span></a>
<a class="sourceLine" id="cb5-59" data-line-number="59"><span class="co">#&gt; 2022-02-26 14:31:16 INFO::Fitting model to feature number 34, TF</span></a>
<a class="sourceLine" id="cb5-60" data-line-number="60"><span class="co">#&gt; 2022-02-26 14:31:17 INFO::Fitting model to feature number 35, CALM2</span></a>
<a class="sourceLine" id="cb5-61" data-line-number="61"><span class="co">#&gt; 2022-02-26 14:31:17 INFO::Fitting model to feature number 36, NTRK2</span></a>
<a class="sourceLine" id="cb5-62" data-line-number="62"><span class="co">#&gt; 2022-02-26 14:31:17 INFO::Fitting model to feature number 37, PPP1CB</span></a>
<a class="sourceLine" id="cb5-63" data-line-number="63"><span class="co">#&gt; 2022-02-26 14:31:17 INFO::Fitting model to feature number 38, MACF1</span></a>
<a class="sourceLine" id="cb5-64" data-line-number="64"><span class="co">#&gt; 2022-02-26 14:31:17 INFO::Fitting model to feature number 39, ATP1B2</span></a>
<a class="sourceLine" id="cb5-65" data-line-number="65"><span class="co">#&gt; 2022-02-26 14:31:17 INFO::Fitting model to feature number 40, SPP1</span></a>
<a class="sourceLine" id="cb5-66" data-line-number="66"><span class="co">#&gt; 2022-02-26 14:31:18 INFO::Fitting model to feature number 41, SRSF6</span></a>
<a class="sourceLine" id="cb5-67" data-line-number="67"><span class="co">#&gt; 2022-02-26 14:31:18 INFO::Fitting model to feature number 42, EGR1</span></a>
<a class="sourceLine" id="cb5-68" data-line-number="68"><span class="co">#&gt; 2022-02-26 14:31:18 INFO::Fitting model to feature number 43, SCD</span></a>
<a class="sourceLine" id="cb5-69" data-line-number="69"><span class="co">#&gt; 2022-02-26 14:31:18 INFO::Fitting model to feature number 44, LAMP2</span></a>
<a class="sourceLine" id="cb5-70" data-line-number="70"><span class="co">#&gt; 2022-02-26 14:31:18 INFO::Fitting model to feature number 45, GJA1</span></a>
<a class="sourceLine" id="cb5-71" data-line-number="71"><span class="co">#&gt; 2022-02-26 14:31:18 INFO::Fitting model to feature number 46, LIMCH1</span></a>
<a class="sourceLine" id="cb5-72" data-line-number="72"><span class="co">#&gt; 2022-02-26 14:31:19 INFO::Fitting model to feature number 47, AXL</span></a>
<a class="sourceLine" id="cb5-73" data-line-number="73"><span class="co">#&gt; 2022-02-26 14:31:19 INFO::Fitting model to feature number 48, AGXT2L1</span></a>
<a class="sourceLine" id="cb5-74" data-line-number="74"><span class="co">#&gt; 2022-02-26 14:31:19 INFO::Fitting model to feature number 49, BMPR1B</span></a>
<a class="sourceLine" id="cb5-75" data-line-number="75"><span class="co">#&gt; 2022-02-26 14:31:19 INFO::Fitting model to feature number 50, SLC24A2</span></a>
<a class="sourceLine" id="cb5-76" data-line-number="76"><span class="co">#&gt; 2022-02-26 14:31:19 INFO::Fitting model to feature number 51, HSPA8</span></a>
<a class="sourceLine" id="cb5-77" data-line-number="77"><span class="co">#&gt; 2022-02-26 14:31:19 INFO::Fitting model to feature number 52, APC</span></a>
<a class="sourceLine" id="cb5-78" data-line-number="78"><span class="co">#&gt; 2022-02-26 14:31:20 INFO::Fitting model to feature number 53, DDX5</span></a>
<a class="sourceLine" id="cb5-79" data-line-number="79"><span class="co">#&gt; 2022-02-26 14:31:20 INFO::Fitting model to feature number 54, GPM6B</span></a>
<a class="sourceLine" id="cb5-80" data-line-number="80"><span class="co">#&gt; 2022-02-26 14:31:20 INFO::Fitting model to feature number 55, ENO1</span></a>
<a class="sourceLine" id="cb5-81" data-line-number="81"><span class="co">#&gt; 2022-02-26 14:31:20 INFO::Fitting model to feature number 56, H3F3B</span></a>
<a class="sourceLine" id="cb5-82" data-line-number="82"><span class="co">#&gt; 2022-02-26 14:31:20 INFO::Fitting model to feature number 57, TUBA1A</span></a>
<a class="sourceLine" id="cb5-83" data-line-number="83"><span class="co">#&gt; 2022-02-26 14:31:20 INFO::Fitting model to feature number 58, AASS</span></a>
<a class="sourceLine" id="cb5-84" data-line-number="84"><span class="co">#&gt; 2022-02-26 14:31:21 INFO::Fitting model to feature number 59, F3</span></a>
<a class="sourceLine" id="cb5-85" data-line-number="85"><span class="co">#&gt; 2022-02-26 14:31:21 INFO::Fitting model to feature number 60, STMN1</span></a>
<a class="sourceLine" id="cb5-86" data-line-number="86"><span class="co">#&gt; 2022-02-26 14:31:21 INFO::Fitting model to feature number 61, C6orf62</span></a>
<a class="sourceLine" id="cb5-87" data-line-number="87"><span class="co">#&gt; 2022-02-26 14:31:22 INFO::Fitting model to feature number 62, CRYAB</span></a>
<a class="sourceLine" id="cb5-88" data-line-number="88"><span class="co">#&gt; 2022-02-26 14:31:22 INFO::Fitting model to feature number 63, SRSF3</span></a>
<a class="sourceLine" id="cb5-89" data-line-number="89"><span class="co">#&gt; 2022-02-26 14:31:22 INFO::Fitting model to feature number 64, CKB</span></a>
<a class="sourceLine" id="cb5-90" data-line-number="90"><span class="co">#&gt; 2022-02-26 14:31:22 INFO::Fitting model to feature number 65, TSPAN7</span></a>
<a class="sourceLine" id="cb5-91" data-line-number="91"><span class="co">#&gt; 2022-02-26 14:31:23 INFO::Fitting model to feature number 66, MARCKS</span></a>
<a class="sourceLine" id="cb5-92" data-line-number="92"><span class="co">#&gt; 2022-02-26 14:31:23 INFO::Fitting model to feature number 67, SLCO1C1</span></a>
<a class="sourceLine" id="cb5-93" data-line-number="93"><span class="co">#&gt; 2022-02-26 14:31:23 INFO::Fitting model to feature number 68, RFTN2</span></a>
<a class="sourceLine" id="cb5-94" data-line-number="94"><span class="co">#&gt; 2022-02-26 14:31:24 INFO::Fitting model to feature number 69, OMG</span></a>
<a class="sourceLine" id="cb5-95" data-line-number="95"><span class="co">#&gt; 2022-02-26 14:31:24 INFO::Fitting model to feature number 70, SYNE1</span></a>
<a class="sourceLine" id="cb5-96" data-line-number="96"><span class="co">#&gt; 2022-02-26 14:31:25 INFO::Fitting model to feature number 71, ENPP2</span></a>
<a class="sourceLine" id="cb5-97" data-line-number="97"><span class="co">#&gt; 2022-02-26 14:31:25 INFO::Fitting model to feature number 72, C10orf46</span></a>
<a class="sourceLine" id="cb5-98" data-line-number="98"><span class="co">#&gt; 2022-02-26 14:31:25 INFO::Fitting model to feature number 73, TBL1XR1</span></a>
<a class="sourceLine" id="cb5-99" data-line-number="99"><span class="co">#&gt; 2022-02-26 14:31:25 INFO::Fitting model to feature number 74, STON2</span></a>
<a class="sourceLine" id="cb5-100" data-line-number="100"><span class="co">#&gt; 2022-02-26 14:31:26 INFO::Fitting model to feature number 75, QKI</span></a>
<a class="sourceLine" id="cb5-101" data-line-number="101"><span class="co">#&gt; 2022-02-26 14:31:26 INFO::Fitting model to feature number 76, SLC12A2</span></a>
<a class="sourceLine" id="cb5-102" data-line-number="102"><span class="co">#&gt; 2022-02-26 14:31:26 INFO::Fitting model to feature number 77, LSAMP</span></a>
<a class="sourceLine" id="cb5-103" data-line-number="103"><span class="co">#&gt; 2022-02-26 14:31:26 INFO::Fitting model to feature number 78, NFIA</span></a>
<a class="sourceLine" id="cb5-104" data-line-number="104"><span class="co">#&gt; 2022-02-26 14:31:26 INFO::Fitting model to feature number 79, NFIB</span></a>
<a class="sourceLine" id="cb5-105" data-line-number="105"><span class="co">#&gt; 2022-02-26 14:31:26 INFO::Fitting model to feature number 80, PTPRZ1</span></a>
<a class="sourceLine" id="cb5-106" data-line-number="106"><span class="co">#&gt; 2022-02-26 14:31:27 INFO::Fitting model to feature number 81, DTNA</span></a>
<a class="sourceLine" id="cb5-107" data-line-number="107"><span class="co">#&gt; 2022-02-26 14:31:27 INFO::Fitting model to feature number 82, GAPDH</span></a>
<a class="sourceLine" id="cb5-108" data-line-number="108"><span class="co">#&gt; 2022-02-26 14:31:27 INFO::Fitting model to feature number 83, ITGA6</span></a>
<a class="sourceLine" id="cb5-109" data-line-number="109"><span class="co">#&gt; 2022-02-26 14:31:27 INFO::Fitting model to feature number 84, GLUD1</span></a>
<a class="sourceLine" id="cb5-110" data-line-number="110"><span class="co">#&gt; 2022-02-26 14:31:27 INFO::Fitting model to feature number 85, RBBP6</span></a>
<a class="sourceLine" id="cb5-111" data-line-number="111"><span class="co">#&gt; 2022-02-26 14:31:28 INFO::Fitting model to feature number 86, CLDN11</span></a>
<a class="sourceLine" id="cb5-112" data-line-number="112"><span class="co">#&gt; 2022-02-26 14:31:28 INFO::Fitting model to feature number 87, DNM3</span></a>
<a class="sourceLine" id="cb5-113" data-line-number="113"><span class="co">#&gt; 2022-02-26 14:31:28 INFO::Fitting model to feature number 88, ACTB</span></a>
<a class="sourceLine" id="cb5-114" data-line-number="114"><span class="co">#&gt; 2022-02-26 14:31:28 INFO::Fitting model to feature number 89, SEPT7</span></a>
<a class="sourceLine" id="cb5-115" data-line-number="115"><span class="co">#&gt; 2022-02-26 14:31:28 INFO::Fitting model to feature number 90, CLK1</span></a>
<a class="sourceLine" id="cb5-116" data-line-number="116"><span class="co">#&gt; 2022-02-26 14:31:28 INFO::Fitting model to feature number 91, PDK4</span></a>
<a class="sourceLine" id="cb5-117" data-line-number="117"><span class="co">#&gt; 2022-02-26 14:31:28 INFO::Fitting model to feature number 92, SLAIN1</span></a>
<a class="sourceLine" id="cb5-118" data-line-number="118"><span class="co">#&gt; 2022-02-26 14:31:29 INFO::Fitting model to feature number 93, ABCA1</span></a>
<a class="sourceLine" id="cb5-119" data-line-number="119"><span class="co">#&gt; 2022-02-26 14:31:29 INFO::Fitting model to feature number 94, RYR3</span></a>
<a class="sourceLine" id="cb5-120" data-line-number="120"><span class="co">#&gt; 2022-02-26 14:31:29 INFO::Fitting model to feature number 95, EZR</span></a>
<a class="sourceLine" id="cb5-121" data-line-number="121"><span class="co">#&gt; 2022-02-26 14:31:29 INFO::Fitting model to feature number 96, MGEA5</span></a>
<a class="sourceLine" id="cb5-122" data-line-number="122"><span class="co">#&gt; 2022-02-26 14:31:29 INFO::Fitting model to feature number 97, TMEM33</span></a>
<a class="sourceLine" id="cb5-123" data-line-number="123"><span class="co">#&gt; 2022-02-26 14:31:30 INFO::Fitting model to feature number 98, HSPH1</span></a>
<a class="sourceLine" id="cb5-124" data-line-number="124"><span class="co">#&gt; 2022-02-26 14:31:30 INFO::Fitting model to feature number 99, TIMP3</span></a>
<a class="sourceLine" id="cb5-125" data-line-number="125"><span class="co">#&gt; 2022-02-26 14:31:30 INFO::Fitting model to feature number 100, IDI1</span></a>
<a class="sourceLine" id="cb5-126" data-line-number="126"><span class="co">#&gt; 2022-02-26 14:31:30 INFO::Counting prevalence for each feature</span></a>
<a class="sourceLine" id="cb5-127" data-line-number="127"><span class="co">#&gt; 2022-02-26 14:31:30 INFO::Writing all results to file (ordered by increasing q-values): demo_output/Brain/all_results.tsv</span></a>
<a class="sourceLine" id="cb5-128" data-line-number="128"><span class="co">#&gt; 2022-02-26 14:31:30 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.050000 ) to file (ordered by increasing q-values): demo_output/Brain/significant_results.tsv</span></a></code></pre></div>
</div>
<div id="output" class="section level2">
<h2>Output</h2>
<p>After computation, <code>Brain</code> is a data frame containing coefficient estimates, p-values, and q-values (multiplicity-adjusted p-values) along with other parameter estimates from the fitted per-feature models. By default, p-values are adjusted with the Benjamini-Hochberg method.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co"># Table with DE analysis results</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="kw">head</span>(Brain)</a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="co">#&gt;    feature metadata            value      coef    stderr         pval</span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co">#&gt; 1   ENPP2  celltype oligodendrocytes  5.729342 0.3161434 2.113508e-73</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="co">#&gt; 2 SPARCL1  celltype oligodendrocytes -4.372554 0.2655240 6.266412e-61</span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co">#&gt; 3  SLC1A2  celltype oligodendrocytes -4.608658 0.2864684 3.104730e-58</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">#&gt; 4   GPR98  celltype oligodendrocytes -6.325255 0.4257395 6.257482e-50</span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="co">#&gt; 5 PIP4K2A  celltype oligodendrocytes  3.258072 0.2243366 8.646200e-48</span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="co">#&gt; 6    PLP1  celltype oligodendrocytes  4.685701 0.3292825 5.965336e-46</span></a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="co">#&gt;           qval base.model tweedie.index   N N.not.zero percent.zero</span></a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="co">#&gt; 1 2.113508e-71       ZICP         1.615 100         62           38</span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12"><span class="co">#&gt; 2 3.133206e-59       ZICP         1.643 100         87           13</span></a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="co">#&gt; 3 1.034910e-56       ZICP         1.595 100         80           20</span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="co">#&gt; 4 1.564371e-48       ZICP         1.577 100         72           28</span></a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="co">#&gt; 5 1.729240e-46       ZICP         1.671 100         85           15</span></a>
<a class="sourceLine" id="cb6-16" data-line-number="16"><span class="co">#&gt; 6 9.942227e-45       ZICP         1.672 100         60           40</span></a>
<a class="sourceLine" id="cb6-17" data-line-number="17"></a>
<a class="sourceLine" id="cb6-18" data-line-number="18"><span class="co"># How many significant at 5% FDR</span></a>
<a class="sourceLine" id="cb6-19" data-line-number="19"><span class="kw">nrow</span>(Brain[Brain<span class="op">$</span>qval<span class="op">&lt;</span><span class="fl">0.05</span>,]) <span class="co"># 85</span></a>
<a class="sourceLine" id="cb6-20" data-line-number="20"><span class="co">#&gt; [1] 85</span></a></code></pre></div>
<p>Using the adjusted p-values, there are 85 genes significant at 5% FDR.</p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">sessionInfo</span>()</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="co">#&gt; R version 3.6.3 (2020-02-29)</span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="co">#&gt; Platform: x86_64-apple-darwin15.6.0 (64-bit)</span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="co">#&gt; Running under: macOS  10.16</span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="co">#&gt; Matrix products: default</span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib</span></a>
<a class="sourceLine" id="cb7-8" data-line-number="8"><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib</span></a>
<a class="sourceLine" id="cb7-9" data-line-number="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb7-10" data-line-number="10"><span class="co">#&gt; locale:</span></a>
<a class="sourceLine" id="cb7-11" data-line-number="11"><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></a>
<a class="sourceLine" id="cb7-12" data-line-number="12"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb7-13" data-line-number="13"><span class="co">#&gt; attached base packages:</span></a>
<a class="sourceLine" id="cb7-14" data-line-number="14"><span class="co">#&gt; [1] stats4    parallel  stats     graphics  grDevices utils     datasets </span></a>
<a class="sourceLine" id="cb7-15" data-line-number="15"><span class="co">#&gt; [8] methods   base     </span></a>
<a class="sourceLine" id="cb7-16" data-line-number="16"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb7-17" data-line-number="17"><span class="co">#&gt; other attached packages:</span></a>
<a class="sourceLine" id="cb7-18" data-line-number="18"><span class="co">#&gt;  [1] scran_1.14.6                SingleCellExperiment_1.8.0 </span></a>
<a class="sourceLine" id="cb7-19" data-line-number="19"><span class="co">#&gt;  [3] SummarizedExperiment_1.16.1 DelayedArray_0.12.3        </span></a>
<a class="sourceLine" id="cb7-20" data-line-number="20"><span class="co">#&gt;  [5] BiocParallel_1.20.1         matrixStats_0.58.0         </span></a>
<a class="sourceLine" id="cb7-21" data-line-number="21"><span class="co">#&gt;  [7] GenomicRanges_1.38.0        GenomeInfoDb_1.22.1        </span></a>
<a class="sourceLine" id="cb7-22" data-line-number="22"><span class="co">#&gt;  [9] IRanges_2.20.2              S4Vectors_0.24.4           </span></a>
<a class="sourceLine" id="cb7-23" data-line-number="23"><span class="co">#&gt; [11] SC2P_1.0.2                  limma_3.42.2               </span></a>
<a class="sourceLine" id="cb7-24" data-line-number="24"><span class="co">#&gt; [13] Biobase_2.46.0              BiocGenerics_0.32.0        </span></a>
<a class="sourceLine" id="cb7-25" data-line-number="25"><span class="co">#&gt; [15] forcats_0.5.1               stringr_1.4.0              </span></a>
<a class="sourceLine" id="cb7-26" data-line-number="26"><span class="co">#&gt; [17] dplyr_1.0.7                 purrr_0.3.4                </span></a>
<a class="sourceLine" id="cb7-27" data-line-number="27"><span class="co">#&gt; [19] readr_1.4.0                 tidyr_1.1.3                </span></a>
<a class="sourceLine" id="cb7-28" data-line-number="28"><span class="co">#&gt; [21] tibble_3.1.3                ggplot2_3.3.5              </span></a>
<a class="sourceLine" id="cb7-29" data-line-number="29"><span class="co">#&gt; [23] tidyverse_1.3.1             Tweedieverse_0.0.1         </span></a>
<a class="sourceLine" id="cb7-30" data-line-number="30"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb7-31" data-line-number="31"><span class="co">#&gt; loaded via a namespace (and not attached):</span></a>
<a class="sourceLine" id="cb7-32" data-line-number="32"><span class="co">#&gt;  [1] ggbeeswarm_0.6.0         minqa_1.2.4              colorspace_2.0-2        </span></a>
<a class="sourceLine" id="cb7-33" data-line-number="33"><span class="co">#&gt;  [4] ellipsis_0.3.2           XVector_0.26.0           BiocNeighbors_1.4.2     </span></a>
<a class="sourceLine" id="cb7-34" data-line-number="34"><span class="co">#&gt;  [7] fs_1.5.0                 rstudioapi_0.13          fansi_0.5.0             </span></a>
<a class="sourceLine" id="cb7-35" data-line-number="35"><span class="co">#&gt; [10] lubridate_1.7.10         xml2_1.3.2               splines_3.6.3           </span></a>
<a class="sourceLine" id="cb7-36" data-line-number="36"><span class="co">#&gt; [13] logging_0.10-108         knitr_1.31               scater_1.14.6           </span></a>
<a class="sourceLine" id="cb7-37" data-line-number="37"><span class="co">#&gt; [16] jsonlite_1.7.2           broom_0.7.7              dbplyr_2.1.1            </span></a>
<a class="sourceLine" id="cb7-38" data-line-number="38"><span class="co">#&gt; [19] compiler_3.6.3           httr_1.4.2               dqrng_0.2.1             </span></a>
<a class="sourceLine" id="cb7-39" data-line-number="39"><span class="co">#&gt; [22] backports_1.2.1          assertthat_0.2.1         Matrix_1.2-18           </span></a>
<a class="sourceLine" id="cb7-40" data-line-number="40"><span class="co">#&gt; [25] cli_3.0.1                BiocSingular_1.2.2       htmltools_0.5.1.1       </span></a>
<a class="sourceLine" id="cb7-41" data-line-number="41"><span class="co">#&gt; [28] tools_3.6.3              rsvd_1.0.3               igraph_1.2.6            </span></a>
<a class="sourceLine" id="cb7-42" data-line-number="42"><span class="co">#&gt; [31] coda_0.19-4              gtable_0.3.0             glue_1.4.2              </span></a>
<a class="sourceLine" id="cb7-43" data-line-number="43"><span class="co">#&gt; [34] GenomeInfoDbData_1.2.2   reshape2_1.4.4           Rcpp_1.0.7              </span></a>
<a class="sourceLine" id="cb7-44" data-line-number="44"><span class="co">#&gt; [37] biglm_0.9-2.1            cellranger_1.1.0         vctrs_0.3.8             </span></a>
<a class="sourceLine" id="cb7-45" data-line-number="45"><span class="co">#&gt; [40] nlme_3.1-144             DelayedMatrixStats_1.8.0 xfun_0.20               </span></a>
<a class="sourceLine" id="cb7-46" data-line-number="46"><span class="co">#&gt; [43] rvest_1.0.0              lifecycle_1.0.0          irlba_2.3.3             </span></a>
<a class="sourceLine" id="cb7-47" data-line-number="47"><span class="co">#&gt; [46] statmod_1.4.33           edgeR_3.28.1             zlibbioc_1.32.0         </span></a>
<a class="sourceLine" id="cb7-48" data-line-number="48"><span class="co">#&gt; [49] scales_1.1.1             hms_1.0.0                yaml_2.2.1              </span></a>
<a class="sourceLine" id="cb7-49" data-line-number="49"><span class="co">#&gt; [52] pbapply_1.4-3            gridExtra_2.3            stringi_1.7.3           </span></a>
<a class="sourceLine" id="cb7-50" data-line-number="50"><span class="co">#&gt; [55] rlang_0.4.11             pkgconfig_2.0.3          bitops_1.0-6            </span></a>
<a class="sourceLine" id="cb7-51" data-line-number="51"><span class="co">#&gt; [58] cplm_0.7-8               evaluate_0.14            lattice_0.20-38         </span></a>
<a class="sourceLine" id="cb7-52" data-line-number="52"><span class="co">#&gt; [61] tidyselect_1.1.0         plyr_1.8.6               magrittr_2.0.1          </span></a>
<a class="sourceLine" id="cb7-53" data-line-number="53"><span class="co">#&gt; [64] R6_2.5.1                 generics_0.1.0           DBI_1.1.1               </span></a>
<a class="sourceLine" id="cb7-54" data-line-number="54"><span class="co">#&gt; [67] pillar_1.6.2             haven_2.3.1              withr_2.4.2             </span></a>
<a class="sourceLine" id="cb7-55" data-line-number="55"><span class="co">#&gt; [70] RCurl_1.98-1.2           modelr_0.1.8             crayon_1.4.1            </span></a>
<a class="sourceLine" id="cb7-56" data-line-number="56"><span class="co">#&gt; [73] utf8_1.2.2               rmarkdown_2.6            viridis_0.5.1           </span></a>
<a class="sourceLine" id="cb7-57" data-line-number="57"><span class="co">#&gt; [76] locfit_1.5-9.4           grid_3.6.3               readxl_1.3.1            </span></a>
<a class="sourceLine" id="cb7-58" data-line-number="58"><span class="co">#&gt; [79] reprex_2.0.0             digest_0.6.27            munsell_0.5.0           </span></a>
<a class="sourceLine" id="cb7-59" data-line-number="59"><span class="co">#&gt; [82] beeswarm_0.2.3           viridisLite_0.4.0        tweedie_2.3.3           </span></a>
<a class="sourceLine" id="cb7-60" data-line-number="60"><span class="co">#&gt; [85] vipor_0.4.5</span></a></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
